---
title: "Fungal community composition and genetic potential regulate fine root decay in northern temperate forests"
author: "William A. Argiroff, Donald R. Zak, Rima A. Upchurch, Peter T. Pellitier, and Julia P. Belke"
output: html_notebook
---

```{r load_packages}
## Load
library(tidyverse)

library(viridis)

library(geosphere)

library(phyloseq)

library(ggpubr)

library(vegan)

library(TITAN2)

library(mgcv)

library(ape)

library(cowplot)

#library(mgcv.helper)

library(psych)

library(scales)
```

```{r calculate_root_mass_loss}
## Calculate initial moisture content
decomp_root_initial <- read_tsv(file = "data/root_mass_loss/manistee_roots_decomposed_masses_july_2020.txt") %>% 
  
  ## Rename columns
  rename(
    
    PLOT = "Plot", 
    
    TIN.MASS = "Tin (g)", 
    
    FRESH.TIN.MASS = "Fresh + tin (g)", 
    
    DRY.TIN.MASS = "Dry + tin (g)"
    
  ) %>% 
  
  ## Calculate fresh and dry root mass, add measurement ID
  mutate(
    
    FRESH.ROOT.MASS = FRESH.TIN.MASS - TIN.MASS, 
    
    DRY.ROOT.MASS = DRY.TIN.MASS - TIN.MASS, 
    
    MEASUREMENT = "Initial"
    
  ) %>% 
  
  ## Trim table, remove test plots, and drop lost sample
  select(PLOT, MEASUREMENT, FRESH.ROOT.MASS, DRY.ROOT.MASS) %>% 
  
  filter(
    
    PLOT != "73_A" & 
      
      PLOT != "73_B" & 
      
      PLOT != "74_A" & 
      
      PLOT != "74_B"
    
  ) %>% 
  
  mutate(
    
    PLOT = paste("Plot", PLOT, sep = "_"), 
    
    FRESH.ROOT.MASS = ifelse(PLOT == "Plot_57", 0, FRESH.ROOT.MASS), 
    
    DRY.ROOT.MASS = ifelse(PLOT == "Plot_57", 0, DRY.ROOT.MASS)
    
  )

## Read in additional moisture data
decomp_root_moisture <- read_tsv(file = "data/root_mass_loss/decomposed_roots_additional_moisture_data.txt") %>% 
  
  ## Calculate fresh and dry root mass, add measurement ID
  mutate(
    
    FRESH.ROOT.MASS = FRESH.ROOTS.TIN - TIN, 
    
    DRY.ROOT.MASS = DRY.ROOTS.TIN - TIN, 
    
    MEASUREMENT = "Additional"
    
  ) %>% 
  
  ## Trim table and format plots
  select(PLOT, MEASUREMENT, FRESH.ROOT.MASS, DRY.ROOT.MASS) %>% 
  
  mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  
  ## Add initial moisture and calculate combined masses
  bind_rows(decomp_root_initial, .) %>% 
  
  group_by(PLOT) %>% 
  
  summarise(
    
    FRESH.ROOT.COMBINED.MASS = sum(FRESH.ROOT.MASS), 
    
    DRY.ROOT.COMBINED.MASS = sum(DRY.ROOT.MASS)
    
  ) %>% 
  
  ungroup() %>% 
  
  # Calculate moisture content
  mutate(MOISTURE.CONTENT = (FRESH.ROOT.COMBINED.MASS - DRY.ROOT.COMBINED.MASS) / FRESH.ROOT.COMBINED.MASS) %>% 
  
  select(PLOT, MOISTURE.CONTENT)

## Correct for ash content

## Read in pre-incubated ash data
pre_incubated_ash <- read_tsv(file = "data/root_mass_loss/pre_incubated_root_ash_data.txt") %>% 
  
  ## Split sample ID into replicate and site, format stand
  separate(col = SAMPLE, into = c("SITE", "REPL")) %>% 
  
  mutate(SITE = paste("Site", SITE, sep = "_")) %>% 
  
  ## Calculate ash-free proportion as site means
  mutate(
    
    ASH.FREE.PROP = ((ROOT.CRUCIBLE - EMPTY.CRUCIBLE) - (ROOT.CRUCIBLE.ASHED - EMPTY.CRUCIBLE)) / 
      (ROOT.CRUCIBLE - EMPTY.CRUCIBLE)
    
  ) %>% 
  
  select(SITE, REPL, ASH.FREE.PROP) %>% 
  
  ## Format sample ID
  unite(SITE : REPL, col = SAMPLE.ID, sep = "_", remove = FALSE) %>% 
  
  mutate(SAMPLE.ID = gsub("Site_", "", SAMPLE.ID))

## Calculate site means, pre-incubated roots
pre_incubated_ash_mean <- pre_incubated_ash %>% 
  
  group_by(SITE) %>% 
  
  summarise(MEAN.ASH.FREE.PROP = mean(ASH.FREE.PROP)) %>% 
  
  ungroup()

## Read in decomposing root ash data
decomp_ash <- readr::read_tsv(file = "data/root_mass_loss/incubated_root_ash_data.txt") %>% 
  
  ## Reformat plot column
  rename(PLOT = "SAMPLE") %>% 
  
  mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  
  ## Calculate ash-free proportion, trim table
  mutate(ASH.FREE.PROP = ((ROOT.CRUCIBLE - EMPTY.CRUCIBLE) - (ROOT.CRUCIBLE.ASHED - EMPTY.CRUCIBLE)) / 
           (ROOT.CRUCIBLE - EMPTY.CRUCIBLE)) %>% 
  
  select(PLOT, ASH.FREE.PROP)

## Read in pre-incubated root masses
pre_incubated_ash_free_dry_mass <- read_tsv(file = "data/root_mass_loss/manistee_undecomposed_root_masses.txt") %>% 
  
  ## Format plot and site, drop test roots
  rename(
    
    PLOT = "Plot", 
    
    SITE = "Stand", 
    
    ROOT.UNCORRECTED.MASS = "Root mass, mg"
    
  ) %>% 
  
  filter(
    
    PLOT != "73" & 
      
      PLOT != "73_A" & 
      
      PLOT != "73_B" & 
      
      PLOT != "74" & 
      
      PLOT != "74_A" & 
      
      PLOT != "74_B"
    
  ) %>% 
  mutate(
    PLOT = paste("Plot", PLOT, sep = "_"), 
    
    SITE = paste("Site", SITE, sep = "_")
    
  ) %>% 
  
  ## Calculate ash-free dry mass, trim table, combine by plot
  inner_join(., pre_incubated_ash_mean, by = "SITE") %>% 
  
  mutate(PRE.INCUBATED.ASH.FREE.DRY.MASS = (ROOT.UNCORRECTED.MASS * MEAN.ASH.FREE.PROP) / 1000) %>% 
  
  select(
    PLOT, 
    
    SITE, 
    
    PRE.INCUBATED.ASH.FREE.DRY.MASS
    
  ) %>% 
  
  group_by(SITE, PLOT) %>% 
  
  
  summarise(PRE.INCUBATED.ROOT.MASS = sum(PRE.INCUBATED.ASH.FREE.DRY.MASS)) %>% 
  
  ungroup()

# Read in decomposed root masses
decomp_ash_free_dry_mass <- read_tsv(file = "data/root_mass_loss/manistee_roots_decomposed_masses_july_2020.txt") %>% 
  
  # Rename and format columns, and trim table
  rename(
    PLOT = "Plot", 
    
    FRESH.MASS = "Total mass (g)"
  ) %>% 
  
  select(PLOT, FRESH.MASS) %>% 
  
  filter(
    
    PLOT != "73_A" & 
      
      PLOT != "73_B" & 
      
      PLOT != "74_A" & 
      
      PLOT != "74_B"
    
  ) %>% 
  
  mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  
  ## Calculate dry mass
  inner_join(., decomp_root_moisture, by = "PLOT") %>% 
  
  mutate(DECOMP.DRY.MASS = FRESH.MASS * (1 - MOISTURE.CONTENT)) %>% 
  
  ## Calculate ash-free dry mass, trim table
  inner_join(., decomp_ash, by = "PLOT") %>% 
  
  mutate(DECOMP.ROOT.MASS = DECOMP.DRY.MASS * ASH.FREE.PROP) %>% 
  
  select(PLOT, DECOMP.ROOT.MASS)

## Calculate mass loss and remove outlier
root_mass_loss <- decomp_ash_free_dry_mass %>% 
  
  ## Combine masses, calculate mass loss, and trim table
  inner_join(pre_incubated_ash_free_dry_mass, ., by = "PLOT") %>% 
  
  mutate(PROP.MASS.LOSS = (PRE.INCUBATED.ROOT.MASS - DECOMP.ROOT.MASS) / PRE.INCUBATED.ROOT.MASS) %>% 
  
  select(PLOT, PROP.MASS.LOSS)
```

```{r calculate_decomp_root_moisture}
# Read in initial moisture data
decomp.root.initial.tbl <- read_tsv(file = "data/root_mass_loss/manistee_roots_decomposed_masses_july_2020.txt") %>% 
  
  # Rename columns
  dplyr::rename(PLOT = "Plot", 
                TIN.MASS = "Tin (g)", 
                FRESH.TIN.MASS = "Fresh + tin (g)", 
                DRY.TIN.MASS = "Dry + tin (g)") %>% 
  
  # Calculate fresh and dry root mass, add measurement ID
  dplyr::mutate(FRESH.ROOT.MASS = FRESH.TIN.MASS - TIN.MASS, 
                DRY.ROOT.MASS = DRY.TIN.MASS - TIN.MASS, 
                MEASUREMENT = "Initial") %>% 
  
  # Trim table, remove test plots, and drop lost sample
  dplyr::select(PLOT, 
                MEASUREMENT, 
                FRESH.ROOT.MASS, 
                DRY.ROOT.MASS) %>% 
  dplyr::filter(PLOT != "73_A" & 
                  PLOT != "73_B" & 
                  PLOT != "74_A" & 
                  PLOT != "74_B") %>% 
  dplyr::mutate(PLOT = paste("Plot", PLOT, sep = "_"), 
                FRESH.ROOT.MASS = ifelse(PLOT == "Plot_57", 0, FRESH.ROOT.MASS), 
                DRY.ROOT.MASS = ifelse(PLOT == "Plot_57", 0, DRY.ROOT.MASS))

# Read in additional moisture data
decomp_root_moisture <- readr::read_tsv(file = "data/root_mass_loss/decomposed_roots_additional_moisture_data.txt") %>% 
  
  # Calculate fresh and dry root mass, add measurement ID
  dplyr::mutate(FRESH.ROOT.MASS = FRESH.ROOTS.TIN - TIN, 
                DRY.ROOT.MASS = DRY.ROOTS.TIN - TIN, 
                MEASUREMENT = "Additional") %>% 
  
  # Trim table and format plots
  dplyr::select(PLOT, 
                MEASUREMENT, 
                FRESH.ROOT.MASS, 
                DRY.ROOT.MASS) %>% 
  dplyr::mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  
  # Add initial moisture and calculate combined masses
  dplyr::bind_rows(decomp.root.initial.tbl, .) %>% 
  dplyr::group_by(PLOT) %>% 
  dplyr::summarise(FRESH.ROOT.COMBINED.MASS = sum(FRESH.ROOT.MASS), 
                   DRY.ROOT.COMBINED.MASS = sum(DRY.ROOT.MASS)) %>% 
  dplyr::ungroup() %>% 
  
  # Calculate moisture content
  dplyr::mutate(MOISTURE.CONTENT = (FRESH.ROOT.COMBINED.MASS - DRY.ROOT.COMBINED.MASS) / FRESH.ROOT.COMBINED.MASS) %>% 
  dplyr::select(PLOT, 
                MOISTURE.CONTENT)
```

```{r format_aq2_data}
#### 2. Format soil moisture data ####

# Read in soil moisture data
soil.moisture.tbl <- readr::read_tsv(file = "data/root_and_soil_chem/spring_2019_soil_moisture.txt") %>% 
  
  # Calculate percent moisture
  dplyr::mutate(GRAV = (FRESH - (DRY.TOT - TIN )) * (1 / FRESH)) %>% 
  
  # Calculate mean by plot
  dplyr::group_by(STAND, PLOT) %>% 
  dplyr::summarize(GRAV = mean(GRAV)) %>% 
  dplyr::ungroup() %>% 
  
  # Rename stands and plots, trim table
  dplyr::mutate(STAND = paste("Stand", STAND, sep = "_"), 
                PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  dplyr::select(-STAND)

# Write out data
# readr::write_tsv(soil.moisture.tbl, path = "data/working/spring_2019_soil_moisture.txt")

#### 3. Format extraction and incubation masses ####

# Read in extraction and incubation fresh masses
extrac.incub.mass.tbl <- readr::read_tsv("data/root_and_soil_chem/spring_2019_N_mineralization_incubation_and_pre_extraction_mass.txt") %>% 
  
  # Select only fresh masses, format columns
  dplyr::filter(TYPE == "F") %>% 
  dplyr::mutate(STAND = paste("Stand", STAND, sep = "_"), 
                PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  
  # Calculate dry masses of soil for pre- and post- extraction samples
  dplyr::inner_join(., soil.moisture.tbl, by = "PLOT") %>% 
  mutate(PRE.INCUB.EXTRACT.DRY.MASS = (1 - GRAV) * PRE.INCUB.EXTRACT.MASS) %>% 
  mutate(POST.INCUB.EXTRACT.DRY.MASS = (1 - GRAV) * INCUB.MASS) %>% 
  # Add column
  mutate(POST.INCUB.EXTRACT.MASS = rep(NA, length(INCUB.MASS))) %>% 
  # Update masses
  mutate(POST.INCUB.EXTRACT.MASS = ifelse(ADD.H2O == "Yes", INCUB.MASS.H2O.ADDED, INCUB.MASS)) %>% 
  
  # Calculate adjusted gravimetric water content and trim table
  dplyr::mutate(GRAV.CORR = (POST.INCUB.EXTRACT.MASS - POST.INCUB.EXTRACT.DRY.MASS) * (1 / POST.INCUB.EXTRACT.MASS)) %>% 
  dplyr::select(PLOT, 
                GRAV, 
                GRAV.CORR, 
                PRE.INCUB.EXTRACT.DRY.MASS, 
                POST.INCUB.EXTRACT.DRY.MASS)

# Read in incubation times
incub.schedule.tbl <- readr::read_tsv("data/root_and_soil_chem/spring_2019_N_mineralization_incubation_schedule.txt") %>% 
  
  # Rename columns abd trim
  dplyr::rename(STAND = "Stand", 
                PLOT = "Plot", 
                START.DATE = "Date started", 
                START.TIME = "Time started", 
                END.DATE = "Date finished", 
                END.TIME = "Rounded time finished") %>% 
  dplyr::filter(Type == "F") %>% 
  
  # Combine dates and times, calculate time intervals
  dplyr::mutate(START = paste(START.DATE, START.TIME, sep = " "), 
                END = paste(END.DATE, END.TIME, sep = " ")) %>% 
  dplyr::mutate(START = as.POSIXct(strptime(START, format = "%Y-%m-%d %H:%M", tz = "EST5EDT")), 
                END = as.POSIXct(strptime(END, format = "%Y-%m-%d %H:%M", tz = "EST5EDT"))) %>% 
  # Time interval
  dplyr::mutate(TIME.INTERVAL = difftime(END, START, tz = "EST5EDT", units = "days")) %>% 
  dplyr::mutate(TIME.INTERVAL = as.numeric(TIME.INTERVAL)) %>% 
  
  # Rename stands and plots, trim table
  dplyr::mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  dplyr::select(PLOT, 
                TIME.INTERVAL)

#### 4. Format AQ2 data ####

# Calculate blank 1 correction factor
aq2.blank.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_23.csv") %>% 
  
  # Redo test column, format standards and blanks
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mg L" & 
                  `Sample ID` != "NO3 1 mg L" & 
                  `Sample ID` != "KCl analytical") %>% 
  
  # Remove unnecessary columns, format other columns
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Select blanks and calculate mean values
  dplyr::filter(SAMPLE.ID == "KCl blank 1 m" | 
                  SAMPLE.ID == "KCl blank 1 n" | 
                  SAMPLE.ID == "KCl blank 2 m" | 
                  SAMPLE.ID == "KCl blank 1 o" | 
                  SAMPLE.ID == "KCl blank 1test a" | 
                  SAMPLE.ID == "KCl blank 1test b" | 
                  SAMPLE.ID == "KCl blank 1test c" | 
                  SAMPLE.ID == "KCl blank 1test d") %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = " ", convert = TRUE) %>% 
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  dplyr::summarise(RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup()

# Calculate ammonium blank ratio
amm.blank.ratio <- unlist(aq2.blank.tbl[aq2.blank.tbl$EXTRAC == "1test" & aq2.blank.tbl$TEST == "Ammonium", "RESULTS"], use.names = FALSE) / unlist(aq2.blank.tbl[aq2.blank.tbl$EXTRAC == "1" & aq2.blank.tbl$TEST == "Ammonium", "RESULTS"], use.names = FALSE)

# Calculate nitrate blank ratio
nit.blank.ratio <- unlist(aq2.blank.tbl[aq2.blank.tbl$EXTRAC == "1test" & aq2.blank.tbl$TEST == "Nitrate", "RESULTS"], use.names = FALSE) / unlist(aq2.blank.tbl[aq2.blank.tbl$EXTRAC == "1" & aq2.blank.tbl$TEST == "Nitrate", "RESULTS"], use.names = FALSE)

# Get blanks, re-run
aq2.7.blanks.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_23.csv") %>% 
  
  # Redo test column, format standards and blanks
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mgL" & 
                  `Sample ID` != "NO3 1 mgL" & 
                  `Sample ID` != "KCl analytical") %>% 
  
  # Reformat columns
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Get blanks only, calculate mean blank values
  dplyr::filter(SAMPLE.ID == "KCl blank 1 m" | 
                  SAMPLE.ID == "KCl blank 1 n" | 
                  SAMPLE.ID == "KCl blank 1 o" | 
                  SAMPLE.ID == "KCl blank 2 m") %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\ ", ".", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_", convert = TRUE) %>% 
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  dplyr::summarise(RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup()

# Read in AQ2 data, re-run
aq2.7.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_23.csv") %>% 
  
  # Redo test column and format blanks
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  filter(`Sample ID` != "Standard 1" & 
           `Sample ID` != "Standard 90" & 
           `Sample ID` != "Standard 91" & 
           `Sample ID` != "Standard 92" & 
           `Sample ID` != "Standard 93" & 
           `Sample ID` != "Standard 94" & 
           `Sample ID` != "Standard 0" & 
           `Sample ID` != "CCV" & 
           `Sample ID` != "CCB" & 
           `Sample ID` != "NO2 1 mgL" & 
           `Sample ID` != "NO3 1 mgL" & 
           `Sample ID` != "KCl analytical") %>% 
  
  # Format columns
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Keep only samples that were re-run
  dplyr::filter((SAMPLE.ID %in% c("22.31.post.c", "22.31.post.d", "24.37.post.c", "24.37.post.d", 
                                  "41.49.post.c", "41.49.post.d", "100.67.pre.c", "100.67.pre.d",
                                  "100.67.post.c", "100.67.post.d", "50.55.pre.c", "50.55.pre.d",
                                  "50.55.post.c", "50.55.post.d", "58.61.pre.c", "58.61.pre.d", 
                                  "58.61.post.c", "58.61.post.d"))) %>% 
  
  # Correct for blanks
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_") %>% 
  dplyr::mutate(KCl.BLANK = rep(NA, length(RESULTS))) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), 
                                   unlist(aq2.7.blanks.tbl[aq2.7.blanks.tbl$EXTRAC == "1" & aq2.7.blanks.tbl$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Ammonium"), 
                                   unlist(aq2.7.blanks.tbl[aq2.7.blanks.tbl$EXTRAC == "2" & aq2.7.blanks.tbl$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), 
                                   unlist(aq2.7.blanks.tbl[aq2.7.blanks.tbl$EXTRAC == "1" & aq2.7.blanks.tbl$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Nitrate"), 
                                   unlist(aq2.7.blanks.tbl[aq2.7.blanks.tbl$EXTRAC == "2" & aq2.7.blanks.tbl$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK))

# Get blanks, run 1
aq2.1.blanks.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_15.csv") %>% 
  
  # Format blanks
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mgL" & 
                  `Sample ID` != "NO3 1 mgL" & 
                  `Sample ID` != "KCl analytical") %>% 
  
  # Format columns
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Correct for blank values
  dplyr::filter(SAMPLE.ID == "KCl blank 1 a" | 
                  SAMPLE.ID == "KCl blank 2 a" | 
                  SAMPLE.ID == "KCl blank 2 b") %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\ ", ".", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_", convert = TRUE) %>% 
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  dplyr::summarise(RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup()

# Read in AQ2 data, run 1
aq2.1.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_15.csv") %>% 
  
  # Format columns
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mgL" & 
                  `Sample ID` != "NO3 1 mgL" & 
                  `Sample ID` != "KCl analytical") %>% 
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Remove samples that needed to be re-run
  dplyr::filter(!((SAMPLE.ID %in% c("22.31.post.a", "22.31.post.b", "24.37.post.a", "24.37.post.b", 
                                    "41.49.post.a", "41.49.post.b", "KCl blank 1 b", "100.67.pre.b", 
                                    "100.67.post.b", "100.67.post.a", "50.55.pre.b", "50.55.post.b", 
                                    "58.61.pre.b", "58.61.post.b")) & (TEST == "Nitrate"))) %>% 
  
  # Update blanks
  dplyr::mutate(SAMPLE.ID = gsub("\\ ", ".", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_", convert = TRUE) %>% 
  
  # Add column of KCl blanks
  dplyr::mutate(KCl.BLANK = rep(NA, length(RESULTS))) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), 
                                   unlist(aq2.1.blanks.tbl[aq2.1.blanks.tbl$EXTRAC == "1" & aq2.1.blanks.tbl$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Ammonium"), 
                                   unlist(aq2.1.blanks.tbl[aq2.1.blanks.tbl$EXTRAC == "2" & aq2.1.blanks.tbl$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), 
                                   unlist(aq2.1.blanks.tbl[aq2.1.blanks.tbl$EXTRAC == "1" & aq2.1.blanks.tbl$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Nitrate"), 
                                   unlist(aq2.1.blanks.tbl[aq2.1.blanks.tbl$EXTRAC == "2" & aq2.1.blanks.tbl$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::filter(STAND != "KCl") %>% 
  
  # Add re-run samples
  dplyr::bind_rows(., aq2.7.tbl) %>% 
  
  # Correct for blank readings
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), KCl.BLANK * amm.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), KCl.BLANK * nit.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.FINAL = ifelse((EXTRAC == "pre"), KCl.BLANK.CORR, KCl.BLANK)) %>% 
  dplyr::mutate(CORR.RESULTS = RESULTS - KCl.BLANK.FINAL) %>% 
  
  # Calculate sample means
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  dplyr::summarise(CORR.RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup() %>% 
  
  # Convert negative values to 0
  dplyr::mutate(CORR.RESULTS = ifelse(CORR.RESULTS < 0, 0, CORR.RESULTS))

# Read in AQ2 data, run 2
aq2.2.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_16.csv") %>% 
  
  # Format columns
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mg L" & 
                  `Sample ID` != "NO3 1 mg L" & 
                  `Sample ID` != "KCl analytical") %>% 
  
  # Format columns
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Update blanks
  dplyr::mutate(SAMPLE.ID = gsub("\\ ", ".", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_", convert = TRUE) %>% 
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  
  # Calculate mean values by sample
  dplyr::summarise(RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup() %>% 
  
  # Create a new column for KCl values
  dplyr::mutate(KCl.BLANK = rep(NA, length(RESULTS))) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::filter(STAND != "KCl") %>% 
  
  # Correct blank readings
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), KCl.BLANK * amm.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), KCl.BLANK * nit.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.FINAL = ifelse((EXTRAC == "pre"), KCl.BLANK.CORR, KCl.BLANK)) %>% 
  dplyr::mutate(CORR.RESULTS = RESULTS - KCl.BLANK.FINAL) %>% 
  
  # Convert negative values to 0
  dplyr::mutate(CORR.RESULTS = ifelse(CORR.RESULTS < 0, 0, CORR.RESULTS)) %>% 
  dplyr::select(STAND, 
                PLOT, 
                EXTRAC, 
                TEST, 
                CORR.RESULTS)

# Read in AQ2 data, run 3
aq2.3.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_17.csv") %>% 
  
  # Format columns
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mg L" & 
                  `Sample ID` != "NO3 1 mg L" & 
                  `Sample ID` != "KCl analytical") %>% 
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  rename(SAMPLE.ID = `Sample ID`, 
         RESULTS = "Results") %>% 
  
  # Update blanks
  dplyr::mutate(SAMPLE.ID = gsub("\\ ", ".", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_", convert = TRUE) %>% 
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  
  # Calculate mean values by sample
  dplyr::summarise(RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup() %>% 
  
  # Create a new column for KCl values
  dplyr::mutate(KCl.BLANK = rep(NA, length(RESULTS))) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::filter(STAND != "KCl") %>% 
  
  # Correct for blank readings
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), KCl.BLANK * amm.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), KCl.BLANK * nit.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.FINAL = ifelse((EXTRAC == "pre"), KCl.BLANK.CORR, KCl.BLANK)) %>% 
  dplyr::mutate(CORR.RESULTS = RESULTS - KCl.BLANK.FINAL) %>% 
  
  # Convert negative values to 0
  dplyr::mutate(CORR.RESULTS = ifelse(CORR.RESULTS < 0, 0, CORR.RESULTS)) %>% 
  dplyr::select(STAND, 
                PLOT, 
                EXTRAC, 
                TEST, 
                CORR.RESULTS)

# Read in AQ2 data, run 4
aq2.4.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_18.csv") %>% 
  
  # Format columns
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mg L" & 
                  `Sample ID` != "NO3 1 mg L" & 
                  `Sample ID` != "KCl analytical") %>% 
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Calculate mean values by sample
  dplyr::mutate(SAMPLE.ID = gsub("\\ ", ".", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_", convert = TRUE) %>% 
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  dplyr::summarise(RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup() %>% 
  
  # Create a new column for KCl values
  dplyr::mutate(KCl.BLANK = rep(NA, length(RESULTS))) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::filter(STAND != "KCl") %>% 
  
  # Correct for blank readings
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), KCl.BLANK * amm.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), KCl.BLANK * nit.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.FINAL = ifelse((EXTRAC == "pre"), KCl.BLANK.CORR, KCl.BLANK)) %>% 
  dplyr::mutate(CORR.RESULTS = RESULTS - KCl.BLANK.FINAL) %>% 
  
  # Convert negative values to 0
  dplyr::mutate(CORR.RESULTS = ifelse(CORR.RESULTS < 0, 0, CORR.RESULTS))  %>% 
  dplyr::select(STAND, 
                PLOT, EXTRAC, 
                TEST, 
                CORR.RESULTS)

# Read in AQ2 data, run 5
aq2.5.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_19.csv") %>% 
  
  # Format columns
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mg L" & 
                  `Sample ID` != "NO3 1 mg L" & 
                  `Sample ID` != "KCl analytical") %>% 
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Update blanks and calculate means by sample
  dplyr::mutate(SAMPLE.ID = gsub("\\ ", ".", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_", convert = TRUE) %>% 
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  dplyr::summarise(RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup() %>% 
  
  # Create a new column for KCl values
  dplyr::mutate(KCl.BLANK = rep(NA, length(RESULTS))) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::filter(STAND != "KCl") %>% 
  
  # Correct blank readings
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), KCl.BLANK * amm.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), KCl.BLANK * nit.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.FINAL = ifelse((EXTRAC == "pre"), KCl.BLANK.CORR, KCl.BLANK)) %>% 
  dplyr::mutate(CORR.RESULTS = RESULTS - KCl.BLANK.FINAL) %>% 
  
  # Convert negative values to 0
  dplyr::mutate(CORR.RESULTS = ifelse(CORR.RESULTS < 0, 0, CORR.RESULTS))  %>% 
  dplyr::select(STAND, 
                PLOT, 
                EXTRAC, 
                TEST, 
                CORR.RESULTS)

# Read in AQ2 data, run 6
aq2.6.tbl <- readr::read_csv("data/root_and_soil_chem/WA_aq2_run_from_19_07_22.csv") %>% 
  
  # Format columns
  dplyr::mutate(TEST = ifelse(Test == "Ammonia 10 KCl", "Ammonium", "Nitrate")) %>% 
  dplyr::filter(`Sample ID` != "Standard 1" & 
                  `Sample ID` != "Standard 90" & 
                  `Sample ID` != "Standard 91" & 
                  `Sample ID` != "Standard 92" & 
                  `Sample ID` != "Standard 93" & 
                  `Sample ID` != "Standard 94" & 
                  `Sample ID` != "Standard 0" & 
                  `Sample ID` != "CCV" & 
                  `Sample ID` != "CCB" & 
                  `Sample ID` != "NO2 1 mg L" & 
                  `Sample ID` != "NO3 1 mg L" & 
                  `Sample ID` != "KCl analytical") %>% 
  dplyr::select(-`Sample Details`, -Test, -Units, -Absorbance, -`QC Pro result`, -Operator, 
                -`Man Dil Factor`, -`Auto Dil Factor`, -`Date and Time`) %>% 
  dplyr::rename(SAMPLE.ID = `Sample ID`, 
                RESULTS = "Results") %>% 
  
  # Update blanks
  dplyr::mutate(SAMPLE.ID = gsub("\\ ", ".", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\.", "_", SAMPLE.ID)) %>% 
  tidyr::separate(SAMPLE.ID, into = c("STAND", "PLOT", "EXTRAC", "REPLICATE"), sep = "_", convert = TRUE) %>% 
  dplyr::group_by(STAND, PLOT, EXTRAC, TEST) %>% 
  dplyr::summarise(RESULTS = mean(RESULTS)) %>% 
  dplyr::ungroup() %>% 
  
  # Create a new column for KCl values
  dplyr::mutate(KCl.BLANK = rep(NA, length(RESULTS))) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Ammonium"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Ammonium", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "1" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK = ifelse((EXTRAC == "post" & TEST == "Nitrate"), 
                                   unlist(.[.$EXTRAC == "2" & .$TEST == "Nitrate", "RESULTS"], use.names = FALSE), 
                                   KCl.BLANK)) %>% 
  dplyr::filter(STAND != "KCl") %>% 
  
  # Correct for blank readings
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Ammonium"), KCl.BLANK * amm.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.CORR = ifelse((EXTRAC == "pre" & TEST == "Nitrate"), KCl.BLANK * nit.blank.ratio, KCl.BLANK)) %>% 
  dplyr::mutate(KCl.BLANK.FINAL = ifelse((EXTRAC == "pre"), KCl.BLANK.CORR, KCl.BLANK)) %>% 
  dplyr::mutate(CORR.RESULTS = RESULTS - KCl.BLANK.FINAL) %>% 
  
  # Convert negative values to 0
  dplyr::mutate(CORR.RESULTS = ifelse(CORR.RESULTS < 0, 0, CORR.RESULTS))  %>% 
  dplyr::select(STAND, 
                PLOT, 
                EXTRAC, 
                TEST, 
                CORR.RESULTS)
```

```{r se_function}
#### Standard error function ####
se <- function(x) {
  
  ## Calculate se
  tmp1 <- sd(x) * (1 / sqrt(length(x)))
  
  ## Return the result
  return(tmp1)
  
}
```

```{r calculate_soil_inorganic_n}
# Combine data from all runs
soil_inorg_N <- bind_rows(aq2.1.tbl, aq2.2.tbl, aq2.3.tbl, aq2.4.tbl, aq2.5.tbl, aq2.6.tbl) %>% 
  
  # Calculate total inorganic N by plot and extraction
  group_by(STAND, PLOT, EXTRAC) %>% 
  summarise(INORG.N = sum(CORR.RESULTS)) %>% 
  ungroup() %>% 
  
  # Wide format, and format columns
  pivot_wider(names_from = "EXTRAC", values_from = "INORG.N") %>% 
  
  rename(
    
    PRE.INCUB.N = "pre", 
    
    POST.INCUB.N = "post"
    
  ) %>% 
  
  select(PLOT, PRE.INCUB.N, POST.INCUB.N) %>% 
  
  # Update plot names
  mutate(PLOT = as.numeric(PLOT)) %>% 
  mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  
  # Add extraction masses, time intervals, and calculate N concentrations per g soil
  inner_join(., extrac.incub.mass.tbl, by = "PLOT") %>% 
  inner_join(., incub.schedule.tbl, by = "PLOT") %>% 
  mutate(
    
    PRE.N.PER.MASS.SOIL = PRE.INCUB.N * (1 / 1000) * 60 * (1 / PRE.INCUB.EXTRACT.DRY.MASS) * 1000, 
    POST.N.PER.MASS.SOIL = POST.INCUB.N * (1 / 1000) * 60 * (1 / POST.INCUB.EXTRACT.DRY.MASS) * 1000
  ) %>% 
  
  # Calculate mineralization rate
  mutate(N.MIN = (POST.N.PER.MASS.SOIL - PRE.N.PER.MASS.SOIL) * (1 / TIME.INTERVAL)) %>% 
  rename(N.INORG = "PRE.N.PER.MASS.SOIL") %>% 
  select(PLOT, N.MIN, N.INORG)
```

```{r calculate_soil_and_root_c_and_n}
# Read in pre-incubated ash data (from "calculate_root_mass_loss.R" script) 
pre.incubated.ash.tbl <- pre_incubated_ash %>% 
  
  # Format sample ID
  tidyr::separate(col = SITE, 
                  into = c("PREFIX", "SITE")) %>% 
  dplyr::select(-PREFIX) %>% 
  tidyr::unite(SITE : REPL, 
               col = SAMPLE.ID, 
               sep = "_", 
               remove = TRUE)

# Read in root LECO data
pre_root_C_and_N <- readr::read_csv(file = "data/root_and_soil_chem/Zak_WA_pre_roots_3-26-19_1.csv") %>% 
  
  # Format sample ID
  dplyr::filter(Project == "Zak_WA_pre_roots") %>% 
  dplyr::select(-Project, 
                -Mass, 
                -Date_time) %>% 
  tidyr::separate(col = Sample, 
                  into = c("SITE", "REPL", "TECH.REP")) %>% 
  tidyr::unite(SITE : REPL, 
               col = SAMPLE.ID, 
               sep = "_", 
               remove = TRUE) %>% 
  
  # Average technical replicates, and correct N and C for ash free dry mass
  dplyr::group_by(SAMPLE.ID) %>% 
  dplyr::summarise(CARBON = mean(Carbon), 
                   NITROGEN = mean(Nitrogen)) %>% 
  dplyr::ungroup() %>% 
  dplyr::inner_join(., pre.incubated.ash.tbl, by = "SAMPLE.ID") %>% 
  dplyr::mutate(PRE.ROOT.C = (((CARBON / ASH.FREE.PROP) * 1 / 100) * 1000), 
                PRE.ROOT.N = (((NITROGEN / ASH.FREE.PROP) * 1 / 100) * 1000)) %>% 
  
  # Calculate C/N and trim table
  dplyr::mutate(PRE.ROOT.CN = PRE.ROOT.C / PRE.ROOT.N) %>% 
  dplyr::select(SAMPLE.ID, 
                PRE.ROOT.C, 
                PRE.ROOT.N, 
                PRE.ROOT.CN)

# Get site means
pre_root_C_and_N_mean <- pre_root_C_and_N %>% 
  
  # Get sites
  tidyr::separate(col = SAMPLE.ID, 
                  into = c("SITE", "REPL"), 
                  sep = "_") %>% 
  dplyr::mutate(SITE = paste("Site", SITE, sep = "_")) %>% 
  
  # Calculate site means and sd
  dplyr::group_by(SITE) %>% 
  dplyr::summarise(PRE.ROOT.C.MEAN = mean(PRE.ROOT.C), 
                   PRE.ROOT.C.SE = se(PRE.ROOT.C), 
                   PRE.ROOT.N.MEAN = mean(PRE.ROOT.N), 
                   PRE.ROOT.N.SE = se(PRE.ROOT.N), 
                   PRE.ROOT.CN.MEAN = mean(PRE.ROOT.CN), 
                   PRE.ROOT.CN.SE = se(PRE.ROOT.CN)) %>% 
  dplyr::group_by()

#### 3. Calculate C and N corrected for ash free dry mass for decomposing roots ####

# Read in decomposing root LECO data
decomp_root_C_and_N <- readr::read_tsv(file = "data/root_and_soil_chem/Zak_WA_post_roots.txt") %>% 
  
  # Format columns
  dplyr::filter(`Sample Group` == "Zak- WA - Manistee Post Roots") %>% 
  dplyr::rename(PLOT = "Sample ID", 
                NITROGEN = "%N", 
                CARBON = "%C") %>% 
  dplyr::select(PLOT, 
                CARBON, 
                NITROGEN) %>% 
  dplyr::mutate(PLOT = gsub("\\ ", "_", PLOT)) %>% 
  
  # Correct for C and N for ash free dry mass (from "calculate_root_mass_loss.R" script)
  dplyr::inner_join(., decomp_ash, by = "PLOT") %>% 
  dplyr::mutate(DECOMP.ROOT.C = (((CARBON / ASH.FREE.PROP) * 1 / 100) * 1000), 
                DECOMP.ROOT.N = (((NITROGEN / ASH.FREE.PROP) * 1 / 100) * 1000)) %>% 
  
  # Calculate C/N and trim table
  dplyr::mutate(DECOMP.ROOT.CN = DECOMP.ROOT.C / DECOMP.ROOT.N) %>% 
  dplyr::select(PLOT, 
                DECOMP.ROOT.C, 
                DECOMP.ROOT.N, 
                DECOMP.ROOT.CN)

#### 4. Format soil C and N data ####

# Read in soil leco data 1
soil_CN_1.tbl <- readr::read_csv(file = "data/root_and_soil_chem/WA_LECO_run_from_19_07_25.csv", 
                                 col_names = FALSE) %>% 
  
  # Format columns
  dplyr::rename(PROJECT.ID = "X1", 
                SAMPLE.ID = "X2", 
                NITROGEN = "X4", 
                CARBON = "X5")

# Read in soil leco data 2
soil_C_and_N_tbl <- readr::read_csv(file = "data/root_and_soil_chem/WA_LECO_run_from_19_07_29.csv", 
                                 col_names = FALSE) %>% 
  
  # Format columns
  dplyr::rename(PROJECT.ID = "X1", 
                SAMPLE.ID = "X2", 
                NITROGEN = "X4", 
                CARBON = "X5") %>% 
  
  # Combine and filter data
  bind_rows(soil_CN_1.tbl, .) %>% 
  dplyr::filter(PROJECT.ID == "Zak_WA_Manistee_soil_2019") %>% 
  tidyr::separate(col = SAMPLE.ID, 
                  into = c("SUBSTRATE", "SITE", "PLOT", "TECH.REP"), 
                  sep = "_") %>% 
  dplyr::mutate(PLOT = paste("Plot", as.numeric(PLOT), sep = "_")) %>% 
  
  # Calculate average across technical replicates, calculate soil CN, and trim table
  dplyr::group_by(PLOT) %>% 
  dplyr::summarise(MEAN.C = mean(CARBON), 
                   MEAN.N = mean(NITROGEN)) %>% 
  dplyr::ungroup() %>% 
  dplyr::mutate(SOIL.C = ((MEAN.C * 1 / 100) * 1000), 
                SOIL.N = ((MEAN.N * 1 / 100) * 1000)) %>% 
  mutate(SOIL.CN = SOIL.C / SOIL.N) %>% 
  dplyr::select(PLOT, SOIL.C, SOIL.N, SOIL.CN)
```

```{r format_soil_and_root_pygcms}
# Read in sample information
soil.py.gcms.info.tbl <- readr::read_tsv("data/root_and_soil_chem/soil_py_gcms_sample_info.txt")
pre.root.py.gcms.info.tbl <- readr::read_tsv("data/root_and_soil_chem/pre_root_py_gcms_sample_info.txt")

# Read in data
pre.root.soil.py.gcms.tbl <- readr::read_tsv("data/root_and_soil_chem/py_gcms_pre_decay_root_som_data.txt") %>% 
  
  # Update column names and sources
  dplyr::rename(COMPOUND = "Compound", 
                TYPE = "Type", 
                SOURCE = "Source") %>% 
  dplyr::mutate(SOURCE = ifelse(SOURCE == "Lignin+TMAH", "Lignin", SOURCE), 
                SOURCE = ifelse(SOURCE == "Phenol+TMAH", "Phenol", SOURCE)) %>% 
  
  # Long format and sum abundances by compound
  tidyr::pivot_longer(-c(COMPOUND, TYPE, SOURCE), names_to = "SAMPLE.ID", values_to = "PROPORTION") %>% 
  dplyr::group_by(SAMPLE.ID, TYPE, SOURCE, COMPOUND) %>% 
  dplyr::summarize(PROPORTION = sum(PROPORTION)) %>% 
  dplyr::ungroup() %>% 
  
  # Merge with sample information
  tidyr::pivot_wider(id_cols = c(TYPE, SOURCE, COMPOUND), names_from = "SAMPLE.ID", values_from = "PROPORTION") %>% 
  dplyr::mutate(COMPOUND.ID = paste("Compound", row_number(), sep = "_")) %>% 
  dplyr::select(TYPE, 
                SOURCE, 
                COMPOUND, 
                COMPOUND.ID, 
                everything()) %>% 
  tidyr::pivot_longer(-c(COMPOUND.ID, COMPOUND, TYPE, SOURCE), names_to = "SAMPLE.ID", values_to = "PROPORTION")

#### 3. Format soil organic matter data ####

# Add soil sample information
soil_pygcms_source_tbl <- pre.root.soil.py.gcms.tbl %>% 
  dplyr::inner_join(soil.py.gcms.info.tbl, ., by = "SAMPLE.ID") %>% 
  dplyr::mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 

  #  Calculate source abundances
  dplyr::group_by(PLOT, SOURCE) %>% 
  dplyr::summarise(RELABUND = sum(PROPORTION)) %>% 
  dplyr::ungroup() %>% 
  
  # Format columns
  tidyr::pivot_wider(names_from = "SOURCE", values_from = "RELABUND") %>% 
  dplyr::rename_all(., toupper) %>% 
  dplyr::rename_all(~ str_replace_all(., '\\-', '\\.')) %>% 
  dplyr::rename_all(~ str_replace_all(., ' ', '\\.')) %>% 
  dplyr::rename_with(.fn = ~ paste("SOIL", .x, sep = "."), .cols = -PLOT)

# Write out data
# readr::write_tsv(soil.py.gcms.source.tbl, path = "data/working/soil_py_GCMS.txt")

#### 4. Format pre-incubated root data ####

# Add soil sample information
pre_root_pygcms_source <- pre.root.soil.py.gcms.tbl %>% 
  dplyr::inner_join(pre.root.py.gcms.info.tbl, ., by = "SAMPLE.ID") %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\_R", "", SAMPLE.ID)) %>% 
  dplyr::mutate(SAMPLE.ID = gsub("\\-", "_", SAMPLE.ID)) %>% 
  
  #  Calculate source abundances
  dplyr::group_by(SAMPLE.ID, SOURCE) %>% 
  dplyr::summarise(RELABUND = sum(PROPORTION)) %>% 
  dplyr::ungroup() %>% 
  
  # Format columns
  tidyr::pivot_wider(names_from = "SOURCE", values_from = "RELABUND") %>% 
  dplyr::rename_all(., toupper) %>% 
  dplyr::rename_all(~ str_replace_all(., '\\-', '\\.')) %>% 
  dplyr::rename_all(~ str_replace_all(., ' ', '\\.'))

# Write out data
# readr::write_tsv(pre.root.py.gcms.source.tbl, path = "data/working/pre_root_py_GCMS.txt")

# Site means
pre_root_pygcms_source_mean <- pre_root_pygcms_source %>% 
  
  # Get sites
  tidyr::separate(col = SAMPLE.ID, 
                  into = c("SITE", "REPL"), 
                  sep = "_") %>% 
  dplyr::mutate(SITE = paste("Site", SITE, sep = "_")) %>% 
  
  # Calculate site means and sd
  tidyr::pivot_longer(cols = AROMATIC : UNKNOWN.ORIGIN, 
                      names_to = "SOURCE", 
                      values_to = "RELABUND") %>% 
  dplyr::group_by(SITE, SOURCE) %>% 
  dplyr::summarise(RELABUND.MEAN = mean(RELABUND), 
                   RELABUND.SE = se(RELABUND)) %>% 
  dplyr::ungroup() %>% 
  
  # Wide format
  tidyr::pivot_wider(id_cols = SITE, 
                     names_from = "SOURCE", 
                     values_from = c("RELABUND.MEAN", "RELABUND.SE")) %>% 
  dplyr::rename_with(~ gsub("RELABUND.", "PRE.ROOT.", .x), .cols = everything()) %>% 
  dplyr::rename_with(~ gsub("\\_", ".", .x), .cols = everything())
```

```{r calculate_root_lignin_N}
## Combine pygcms and C and N data to calculate lignin N
root_lignin_n <- pre_root_pygcms_source %>% 
  
  inner_join(., pre_root_C_and_N, by = "SAMPLE.ID") %>% 
  
  mutate(LIGNIN.N = LIGNIN / (PRE.ROOT.N / 1000)) %>% 
  
  separate(
    
    col = SAMPLE.ID, 
    
    into = c("SITE", "REPL"), 
    
    sep = "_") %>% 
  
  mutate(SITE = paste("Site", SITE, sep = "_")) %>% 
  
  group_by(SITE) %>% 
  
  summarise(
    
    LIGNIN.N.MEAN = mean(LIGNIN.N), 
    
    LIGNIN.N.SE = se(LIGNIN.N)
    
  )
```

```{r format_soil_ph}
# Read in soil pH data
soil_pH <- readr::read_tsv(file = "data/root_and_soil_chem/Manistee_spring_2019_soil_pH.txt") %>% 
  
  # Format plot names
  dplyr::mutate(PLOT = paste("Plot", PLOT, sep = "_"))
```

```{r format_soil_water_and_temperature}
# Read in site 3 data 1
logger.vwc.temp.3.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_3_Oct_19-20_2019.csv", 
                                           skip = 2, 
                                           col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_3")

# Read in site 3 data 2
logger.vwc.temp.3.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_3_Oct_17_2020.csv", 
                                           skip = 2, 
                                           col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_3")

# Read in site 6 data 1
logger.vwc.temp.6.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_6_Oct_19-20_2019.csv", 
                                           skip = 2, 
                                           col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_6")

# Read in site 6 data 2
logger.vwc.temp.6.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_6_Oct_17_2020.csv", 
                                           skip = 2, 
                                           col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_6")

# Read in site 7 data 1
logger.vwc.temp.7.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_7_Oct_19-20_2019.csv", 
                                           skip = 2, 
                                           col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_7")

# Read in site 7 data 2
logger.vwc.temp.7.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_7_Oct_17_2020.csv", 
                                           skip = 2, 
                                           col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_7")

# Read in site 9 data 1
logger.vwc.temp.9.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_9_Oct_19-20_2019.csv", 
                                           skip = 2, 
                                           col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_9")

# Read in site 9 data 2
logger.vwc.temp.9.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_9_Oct_17_2020.csv", 
                                           skip = 2, 
                                           col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_9")

# Read in site 20 data 1
logger.vwc.temp.20.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_20_Oct_19-20_2019.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_20")

# Read in site 20 data 2
logger.vwc.temp.20.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_20_Oct_17_2020.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_20")

# Read in site 22 data 1
logger.vwc.temp.22.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_22_Oct_19-20_2019.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_22")

# Read in site 22 data 2
logger.vwc.temp.22.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_22_Oct_17_2020.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_22")

# Read in site 24 data 1
logger.vwc.temp.24.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_24_Oct_19-20_2019.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_24")

# Read in site 24 data 2
logger.vwc.temp.24.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_24_Oct_17_2020.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_24")

# Read in site 31 data 1
logger.vwc.temp.31.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_31_Oct_19-20_2019.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_31")

# Read in site 31 data 2
logger.vwc.temp.31.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_31_Oct_17_2020.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_31")

# Read in site 41 data 1
logger.vwc.temp.41.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_41_Oct_19-20_2019.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_41")

# Read in site 41 data 2
logger.vwc.temp.41.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_41_Oct_17_2020.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_41")

# Read in site 50 data 1
logger.vwc.temp.50.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_50_Oct_19-20_2019.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_50")

# Read in site 50 data 2
logger.vwc.temp.50.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_50_Oct_17_2020.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_50")

# Read in site 58 data 1
logger.vwc.temp.58.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_58_Oct_19-20_2019.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_58")

# Read in site 58 data 2
logger.vwc.temp.58.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_58_Oct_17_2020.csv", 
                                            skip = 2, 
                                            col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_58")

# Read in site 100 data 1
logger.vwc.temp.100.1.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_100_Oct_19-20_2019.csv", 
                                             skip = 2, 
                                             col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "A", sep = "_"), 
                SITE = "Site_100")

# Read in site 100 data 2
logger.vwc.temp.100.2.tbl <- readr::read_csv(file = "data/soil_water_and_temp/Site_100_Oct_17_2020.csv", 
                                             skip = 2, 
                                             col_names = FALSE) %>% 
  
  # Column names
  dplyr::rename(OBS = "X1", 
                LOGGER.TIME = "X2", 
                LOGGER.TEMP = "X3", 
                LOGGER.VWC = "X4") %>% 
  dplyr::mutate(OBS = paste(OBS, "B", sep = "_"), 
                SITE = "Site_100")

# Combine logger data
logger.vwc.temp.tbl <- dplyr::bind_rows(logger.vwc.temp.3.1.tbl, 
                                        logger.vwc.temp.3.2.tbl, 
                                        logger.vwc.temp.6.1.tbl, 
                                        logger.vwc.temp.6.2.tbl, 
                                        logger.vwc.temp.7.1.tbl, 
                                        logger.vwc.temp.7.2.tbl, 
                                        logger.vwc.temp.9.1.tbl, 
                                        logger.vwc.temp.9.2.tbl, 
                                        logger.vwc.temp.20.1.tbl, 
                                        logger.vwc.temp.20.2.tbl, 
                                        logger.vwc.temp.22.1.tbl, 
                                        logger.vwc.temp.22.2.tbl, 
                                        logger.vwc.temp.24.1.tbl, 
                                        logger.vwc.temp.24.2.tbl, 
                                        logger.vwc.temp.31.1.tbl, 
                                        logger.vwc.temp.31.2.tbl, 
                                        logger.vwc.temp.41.1.tbl, 
                                        logger.vwc.temp.41.2.tbl, 
                                        logger.vwc.temp.50.1.tbl, 
                                        logger.vwc.temp.50.2.tbl, 
                                        logger.vwc.temp.58.1.tbl, 
                                        logger.vwc.temp.58.2.tbl, 
                                        logger.vwc.temp.100.1.tbl, 
                                        logger.vwc.temp.100.2.tbl) %>% 
  
  # Convert character to date and filter for growing season
  dplyr::mutate(LOGGER.TIME = lubridate::mdy_hms(LOGGER.TIME)) %>% 
  dplyr::filter(!LOGGER.TIME < as.Date("2019-06-01") & 
                  !LOGGER.TIME > as.Date("2020-06-25") & 
                  !(LOGGER.TIME > as.Date("2019-10-31") & LOGGER.TIME < as.Date("2020-05-01")))

#### 3. Read in and format data for hand held probe measurements ####

# Data set 1
vwc.temp.probe.1.tbl <- readr::read_tsv(file = "data/soil_water_and_temp/plot_level_vwc_temp_august_03_to_05_2019.txt") %>% 
  
  # Calculate plot means
  dplyr::group_by(STAND, PLOT, DATE, TIME) %>% 
  dplyr::summarise(VWC = mean(VWC), 
                   TEMP = mean(TEMP)) %>% 
  dplyr::ungroup() %>% 
  
  # Format date and time
  dplyr::mutate(PLOT.TIME = paste(DATE, TIME, sep = " "))

# Data set 2
vwc.probe.2.tbl <- readr::read_tsv(file = "data/soil_water_and_temp/plot_level_vwc_temp_august_09_to_10_2019.txt") %>%
  
  # Calculate plot means
  dplyr::group_by(STAND, PLOT, DATE, TIME) %>% 
  dplyr::summarise(VWC = mean(VWC)) %>% 
  dplyr::ungroup() %>% 
  
  # Format date and time
  dplyr::mutate(PLOT.TIME = paste(DATE, TIME, sep = " "))

vwc.temp.probe.2.tbl <- readr::read_tsv(file = "data/soil_water_and_temp/plot_level_vwc_temp_august_09_to_10_2019.txt") %>%
  
  filter(!is.na(TEMP)) %>% 
  
  # Calculate plot means
  dplyr::group_by(STAND, PLOT, DATE, TIME) %>% 
  dplyr::summarise(TEMP = mean(TEMP)) %>% 
  dplyr::ungroup() %>% 
  
  # Format date and time
  dplyr::mutate(PLOT.TIME = paste(DATE, TIME, sep = " ")) %>% 
  
  select(PLOT, TEMP) %>% 
  
  inner_join(vwc.probe.2.tbl, ., by = "PLOT")

# Data set 3
vwc.temp.probe.3.tbl <- readr::read_tsv(file = "data/soil_water_and_temp/plot_level_vwc_temp_september_14_2019.txt") %>% 
  
  # Calculate plot means
  dplyr::group_by(STAND, PLOT, DATE, TIME) %>% 
  dplyr::summarise(VWC = mean(VWC), 
                   TEMP = mean(TEMP)) %>% 
  dplyr::ungroup() %>% 
  
  # Format date and time
  dplyr::mutate(DATE = lubridate::parse_date_time(DATE, orders = "mdy"), 
                DATE = lubridate::ymd(DATE)) %>% 
  dplyr::mutate(PLOT.TIME = paste(DATE, TIME, sep = " "))

# Data set 4
vwc.temp.probe.4.tbl <- readr::read_tsv(file = "data/soil_water_and_temp/plot_level_vwc_temp_october_19_to_20_2019.txt") %>% 
  
  # Calculate plot means
  dplyr::group_by(STAND, PLOT, DATE, TIME) %>% 
  dplyr::summarise(VWC = mean(VWC), 
                   TEMP = mean(TEMP)) %>% 
  dplyr::ungroup() %>% 
  
  # Format date and time, trim
  dplyr::mutate(DATE = lubridate::parse_date_time(DATE, orders = "mdy"), 
                DATE = lubridate::ymd(DATE)) %>% 
  dplyr::mutate(PLOT.TIME = paste(DATE, TIME, sep = " ")) %>% 
  dplyr::filter(PLOT != 73 & PLOT != 74)
    
# Combine plot data
vwc.temp.probe.tbl <- dplyr::bind_rows(vwc.temp.probe.1.tbl, 
                                       vwc.temp.probe.2.tbl, 
                                       vwc.temp.probe.3.tbl, 
                                       vwc.temp.probe.4.tbl) %>% 
  
  ## Drop temp outlier
  filter(TEMP < 32) %>% 
  
  # Format columns
  dplyr::rename(SITE = "STAND", 
                PLOT.VWC = "VWC", 
                PLOT.TEMP = "TEMP") %>% 
  dplyr::mutate(PLOT.VWC = PLOT.VWC / 100, 
                SITE = paste("Site", SITE, sep = "_"), 
                PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  dplyr::select(PLOT, 
                SITE, 
                PLOT.VWC, 
                PLOT.TEMP, 
                PLOT.TIME) %>% 
  
  # Add logger values, calculate time intervals, and select nearest logger measurements
  dplyr::inner_join(., logger.vwc.temp.tbl, by = "SITE") %>% 
  dplyr::mutate(TIME.INTERVAL = lubridate::interval(start = PLOT.TIME, end = LOGGER.TIME)) %>% 
  dplyr::mutate(TIME.LENGTH = lubridate::time_length(TIME.INTERVAL, unit = "second")) %>% 
  dplyr::mutate(TIME.LENGTH = abs(TIME.LENGTH)) %>% 
  dplyr::mutate(PLOT.DATE = lubridate::date(PLOT.TIME), 
                PLOT.DATE = as.character(PLOT.DATE)) %>% 
  dplyr::group_by(SITE, PLOT, PLOT.DATE) %>% 
  dplyr::mutate(MIN.TIME.LENGTH = min(TIME.LENGTH)) %>% 
  dplyr::ungroup() %>% 
  dplyr::filter(TIME.LENGTH == MIN.TIME.LENGTH) %>% 
  
  # Collapse averages for equal intervals
  dplyr::group_by(SITE, 
                  PLOT, 
                  PLOT.DATE) %>% 
  dplyr::summarise(PLOT.VWC = mean(PLOT.VWC), 
                   PLOT.TEMP = mean(PLOT.TEMP), 
                   LOGGER.VWC = mean(LOGGER.VWC), 
                   LOGGER.TEMP = mean(LOGGER.TEMP)) %>% 
  dplyr::ungroup()

#### 4. Correct plot values to match logger values  ####

# Correction factor list
vwc.temp.probe.corr.list <- vwc.temp.probe.tbl %>% 
  dplyr::filter(PLOT == "Plot_75" | PLOT == "Plot_76" | PLOT == "Plot_77" | PLOT == "Plot_78" | 
                  PLOT == "Plot_79" | PLOT == "Plot_80" | PLOT == "Plot_81" | PLOT == "Plot_82" | 
                  PLOT == "Plot_83" | PLOT == "Plot_84" | PLOT == "Plot_85" | PLOT == "Plot_86") %>% 
  dplyr::group_by(SITE) %>% 
  dplyr::group_split()
names(vwc.temp.probe.corr.list) <- unique(vwc.temp.probe.tbl$SITE)

# Write function to get coefficients from linear regressions to correct plot level data to match logger readings
vwc_temp_corr_coeff <- function(x) {
  
  # Get site
  tmp1 <- unique(x$SITE)
  
  # VWC regression
  tmp2 <- lm(LOGGER.VWC ~ PLOT.VWC, data = x)
  # Summary
  tmp3 <- summary(tmp2)
  # Coefficients
  tmp4 <- tmp3$coefficients
  
  # Temp regression
  tmp5 <- lm(LOGGER.TEMP ~ PLOT.TEMP, data = x)
  # Summary
  tmp6 <- summary(tmp5)
  # Coefficients
  tmp7 <- tmp6$coefficients
  
  # Compile data
  tmp8 <- tibble(SITE = tmp1, 
                 VWC.CORR.INTERCEPT = tmp4[1, 1], 
                 VWC.CORR.SLOPE = tmp4[2, 1], 
                 VWC.CORR.INTERCEPT.P = tmp4[1, 4], 
                 VWC.CORR.SLOPE.P = tmp4[2, 4], 
                 VWC.CORR.R2 = tmp3$r.squared, 
                 TEMP.CORR.INTERCEPT = tmp7[1, 1], 
                 TEMP.CORR.SLOPE = tmp7[2, 1], 
                 TEMP.CORR.INTERCEPT.P = tmp7[1, 4], 
                 TEMP.CORR.SLOPE.P = tmp7[2, 4], 
                 TEMP.CORR.R2 = tmp6$r.squared)
  
  # Return
  return(tmp8)
  
}

# Calculate regression coefficients for correction factors, and correct plot point measurements
vwc.temp.corr.tbl <- lapply(vwc.temp.probe.corr.list, vwc_temp_corr_coeff) %>% 
  dplyr::bind_rows() %>% 
  # Add plot level data
  dplyr::inner_join(vwc.temp.probe.tbl, ., by = "SITE") %>% 
  dplyr::filter(PLOT != "Plot_75" & PLOT != "Plot_76" & PLOT != "Plot_77" & PLOT != "Plot_78" & 
                  PLOT != "Plot_79" & PLOT != "Plot_80" & PLOT != "Plot_81" & PLOT != "Plot_82" & 
                  PLOT != "Plot_83" & PLOT != "Plot_84" & PLOT != "Plot_85" & PLOT != "Plot_86") %>% 
  dplyr::mutate(PLOT.VWC.CORR = (VWC.CORR.SLOPE * PLOT.VWC) + VWC.CORR.INTERCEPT, 
                PLOT.TEMP.CORR = (TEMP.CORR.SLOPE * PLOT.TEMP) + TEMP.CORR.INTERCEPT)

#### 5. Run regressions of plot vs. logger readings ####

# Write function to get coefficients from linear regressions of corrected plot vs. logger readings
vwc_temp_coeff <- function(x) {
  
  # Get plot
  tmp1 <- unique(x$PLOT)
  # Get site
  tmp1b <- unique(x$SITE)
  
  # VWC regression
  tmp2 <- lm(PLOT.VWC.CORR ~ LOGGER.VWC, data = x)
  # Summary
  tmp3 <- summary(tmp2)
  # Coefficients
  tmp4 <- tmp3$coefficients
  
  # Temp regression
  tmp5 <- lm(PLOT.TEMP.CORR ~ LOGGER.TEMP, data = x)
  # Summary
  tmp6 <- summary(tmp5)
  # Coefficients
  tmp7 <- tmp6$coefficients
  
  # Compile data
  tmp8 <- tibble(PLOT = tmp1, 
                 SITE = tmp1b, 
                 VWC.INTERCEPT = tmp4[1, 1], 
                 VWC.SLOPE = tmp4[2, 1], 
                 VWC.INTERCEPT.P = tmp4[1, 4], 
                 VWC.SLOPE.P = tmp4[2, 4], 
                 VWC.R2 = tmp3$r.squared, 
                 TEMP.INTERCEPT = tmp7[1, 1], 
                 TEMP.SLOPE = tmp7[2, 1], 
                 TEMP.INTERCEPT.P = tmp7[1, 4], 
                 TEMP.SLOPE.P = tmp7[2, 4], 
                 TEMP.R2 = tmp6$r.squared)
  
  # Return
  return(tmp8)
  
}

# Input list
vwc.temp.list <- vwc.temp.corr.tbl %>% 
  dplyr::group_by(PLOT) %>% 
  dplyr::group_split()
names(vwc.temp.list) <- unique(vwc.temp.corr.tbl$PLOT)

# Get coefficients for regressions predicting plot values from logger values
vwc.temp.coeff.tbl <- lapply(vwc.temp.list, vwc_temp_coeff) %>% 
  dplyr::bind_rows()

# Get hourly logger data
soil_vwc_temp_daily <- logger.vwc.temp.tbl %>% 
  dplyr::rename(VWC.LOGGER.HOURLY = "LOGGER.VWC", 
                TEMP.LOGGER.HOURLY = "LOGGER.TEMP") %>% 
  
  # Add coefficients to calculate plot level hourly data
  dplyr::inner_join(., vwc.temp.coeff.tbl, by = "SITE") %>% 
  
  # Calculate hourly values for plots
  dplyr::mutate(VWC.PLOT.HOURLY = (VWC.SLOPE * VWC.LOGGER.HOURLY) + VWC.INTERCEPT, 
                TEMP.PLOT.HOURLY = (TEMP.SLOPE * TEMP.LOGGER.HOURLY) + TEMP.INTERCEPT) %>% 
  
  # Calculate daily means by plot over growing season
  dplyr::mutate(PLOT.DATE = lubridate::date(LOGGER.TIME), 
                PLOT.DATE = as.character(PLOT.DATE)) %>% 
  dplyr::group_by(PLOT, PLOT.DATE) %>% 
  dplyr::summarise(VWC.DAILY.MEAN = mean(VWC.PLOT.HOURLY), 
                   TEMP.DAILY.MEAN = mean(TEMP.PLOT.HOURLY)) %>% 
  dplyr::ungroup()

## Water
soil_vwc <- soil_vwc_temp_daily %>% 
  
  group_by(PLOT) %>% 
  
  mutate(VWC.THRESHOLD = 0.8 * max(VWC.DAILY.MEAN)) %>% 
  
  ungroup() %>% 
  
  mutate(VWC.ABOVE = ifelse(VWC.DAILY.MEAN >= VWC.THRESHOLD, 1, 0)) %>% 
  
  group_by(PLOT) %>% 
  
  summarise(
    
    VWC.MEAN = mean(VWC.DAILY.MEAN), 
    
    VWC.SD = sd(VWC.DAILY.MEAN), 
    
    VWC.ABOVE.TOTAL = sum(VWC.ABOVE)                           
    
  )

## Water
soil_temp <- soil_vwc_temp_daily %>% 
  
  group_by(PLOT) %>% 
  
  summarise(
    
    TEMP.MEAN = mean(TEMP.DAILY.MEAN), 
    
    TEMP.SD = sd(TEMP.DAILY.MEAN)
    
  )

## Combine
soil_vwc_temp <- soil_vwc %>% 
  
  inner_join(., soil_temp, by = "PLOT")
```

```{r determine_plot_locations}
#### 2. Calculate coordinates for each plot

# Read in sample information
plot.locations.tbl <- readr::read_tsv("data/site/plot.locations.updated.2019.10.19.and.20.V2.txt") %>% 
  
  # Format columns
  dplyr::select(Site, Plot, `Bag angle`, `Bag distance (m)`, `Site latitude degrees (N)`, `Site latitude minutes (N)`, `Site longitude degrees (W)`, `Site longitude minutes (W)`) %>% 
  dplyr::rename(STAND = "Site", 
                PLOT = "Plot", 
                PLOT.BEARING = "Bag angle", 
                PLOT.DIST = "Bag distance (m)", 
                SITE.LAT.DEG = "Site latitude degrees (N)", 
                SITE.LAT.MIN = "Site latitude minutes (N)", 
                SITE.LON.DEG = "Site longitude degrees (W)", 
                SITE.LON.MIN = "Site longitude minutes (W)") %>% 
  dplyr::filter(PLOT <= 72) %>% 
  mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  
  # Calculate decimal degrees
  mutate(SITE.LAT = SITE.LAT.DEG + (SITE.LAT.MIN / 60), 
         SITE.LON = -1 * (SITE.LON.DEG + (SITE.LON.MIN / 60)))

# Format matrix of stand latitude and longitude
plot.locations.df <- as.data.frame(plot.locations.tbl)
row.names(plot.locations.df) <- plot.locations.df$PLOT
plot.locations.mat <- as.matrix(plot.locations.df[c("SITE.LON", "SITE.LAT")])

# Format plot bearings and distances
plot.bearing.mat <- as.matrix(plot.locations.df["PLOT.BEARING"])
plot.dist.mat <- as.matrix(plot.locations.df["PLOT.DIST"])

# Calculate coordinates for each plot
plot.coordinates.df <- as.data.frame(destPoint(plot.locations.mat, plot.bearing.mat, plot.dist.mat))
row.names(plot.coordinates.df) <- row.names(plot.locations.df)

plot_coordinates <- as_tibble(plot.coordinates.df, rownames = "PLOT") %>% 
  
  # Update column names
  dplyr::rename(LON = "lon", 
                LAT = "lat")
```

```{r sequence_processing_functions}
#### Write a function to tabulate primer hits ####
primerHits <- function(primer, fn) {
  
  ## Counts number of reads in which the primer is found
  nhits <- vcountPattern(primer, sread(readFastq(fn)), fixed = FALSE)
  
  return(sum(nhits > 0))
  
}

#### Write a function to get sample type ####
sample_type <- function(x) {
  
  if(x == "MOCK1" | x == "MOCK2" | x == "MOCK1nano" | x == "MOCK2nano") {
    tmp1 <- "mock"
  } else if(x == "NC1" | x == "NC2" | x == "NC1nano" | x == "NC2nano") {
    tmp1 <- "negative.control"
  } else {
    tmp1 <- "environmental"
  }
  
  return(tmp1)
  
}

#### Write a function to get plot number ####
sample_plot <- function(sample.id, sample.type) {
  
  if(sample.type == "environmental") {
    tmp1 <- gsub("[a-zA-Z]", "", sample.id)
  } else {
    tmp1 <- sample.id
  }
  
  return(tmp1)
  
}

#### Write a function to get stand number ####
sample_stand <- function(plot.number, sample.type) {
  
  if(sample.type == "environmental" & as.numeric(plot.number) >= 1 & as.numeric(plot.number) <= 6) {
    tmp1 <- 3
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 7 & as.numeric(plot.number) <= 12) {
    tmp1 <- 6
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 13 & as.numeric(plot.number) <= 18) {
    tmp1 <- 7
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 19 & as.numeric(plot.number) <= 24) {
    tmp1 <- 9
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 25 & as.numeric(plot.number) <= 30) {
    tmp1 <- 20
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 31 & as.numeric(plot.number) <= 36) {
    tmp1 <- 22
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 37 & as.numeric(plot.number) <= 42) {
    tmp1 <- 24
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 43 & as.numeric(plot.number) <= 48) {
    tmp1 <- 31
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 49 & as.numeric(plot.number) <= 54) {
    tmp1 <- 41
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 55 & as.numeric(plot.number) <= 60) {
    tmp1 <- 50
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 61 & as.numeric(plot.number) <= 66) {
    tmp1 <- 58
  } else if(sample.type == "environmental" & as.numeric(plot.number) >= 67 & as.numeric(plot.number) <= 72) {
    tmp1 <- 100
  } else if(sample.type == "mock") {
    tmp1 <- plot.number
  } else if(sample.type == "negative.control") {
    tmp1 <- plot.number
  } else {
    tmp1 <- NA
  }
  
  return(tmp1)
}

#### Write function to update plot ####
sample_plot_update <- function(plot.number, sample.type) {
  
  if(sample.type == "environmental") {
    tmp1 <- paste("Plot", plot.number, sep = "_")
  } else {
    tmp1 <- plot.number
  }
  
  return(tmp1)
  
}

#### Write function to update stand ####
sample_stand_update <- function(stand.number, sample.type) {
  
  if(sample.type == "environmental") {
    tmp1 <- paste("Stand", stand.number, sep = "_")
  } else {
    tmp1 <- stand.number
  }
  
  return(tmp1)
  
}

#### Write function to get substrate ####
sample_substrate <- function(sample.id, sample.type) {
  
  if(sample.type == "environmental") {
    tmp1 <- gsub("[[:digit:]]", "", sample.id)
    tmp2 <- ifelse(tmp1 == "S" | tmp1 == "Snano", "soil", "decomp.roots")
  } else {
    tmp2 <- sample.id
  }
  
  return(tmp2)
  
}

#### Write a function to get sequence run ####
sample_seqRun <- function(sample.id, sample.type) {
  
  if(sample.type == "environmental") {
    tmp1 <- gsub("[[:digit:]]", "", sample.id)
    tmp2 <- ifelse(tmp1 == "S" | tmp1 == "R" | tmp1 == "RA" | tmp1 == "RB", "full", "nano")
  } else if(sample.id == "MOCK1" | sample.id == "MOCK2" | sample.id == "NC1" | sample.id == "NC2") {
    tmp2 <- "full"
  } else {
    tmp2 <- "nano"
  }
  
  return(tmp2)
  
}

#### Write function to clean up taxa ####
tax_clean <- function(x) {
  
  if(!is.na(x)) {
    tmp1 <- gsub(".*_", "", x)
  } else {
    tmp1 <- x
  }
  
  return(tmp1)
  
}

#### Write a function to handle unclassified species ####
s_unclass <- function(kingdom, phylum, class, order, family, genus, species) {
  
  if(!is.na(species)) {
    tmp1 <- species
  } else if(is.na(species) & !is.na(genus) & genus != "sedis") {
    tmp1 <- paste("unclassified", genus, sep = " ")
  } else if(is.na(species) & is.na(genus) & !is.na(family) & family != "sedis") {
    tmp1 <- paste("unclassified", family, sep = " ")
  } else if(is.na(species) & is.na(genus) & is.na(family) & !is.na(order) & order != "sedis") {
    tmp1 <- paste("unclassified", order, sep = " ")
  } else if(is.na(species) & is.na(genus) & is.na(family) & is.na(order) & !is.na(class) & class != "sedis") {
    tmp1 <- paste("unclassified", class, sep = " ")
  } else if(is.na(species) & is.na(genus) & is.na(family) & is.na(order) & is.na(class) & !is.na(phylum) & phylum != "sedis") {
    tmp1 <- paste("unclassified", phylum, sep = " ")
  } else if(is.na(species) & is.na(genus) & is.na(family) & is.na(order) & is.na(class) & is.na(phylum) & !is.na(kingdom) & kingdom != "sedis") {
    tmp1 <- paste("unclassified", kingdom, sep = " ")
  } else {
    tmp1 <- NA
  }
  
  return(tmp1)
  
}

#### Write a function to handle unclassified genera ####
g_unclass <- function(kingdom, phylum, class, order, family, genus) {
  
  if(!is.na(genus)) {
    tmp1 <- genus
  } else if(is.na(genus) & !is.na(family) & family != "sedis") {
    tmp1 <- paste("unclassified", family, sep = " ")
  } else if(is.na(genus) & is.na(family) & !is.na(order) & order != "sedis") {
    tmp1 <- paste("unclassified", order, sep = " ")
  } else if(is.na(genus) & is.na(family) & is.na(order) & !is.na(class) & class != "sedis") {
    tmp1 <- paste("unclassified", class, sep = " ")
  } else if(is.na(genus) & is.na(family) & is.na(order) & is.na(class) & !is.na(phylum) & phylum != "sedis") {
    tmp1 <- paste("unclassified", phylum, sep = " ")
  } else if(is.na(genus) & is.na(family) & is.na(order) & is.na(class) & is.na(phylum) & !is.na(kingdom) & kingdom != "sedis") {
    tmp1 <- paste("unclassified", kingdom, sep = " ")
  } else {
    tmp1 <- NA
  }
  
  return(tmp1)
  
}

#### Write a function to handle unclassified families ####
f_unclass <- function(kingdom, phylum, class, order, family) {
  
  if(!is.na(family)) {
    tmp1 <- family
  } else if(is.na(family) & !is.na(order) & order != "sedis") {
    tmp1 <- paste("unclassified", order, sep = " ")
  } else if(is.na(family) & is.na(order) & !is.na(class) & class != "sedis") {
    tmp1 <- paste("unclassified", class, sep = " ")
  } else if(is.na(family) & is.na(order) & is.na(class) & !is.na(phylum) & phylum != "sedis") {
    tmp1 <- paste("unclassified", phylum, sep = " ")
  } else if(is.na(family) & is.na(order) & is.na(class) & is.na(phylum) & !is.na(kingdom) & kingdom != "sedis") {
    tmp1 <- paste("unclassified", kingdom, sep = " ")
  } else {
    tmp1 <- NA
  }
  
  return(tmp1)
  
}

#### Write a function to handle unclassified orders ####
o_unclass <- function(kingdom, phylum, class, order) {
  
  if(!is.na(order)) {
    tmp1 <- order
  } else if(is.na(order) & !is.na(class) & class != "sedis") {
    tmp1 <- paste("unclassified", class, sep = " ")
  } else if(is.na(order) & is.na(class) & !is.na(phylum) & phylum != "sedis") {
    tmp1 <- paste("unclassified", phylum, sep = " ")
  } else if(is.na(order) & is.na(class) & is.na(phylum) & !is.na(kingdom) & kingdom != "sedis") {
    tmp1 <- paste("unclassified", kingdom, sep = " ")
  } else {
    tmp1 <- NA
  }
  
  return(tmp1)
  
}

#### Write a function to handle unclassified classes ####
c_unclass <- function(kingdom, phylum, class) {
  
  if(!is.na(class)) {
    tmp1 <- class
  } else if(is.na(class) & !is.na(phylum) & phylum != "sedis") {
    tmp1 <- paste("unclassified", phylum, sep = " ")
  } else if(is.na(class) & is.na(phylum) & !is.na(kingdom) & kingdom != "sedis") {
    tmp1 <- paste("unclassified", kingdom, sep = " ")
  } else {
    tmp1 <- NA
  }
  
  return(tmp1)
  
}

#### Write a function to handle unclassified phyla ####
p_unclass <- function(kingdom, phylum) {
  
  if(!is.na(phylum)) {
    tmp1 <- phylum
  } else if(is.na(phylum) & !is.na(kingdom) & kingdom != "sedis") {
    tmp1 <- paste("unclassified", kingdom, sep = " ")
  } else {
    tmp1 <- NA
  }
  
  return(tmp1)
  
}

#### Get raw sequence counts ####
raw_seq_counts <- function(x) {
  
  tmp1 <- sum(x$data$Count) / 251
  
  return(tmp1)
  
}
```

```{r dada2}
# Uncomment to run
# 
# #### 2. Check initial read quality ####
# 
# # Directory containing fastq files
# path <- "data/Manistee_MiSeq_data"
# 
# # Get a list of files in the directory
# list.files(path)
# 
# # Create vectors of paths to each file
# fnFs <- sort(list.files(path, pattern = "_R1_001.fastq.gz", full.names = TRUE))
# fnRs <- sort(list.files(path, pattern = "_R2_001.fastq.gz", full.names = TRUE))
# 
# # Check initial quality of reads
# fnFs.qual <- vector(mode = "list", length = length(fnFs))
# get.sample.name <- function(fname) strsplit(basename(fname), "_")[[1]][1]
# sample.names <- unname(sapply(fnFs, get.sample.name))
# names(fnFs.qual) <- sample.names
# 
# # Uncomment to run (takes a while)
# #for(i in seq_along(fnFs)) {
# #  
# #  print(paste("Processing:", sample.names[i], sep = " "))
# #  fnFs.qual[[i]] <- plotQualityProfile(fnFs[i])
# #  
# #}
# 
# # Save plots
# #saveRDS(fnFs.qual, file = "data/Manistee_MiSeq_data_v2/quality_output/fnFs.qual.rds")
# 
# # Check initial quality of reads
# fnRs.qual <- vector(mode = "list", length = length(fnRs))
# names(fnRs.qual) <- sample.names
# 
# # Uncomment to run (takes a while)
# #for(i in seq_along(fnRs)) {
# #  
# #  print(paste("Processing:", sample.names[i], sep = " "))
# #  fnRs.qual[[i]] <- plotQualityProfile(fnRs[i])
# #  
# #}
# 
# # Save plots
# #saveRDS(fnRs.qual, file = "data/Manistee_MiSeq_data/quality_output/fnRs.qual.rds")
# 
# # Drop reverse reads from the rest of the analysis due to low quality
# 
# #### 3. Filter Ns and check for primer sequences ####
# 
# # Forward and reverse primer sequences
# FWD <- "AGCCTCCGCTTATTGATATGCTTAART"
# REV <- "AACTTTYRRCAAYGGATCWCT"
# 
# # Create function for all possible orientations of each primer
# allOrients <- function(primer) {
#   
#   # Create all orientations of the input sequence
#   require(Biostrings)
#   # The Biostrings works w/ DNAString objects rather than character vectors
#   dna <- DNAString(primer)
#   orients <- c(Forward = dna, Complement = complement(dna), Reverse = reverse(dna), 
#                RevComp = reverseComplement(dna))
#   # Convert back to character vector
#   return(sapply(orients, toString))
# }
# 
# # Create all possible orientations of each primer
# FWD.orients <- allOrients(FWD)
# REV.orients <- allOrients(REV)
# 
# # Pre-filter sequences to remove sequences with any ambiguous base calls; put N-filtered files in filtN/ subdirectory
# fnFs.filtN <- file.path(path, "filtN", basename(fnFs))
# # Filter
# filterAndTrim(fnFs, fnFs.filtN, maxN = 0, multithread = TRUE)
# 
# # Tabulate primer hits
# fnFs.filtN.primerHits <- vector(mode = "list", length = length(fnFs.filtN))
# names(fnFs.filtN.primerHits) <- sample.names
# 
# # Uncomment to run (takes a while)
# #for(i in seq_along(fnFs.filtN.primerHits)) {
# #  
# #  print(paste("Processing:", sample.names[i], sep = " "))
# #  fnFs.filtN.primerHits[[i]] <- rbind(FWD.ForwardReads = sapply(FWD.orients, primerHits, fn = fnFs.filtN[[i]]), 
# #                                      REV.ForwardReads = sapply(REV.orients, primerHits, fn = fnFs.filtN[[i]]))
# #  
# #}
# 
# # Save data
# saveRDS(fnFs.filtN.primerHits, file = "data/Manistee_MiSeq_data/primer_hits_output/fnFs.primer.hits.rds")
# 
# #### 4. Remove primers using cutadapt ####
# 
# # Specify path to cutadapt
# cutadapt <- "/Users/williamargiroff/.local/bin/cutadapt"
# # Check that cutadapt is reached from R
# system2(cutadapt, args = "--version")
# 
# # Create a directory to hold the cutadapt-trimmed sequences
# path.cut <- file.path(path, "cutadapt_processed")
# if(!dir.exists(path.cut)) dir.create(path.cut)
# 
# # Create names for the files in the directory
# fnFs.cut <- file.path(path.cut, basename(fnFs))
# 
# # Reverse complement the reverse primer (as it would show up in the forward reads if the read sequenced into the reverse primer)
# REV.RC <- dada2:::rc(REV)
# 
# # Create flags to trim FWD and the reverse-complement of REV off of R1 (forward reads)
# R1.flags <- paste("-g", FWD, "-a", REV.RC) 
# 
# # Uncomment to run (takes a while)
# # Run Cutadapt
# #for(i in seq_along(fnFs.cut)) {
# #  
#   #-n 2 required to remove FWD and REV primers from reads
# #  system2(cutadapt, args = c(R1.flags, "-n", 2,
#                              # output files
# #                             "-o", fnFs.cut[i],
#                              # input files
# #                             fnFs.filtN[i]))
# #  
# #}
# 
# # Tabulate primer hits
# fnFs.cut.primerHits <- vector(mode = "list", length = length(fnFs.cut))
# names(fnFs.cut.primerHits) <- sample.names
# 
# for(i in seq_along(fnFs.cut.primerHits)) {
#   
#   print(paste("Processing:", sample.names[i], sep = " "))
#   fnFs.cut.primerHits[[i]] <- rbind(FWD.ForwardReads = sapply(FWD.orients, primerHits, fn = fnFs.cut[[i]]), 
#                                     REV.ForwardReads = sapply(REV.orients, primerHits, fn = fnFs.cut[[i]]))
#   
# }
# 
# # Save data
# saveRDS(fnFs.cut.primerHits, file = "data/Manistee_MiSeq_data/primer_hits_output/fnFs.cut.primer.hits.rds")
# 
# # Forward read fastq filenames have the following format
# cutFs <- sort(list.files(path.cut, pattern = "_R1_001.fastq.gz", full.names = TRUE))
# 
# # Extract sample names, assuming filenames have format from above
# sample.names <- unname(sapply(cutFs, get.sample.name))
# 
# # Quality
# cutFs.qual <- vector(mode = "list", length = length(cutFs))
# names(cutFs.qual) <- sample.names
# 
# for(i in seq_along(cutFs)) {
#   
#   print(paste("Processing:", sample.names[i], sep = " "))
#   cutFs.qual[[i]] <- plotQualityProfile(cutFs[i])
#   
# }
# 
# # Save plots
# saveRDS(cutFs.qual, file = "data/Manistee_MiSeq_data/quality_output/cutFs.qual.rds")
# 
# #### 5. Filter reads based on errors and length; quantify error rate ####
# 
# # Create a directory for filtered reads
# filtFs <- file.path(path.cut, "filtered", basename(cutFs))
# 
# # filter reads
# filter.summary <- filterAndTrim(cutFs, filtFs, maxN = 0, maxEE = 2, 
#                                 truncQ = 2, minLen = 100, rm.phix = TRUE, compress = TRUE, multithread = TRUE)
# 
# # Quantify error rate
# errF <- learnErrors(filtFs, multithread = TRUE)
# 
# # Plot error rates
# errF.plot <- plotErrors(errF, nominalQ = TRUE)
# 
# # Quality
# filtFs.qual <- vector(mode = "list", length = length(filtFs))
# names(filtFs.qual) <- sample.names
# 
# for(i in seq_along(filtFs)) {
#   
#   print(paste("Processing:", sample.names[i], sep = " "))
#   filtFs.qual[[i]] <- plotQualityProfile(filtFs[i])
#   
# }
# 
# # Save plots
# saveRDS(filtFs.qual, file = "data/Manistee_MiSeq_data/quality_output/filtFs.qual.rds")
# 
# #### 6. De-replicate seqs, run ASV inference, remove chimeras, ASV table ####
# 
# # De-replicate the sequences
# derepFs <- derepFastq(filtFs, verbose = TRUE)
# 
# # Name the derep-class objects by the sample names
# names(derepFs) <- sample.names
# 
# # Run sample inference on dereplicated sequences
# dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
# 
# # Create ASV table
# seqtab <- makeSequenceTable(dadaFs)
# dim(seqtab)
# 
# # Inspect distribution of sequence lengths
# seqtab.table <- table(nchar(getSequences(seqtab)))
# saveRDS(seqtab.table, file = "data/Manistee_MiSeq_data/seq_summary/seqtab.table.rds")
# 
# # Remove chimeras
# seqtab.nochim <- removeBimeraDenovo(seqtab, method = "consensus", multithread = TRUE, verbose = TRUE)
# dim(seqtab.nochim)
# sum(seqtab.nochim)/sum(seqtab)
# seqtab.nochim.table <- table(nchar(getSequences(seqtab.nochim)))
# saveRDS(seqtab.nochim.table, file = "data/Manistee_MiSeq_data_v2/seq_summary/seqtab.nochim.table.rds")
# 
# #### 7. Summarize filtering output ####
# 
# # Write function to get uniques to start tracking progress through the pipeline
# getN <- function(x) sum(getUniques(x))
# 
# # If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
# track <- as.data.frame(cbind(filter.summary, sapply(dadaFs, getN), rowSums(seqtab.nochim)))
# colnames(track) <- c("input", "filtered", "denoisedF", "nonchim")
# rownames(track) <- sample.names
# # Percent remaining
# track$PERC.REMAINING <- 100 * (track$nonchim / track$input)
# 
# # Track sequences remaining after different steps; save data
# saveRDS(track, file = "data/Manistee_MiSeq_data/seq_summary/dada2.seq.summary.rds")
# 
# #### 8. Assign taxonomy using UNITE ####
# 
# # Assign taxonomy; runs overnight
# unite.ref <- "data/Manistee_MiSeq_data/unite/sh_general_release_dynamic_s_04.02.2020.fasta"
# taxa <- assignTaxonomy(seqtab.nochim, unite.ref, 
#                        multithread = TRUE, 
#                        tryRC = TRUE, 
#                        minBoot = 80, 
#                        outputBootstraps = TRUE, 
#                        verbose = TRUE)
# 
# saveRDS(taxa, file = "data/Manistee_MiSeq_data/unite/taxa.rds")
# 
# #### 9. Save data as a phyloseq object -- this is specific to my data, but is convenient downstream ####
# 
# # Construct a sample data frame
# samples.out <- rownames(seqtab.nochim)
# samdf <- data.frame(SAMPLE.ID = samples.out, stringsAsFactors = FALSE)
# row.names(samdf) <- samples.out
# 
# # Get sample type
# samdf$TYPE <- sapply(samdf$SAMPLE.ID, sample_type, simplify = TRUE)
# # Get sample plot
# samdf$PLOT <- mapply(sample_plot, samdf$SAMPLE.ID, samdf$TYPE, SIMPLIFY = TRUE)
# # Get sample stand
# samdf$STAND <- mapply(sample_stand, samdf$PLOT, samdf$TYPE, SIMPLIFY = TRUE)
# # Update plot
# samdf$PLOT <- mapply(sample_plot_update, samdf$PLOT, samdf$TYPE, SIMPLIFY = TRUE)
# # Update stand
# samdf$STAND <- mapply(sample_stand_update, samdf$STAND, samdf$TYPE, SIMPLIFY = TRUE)
# # Add substrate
# samdf$SUBSTRATE <- mapply(sample_substrate, samdf$SAMPLE.ID, samdf$TYPE, SIMPLIFY = TRUE)
# # Add run
# samdf$RUN.TYPE <- mapply(sample_seqRun, samdf$SAMPLE.ID, samdf$TYPE, SIMPLIFY = TRUE)
# 
# # Load phyloseq
# library(phyloseq)
# 
# # Create phyloseq object
# ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows = FALSE), 
#                sample_data(samdf), 
#                tax_table(taxa$tax))
# 
# # Get refseqs
# dna <- Biostrings::DNAStringSet(taxa_names(ps))
# # Add seqs to phyloseq object
# names(dna) <- taxa_names(ps)
# ps <- merge_phyloseq(ps, dna)
# # Update ASV names
# taxa_names(ps) <- paste0("ASV_", seq(ntaxa(ps)))
# 
# # Save data
# saveRDS(ps, file = "data/Manistee_MiSeq_data/asv_data/phyloseq.rds")
```

```{r format_asv_table}
# Read in data
ps <- readRDS("data/Manistee_MiSeq_data/asv_data/phyloseq.rds")

#### 2. Format ASV table ####

## Get ASV table from phyloseq object
ps_ASV <- as.data.frame(otu_table(ps)) %>% 
  
  as_tibble(., rownames = "SAMPLE.ID")

## Get sample data from phyloseq object
ps_sample <- as.data.frame(sample_data(ps)) %>% 
  
  as_tibble(.)

## Get taxonomy data
ps_tax <- as.data.frame(tax_table(ps)) %>% 
  
  as_tibble(., rownames = "ASV") %>% 
  
  rename_all(toupper) %>% 
  
  pivot_longer(-ASV, names_to = "RANKING", values_to = "TAXON") %>% 
  
  mutate(TAXON = sapply(TAXON, tax_clean, simplify = TRUE)) %>% 
  
  pivot_wider(id_cols = ASV, names_from = "RANKING", values_from = TAXON) %>% 
  
  mutate(
    SPECIES = mapply(s_unclass, KINGDOM, PHYLUM, CLASS, ORDER, FAMILY, GENUS, SPECIES, SIMPLIFY = TRUE), 
    
    GENUS = mapply(g_unclass, KINGDOM, PHYLUM, CLASS, ORDER, FAMILY, GENUS, SIMPLIFY = TRUE), 
    
    FAMILY = mapply(f_unclass, KINGDOM, PHYLUM, CLASS, ORDER, FAMILY, SIMPLIFY = TRUE), 
    
    ORDER = mapply(o_unclass, KINGDOM, PHYLUM, CLASS, ORDER, SIMPLIFY = TRUE), 
    
    CLASS = mapply(c_unclass, KINGDOM, PHYLUM, CLASS, SIMPLIFY = TRUE), 
    
    PHYLUM = mapply(p_unclass, KINGDOM, PHYLUM, SIMPLIFY = TRUE)
    
  )

## Format
roots_asv <- ps_ASV %>% 
  
  pivot_longer(-SAMPLE.ID, names_to = "ASV", values_to = "COUNT") %>% 
  
  ## Add sample data
  inner_join(ps_sample, ., by = "SAMPLE.ID") %>% 
  
  ## Remove mock communities and negative controls
  filter(TYPE == "environmental" & SUBSTRATE == "decomp.roots") %>% 
  
  ## Combine sequences by plot and ASV
  rename(SITE = "STAND") %>% 
  
  mutate(SITE = gsub("Stand", "Site", SITE)) %>% 

  group_by(PLOT, SITE, ASV) %>% 
  
  summarise(COUNT = sum(COUNT)) %>% 
  
  ungroup() %>% 
  
  ## Remove any ASVs with no sequences (after mock and negative controls removed)
  group_by(ASV) %>% 
  
  mutate(ASV.TOTAL = sum(COUNT)) %>% 
  
  ungroup() %>% 
  
  filter(ASV.TOTAL > 0) %>% 
  
  select(-ASV.TOTAL)
```

```{r format_genera}
## Update tax table
tax_modified <- ps_tax %>% 
  
  mutate(
    
    FAMILY = ifelse(GENUS == "Cryptococcus" & FAMILY == "Cryptococcaceae", "Tremellaceae", FAMILY), 
    
    GENUS = ifelse(is.na(GENUS) & CLASS == "Lecanoromycetes", "Lecanoromycetes incertae sedis", GENUS), 
    
    GENUS = ifelse(is.na(GENUS) & ORDER == "Helotiales", "Helotiales incertae sedis", GENUS), 
    
    GENUS = ifelse(is.na(GENUS) & ORDER == "Hypocreales", "Hypocreales incertae sedis", GENUS), 
    
    GENUS = ifelse(is.na(GENUS) & PHYLUM == "Rozellomycota","Rozellomycota incertae sedis", GENUS), 
    
    GENUS = ifelse(GENUS == "unclassified GS35" & ORDER == "unclassified GS35","unclassified GS35 (c)", GENUS), 
    
    GENUS = ifelse(GENUS == "unclassified GS35" & ORDER == "GS35","unclassified GS35 (o)", GENUS), 
    
    GENUS = ifelse(GENUS == "unclassified GS18" & ORDER == "unclassified GS18","unclassified GS18 (c)", GENUS), 
    
    GENUS = ifelse(GENUS == "unclassified GS18" & ORDER == "GS18","unclassified GS18 (c)", GENUS)
    
  )

## Sum sequence counts by plot and genus, get relative abundances
genus <- roots_asv %>% 
  
  inner_join(tax_modified, ., by = "ASV") %>% 
  
  group_by(SITE, PLOT, GENUS) %>% 
  
  summarise(COUNT = sum(COUNT)) %>% 
  
  ungroup() %>% 
  
  group_by(PLOT) %>% 
  
  mutate(PLOT.COUNT = sum(COUNT)) %>% 
  
  ungroup() %>% 
  
  mutate(PROP.ABUND = COUNT / PLOT.COUNT) %>% 
  
  inner_join(., root_mass_loss, by = "PLOT") %>% 
  
  ## Filter outliers
  filter(PLOT != "Plot_19" & PROP.MASS.LOSS < 0.4) %>%  
  
  select(-PLOT.COUNT, -PROP.MASS.LOSS)

## Get genus info
genus_all <- genus %>% 
  
  group_by(GENUS) %>% 
  
  mutate(GENUS.COUNT = sum(COUNT)) %>% 
  
  ungroup() %>% 
  
  mutate(
    
    GENUS.PERC = 100 * (GENUS.COUNT / sum(COUNT)), 
    
    PA = ifelse(COUNT > 0, 1, 0)
    
  ) %>% 
  
  group_by(GENUS) %>% 
  
  mutate(OCCURRENCE = sum(PA)) %>% 
  
  ungroup()

## Get abundant genera only
genus_relabund <- genus_all %>% 
  
  separate(col = GENUS, into = c("UNCLASSIFIED", "UNCLASSIFIED.2"), remove = FALSE, sep = " ") %>% 
  
  filter(UNCLASSIFIED != "unclassified" & is.na(UNCLASSIFIED.2)) %>% 
  
  select(-UNCLASSIFIED, -UNCLASSIFIED.2) %>% 
  
  filter(GENUS.PERC >= 0.1 & OCCURRENCE >= 5) %>% 
  
  select(SITE, PLOT, GENUS, PROP.ABUND)
```

```{r calculate_functional_group_abundances}
## Read in genus classifications
classified_genera <- read_tsv("data/working/classified_genera.txt") %>% 
  
  select(GENUS, FUNCTIONAL.GROUP)

## Calculate functional group abundances
functional_group_relabund <- genus_relabund %>% 
  
  inner_join(classified_genera, ., by = "GENUS") %>% 
  
  group_by(SITE, PLOT, FUNCTIONAL.GROUP) %>% 
  
  summarise(PROP.ABUND = sum(PROP.ABUND))  %>% 
  
  ungroup()
```

```{r combine_environmental_data}
## Read in plot organization
plot_org <- read_tsv(file = "data/site/plot_organization.txt")

## Combine and format environmental data
env_data <- plot_coordinates %>% 
  
  inner_join(plot_org, ., by = "PLOT") %>% 
  
  inner_join(., root_mass_loss, by = "PLOT") %>% 
  
  inner_join(., decomp_root_moisture, by = "PLOT") %>% 
  
  inner_join(., soil_inorg_N, by = "PLOT") %>% 
  
  inner_join(., soil_vwc_temp, by = "PLOT") %>% 
  
  inner_join(., soil_pH, by = "PLOT") %>% 
  
  inner_join(., pre_root_C_and_N_mean, by = "SITE") %>% 
  
  inner_join(., decomp_root_C_and_N, by = "PLOT") %>% 
  
  inner_join(., soil_C_and_N_tbl, by = "PLOT") %>% 
  
  inner_join(., pre_root_pygcms_source_mean, by = "SITE")

# Filter outliers
env_data_trimmed <- env_data %>% 
  
  ## Filter outliers
  filter(PLOT != "Plot_19" & PROP.MASS.LOSS < 0.4)
```

```{r calculate_absolute_fungal_abundance}
# Get root DNA extraction masses
roots.DNA.extraction.masses.tbl <- read_tsv("data/qPCR/root_extraction_mass.txt") %>% 
  
  # Format
  dplyr::rename(PLOT = "Sample number (plot)", 
                REPL = "Replicate", 
                MASS = "sample mass") %>% 
  dplyr::mutate(PLOT = gsub("\\ ", "_", PLOT)) %>% 
  
  # Calculate total mass
  dplyr::group_by(PLOT) %>% 
  dplyr::summarise(MASS = sum(MASS)) %>% 
  dplyr::ungroup() %>% 
  
  # Add root moisture and ash content, calculate dry mass
  dplyr::inner_join(., decomp_root_moisture, by = "PLOT") %>% 
  dplyr::inner_join(., decomp_ash, by = "PLOT") %>% 
  dplyr::mutate(DRY.MASS = (MASS * (1 - MOISTURE.CONTENT) * ASH.FREE.PROP))
  
# Read in qPCR data
roots.ITS.copies.tbl1 <- read_tsv(file = "data/qPCR/ITS1_R_qPCR1_individual_full_curve_data.txt") %>% 
  
  # Trim
  dplyr::filter(`Well Type` == "Unknown" & Threshold != "Reference") %>% 
  dplyr::mutate(Quantity = as.numeric(Quantity), 
                PLOT = c(1 : nrow(.)), 
                PLOT = paste("Plot", PLOT, sep = "_"), 
                RUN = "Run_1") %>% 
  rename(INITIAL.COPIES = "Quantity") %>% 
  dplyr::select(PLOT, 
                RUN, 
                INITIAL.COPIES)

# Read in qPCR data
roots.ITS.copies.tbl2 <- read_tsv(file = "data/qPCR/ITS1_R_qPCR2_individual_full_curve_data.txt") %>% 
  
  # Trim
  dplyr::filter(`Well Type` == "Unknown" & Threshold != "Reference") %>% 
  dplyr::mutate(Quantity = as.numeric(Quantity), 
                PLOT = c(1 : nrow(.)), 
                PLOT = paste("Plot", PLOT, sep = "_"), 
                RUN = "Run_2") %>% 
  rename(INITIAL.COPIES = "Quantity") %>% 
  dplyr::select(PLOT, 
                RUN, 
                INITIAL.COPIES)

# Combine
roots_copy_number <- dplyr::bind_rows(roots.ITS.copies.tbl1, 
                                      roots.ITS.copies.tbl2) %>% 
  dplyr::group_by(PLOT) %>% 
  dplyr::summarise(ROOTS.INITIAL.COPIES = mean(INITIAL.COPIES)) %>% 
  dplyr::ungroup() %>% 
  
  # Add root mass
  dplyr::inner_join(., roots.DNA.extraction.masses.tbl, by = "PLOT") %>% 
  dplyr::mutate(VOLUME = 1, 
                DILUTION.FACTOR = 10, 
                CLEANUP.OUTPUT = 100, 
                CLEANUP.INPUT = 150, 
                EXTR.VOLUME = 600) %>% 
  
  # Calculate copy number
  dplyr::mutate(ROOTS.COPY.NUMBER = (ROOTS.INITIAL.COPIES * VOLUME * DILUTION.FACTOR * CLEANUP.OUTPUT * (1 / CLEANUP.INPUT) * EXTR.VOLUME * (1 / DRY.MASS))) %>% 
  
  ## Add site
  inner_join(plot_org, ., by = "PLOT") %>% 
  
  ## Add root mass loss
  inner_join(., root_mass_loss, by = "PLOT") %>% 
  
  ## Filter outliers
  filter(PLOT != "Plot_19" & PROP.MASS.LOSS < 0.4) %>% 
  
  select(PLOT, ROOTS.COPY.NUMBER)
```

```{r calculate_species_level_normalized_pcwde_gene_counts}
## Read in PCWDE gene names
pcwde_cazymes <- read_tsv("data/genome_data/pcwde_genes.txt")

## Read in CAZyme data into a list of data frames
cazyme_counts_files <- list.files(path = "data/genome_data/cazyme_counts")

cazyme_counts_files <- paste("data/genome_data/cazyme_counts/", cazyme_counts_files, sep = "")

cazyme_counts <- lapply(cazyme_counts_files, read_tsv)

## Write a function to obtain annotation information
obtain_annotations <- function(x) {
  
  ## Get first and last columns
  tmp1 <- x %>% 
    select(`Annotations/Genomes`, `Annotation Description`)
  
  ## Return result
  return(tmp1)
  
}

## Get annotation descriptions and compile into a single table without duplicates
annotation_descriptions <- lapply(cazyme_counts, obtain_annotations) %>% 
  
  bind_rows(.) %>% 
  
  distinct(.)

## Write a function to trim each table to include only target PCWDE
pcwde_only <- function(x) {
  
  ## Get table, update column name, and remove last 2 columns
  tmp1 <- x %>% 
    
    rename(cazyme = "Annotations/Genomes") %>% 
    
    select(-`Annotation Description`)
  
  ## Merge with PCWDE table, and keep only PCWDEs
  tmp2 <- tmp1 %>% 
    
    left_join(pcwde_cazymes, ., by = "cazyme") %>% 
    
    select(-gene_category)
  
  ## Return
  return(tmp2)
  
}

## Trim CAZyme counts to include PCWDEs only, combine into single data frame, and convert to long format
pcwde_counts <- lapply(cazyme_counts, pcwde_only) %>% 
  
  reduce(., left_join, by = "cazyme") %>% 
  
  pivot_longer(., cols = -cazyme, names_to = "mycocosm_species_code", values_to = "gene_number") %>% 
  
  inner_join(pcwde_cazymes, ., by = "cazyme")

## Read in species with sequenced genomes matching genera in the roots dataset
species_with_genomes <- read_tsv(file = "data/genome_data/species_with_genomes.txt")

## Get a table of genera in the roots dataset
roots_genera <- genus_relabund %>% 
  
  select(GENUS) %>% 
  
  distinct()

## Merge species codes with PCWDE count data, remove genera without genomes, 
## normalize counts to total genes, replace NA with 0, and trim unnecessary columns
species_level_normalized_pcwde_counts <- pcwde_counts %>% 
  
  inner_join(species_with_genomes, ., by = "mycocosm_species_code") %>% 
  
  mutate(
    
    normalized_gene_number = 10000 * (gene_number / total_gene_number), ## genes per 10 thousand genes
    
    gene_number = ifelse(is.na(gene_number), 0, gene_number), 
    
    normalized_gene_number = ifelse(is.na(normalized_gene_number), 0, normalized_gene_number)
    
  ) %>% 
  
  select(mycocosm_species_code, genus, cazyme, gene_category, gene_number, normalized_gene_number) %>% 
  
  rename(GENUS = "genus") %>% 
  
  ## Trim to include only genera in the roots dataset
  inner_join(., roots_genera, by = "GENUS") %>% 
  
  ## Add functional group assignments
  inner_join(., classified_genera, by = "GENUS") %>% 
  
  ## Trim to include the 4 main functional groups
  filter(
    
    FUNCTIONAL.GROUP == "SAP.LIG" | 
      
      FUNCTIONAL.GROUP == "SAP.NONLIG" | 
      
      FUNCTIONAL.GROUP == "ECM.PEROX" | 
      
      FUNCTIONAL.GROUP == "ECM.NOPEROX"
    
  )
```

```{r calculate_pcwde_cwm}
## Calculate average normalized gene number per PCWDE CAZyme by genus
genus_level_normalized_pcwde_counts <- species_level_normalized_pcwde_counts %>% 
  
  group_by(GENUS, cazyme) %>% 
  
  summarise(mean_normalized_gene_number = mean(normalized_gene_number))

## CWM
pcwde_cwm <- genus_level_normalized_pcwde_counts %>% 
  
  inner_join(genus_relabund, ., by = "GENUS") %>% 
  
  group_by(SITE, PLOT, cazyme) %>% 
  
  summarise(CWM = weighted.mean(mean_normalized_gene_number, PROP.ABUND)) %>% 
  
  ungroup() %>% 
  
  filter(cazyme != "GH42")

total_pcwde_cwm <- pcwde_cwm %>% 
  
  group_by(PLOT) %>% 
  
  summarise(TOTAL.CWM = sum(CWM)) %>% 
  
  ungroup()
```

```{r pcwde_titan}
## Mass loss gradient
root_mass_loss_titan_input <- root_mass_loss %>% 
  
  ## Filter outliers
  filter(PLOT != "Plot_19" & PROP.MASS.LOSS < 0.4) %>% 
  
  # Format
  arrange(PLOT) %>% 
  
  column_to_rownames(var = "PLOT") %>% 
  
  as.data.frame(.)

## Input
pcwde_titan_input <- pcwde_cwm %>% 
  
  # Format
  select(PLOT, cazyme, CWM) %>% 
  
  arrange(cazyme) %>% 
  
  pivot_wider(id_cols = PLOT, names_from = "cazyme", values_from = "CWM") %>% 
  
  arrange(PLOT) %>% 
  
  column_to_rownames(var = "PLOT") %>% 
  
  as.data.frame(.)

## pcwde TITAN
# pcwde_titan3 <- titan(
# 
#   root_mass_loss_titan_input,
# 
#   pcwde_titan_input,
# 
#   minSplt = 5,
# 
#   numPerm = 1000,
# 
#   boot = TRUE,
# 
#   nBoot = 1000,
# 
#   imax = FALSE,
# 
#   ivTot = FALSE,
# 
#   pur.cut = 0.95,
# 
#   rel.cut = 0.95,
# 
#   ncpus = 4,
# 
#   memory = TRUE
# 
# )

## Save data to reduce computational time later
# saveRDS(pcwde_titan3, file = "data/working/pcwde_titan3.rData")
```

```{r get_fast_decay_pcwde}
## Read in TITAN results to save time
pcwde_titan3 <- readRDS(file = "data/working/pcwde_titan3.rData")

## Get genes associated with fast decomposition
fast_pcwde <- as.data.frame(pcwde_titan3$sppmax) %>% 
  
  rownames_to_column(var = "cazyme") %>% 
  
  as_tibble(.) %>% 
  
  filter(filter == 2) %>% 
  
  select(cazyme)

## Write out
# write_tsv(fast_pcwde, file = "data/working/fast_pcwde3.txt")
```

```{r calculate_genus_level_fast_pcwde_gene_counts}
## Calculate average normalized gene number per PCWDE CAZyme by genus
genus_fast_pcwde <- species_level_normalized_pcwde_counts %>% 
  
  inner_join(., fast_pcwde, by = "cazyme") %>% 
  
  group_by(FUNCTIONAL.GROUP, GENUS, cazyme) %>% 
  
  summarise(
    
    MEAN.FAST = mean(normalized_gene_number), 
    
    SD.FAST = sd(normalized_gene_number)
    
  ) %>% 
  
  ungroup()
```

```{r calculate_fast_pcwde_cwm}
#### Calculate community-weighted mean fast pcwde counts by plot ####
fast_pcwde_cwm <- genus_relabund %>% 
  
  inner_join(., genus_fast_pcwde, by = "GENUS") %>% 
  
  group_by(SITE, PLOT, cazyme) %>% 
  
  summarise(CWM.FAST = weighted.mean(MEAN.FAST, PROP.ABUND)) %>% 
  
  ungroup() %>% 
  
  group_by(SITE, PLOT) %>% 
  
  summarise(CWM.FAST.TOTAL = sum(CWM.FAST)) %>% 
  
  ungroup() %>% 
  
  select(PLOT, CWM.FAST.TOTAL)
```

## Figure 1a  

```{r format_data_for_fast_decay_pcwde_figure_fig1a}
## Read in substrates
fast_pcwde_substrates <- read_tsv(file = "data/working/fast_pcwde_substrates.txt")

## Get genes associated with fast decomposition
fast_pcwde_input <- as.data.frame(pcwde_titan3$sppmax) %>% 
  
  rownames_to_column(var = "cazyme") %>% 
  
  as_tibble(.) %>% 
  
  filter(filter > 0) %>% 
  
  inner_join(., pcwde_cazymes, by = "cazyme") %>% 
  
  mutate(
    
    z.median = ifelse(filter == 1, -1 * z.median, z.median), 
    
    GENE.CATEGORY.LABEL = NA, 
    
    GENE.CATEGORY.LABEL = ifelse(gene_category == "OX_RE", "Oxidoreductase", GENE.CATEGORY.LABEL), 
    
    GENE.CATEGORY.LABEL = ifelse(gene_category == "CE", "Carbohydrate esterase", GENE.CATEGORY.LABEL), 
    
    GENE.CATEGORY.LABEL = ifelse(gene_category == "GH", "Glycoside hydrolase", GENE.CATEGORY.LABEL), 
    
    GENE.CATEGORY.LABEL = ifelse(gene_category == "PL", "Polysaccharide lyase", GENE.CATEGORY.LABEL), 
    
    GENE.CATEGORY.LABEL = ifelse(gene_category == "CBM", "Carbohydrate-binding module", GENE.CATEGORY.LABEL), 
    
    GENE.CATEGORY.LABEL = factor(
      
      GENE.CATEGORY.LABEL, 
      
      levels = c("Oxidoreductase", "Carbohydrate esterase", "Carbohydrate-binding module", "Polysaccharide lyase", "Glycoside hydrolase")
      
    )
    
  ) %>% 
  
  full_join(., fast_pcwde_substrates, by = "cazyme") %>% 
  
  mutate(
    
    substrate = ifelse(cazyme == "GH88", "Hemicellulose", substrate), 
    
    substrate = factor(substrate, levels = c(
      
      "Cellulose", 
      
      "Cellulose and hemicellulose", 
      
      "Hemicellulose", 
      
      "Hemicellulose and cutin", 
      
      "Pectin"
      
    )
    
    )
    
  ) %>% 
  
  arrange(GENE.CATEGORY.LABEL, desc(z.median)) %>% 
  
  mutate(cazyme = factor(cazyme, levels = unique(cazyme)))
```

```{r fig1a_fast_pcwde}
## Fast decomp PCWDE figure
fig1a_fast_pcwde_z <- ggplot() + 
  
  geom_point(
    
    data = fast_pcwde_input, 
    
    aes(x = cazyme, y = z.median, shape = GENE.CATEGORY.LABEL, fill = substrate), 
    
    size = 3
    
  ) + 
  
  coord_flip() + 
  
  geom_hline(
    
    yintercept = 0, 
    
    colour = "black", 
    
    linetype = "dashed", 
    
    size = 0.25
    
  ) + 
  
  scale_x_discrete(limits = rev(levels(fast_pcwde_input$cazyme))) + 
  
  ## Shapes
  scale_shape_manual(
    
    name = "Gene category", 
    
    values = c(21, 24, 25, 23, 22)
    
  ) + 
  
  ## Colors
  scale_fill_viridis(
    
    name = "Substrate", 
    
    discrete = TRUE, 
    
    option = "plasma"
    
  ) + 
  
  ## Titles
  labs(
    
    title = "(a)", 
    
    x = "CAZyme gene family", 
    
    y = "Median Z-score\n(association with fine root decay)"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_line(colour = "grey80", size = 0.25), 
    
    panel.grid.minor = element_line(colour = "grey80", size = 0.25)
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    panel.grid.major.x = element_blank(), 
    
    panel.grid.minor.x = element_blank(), 
    
    axis.title.y = element_text(colour = "black", size = 10), 
    
    axis.title.x = element_text(colour = "black", size = 10), 
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  ) + 
  
  ## Format legends
  theme( 
    
    legend.key = element_blank(), 
    
    legend.title = element_text(size = 10), 
    
    legend.text = element_text(size = 8)
    
  ) + 
  
  guides(fill = guide_legend(override.aes = list(
    
    size = 4, 
    
    pch =  21
    
  )))
```

```{r genus_summary}
genus_level_normalized_pcwde <- genus_level_normalized_pcwde_counts %>% 
  
  select(GENUS) %>% 
  
  distinct() %>% 
  
  pull(GENUS)

abund_genera <- genus_relabund %>% 
  
  select(GENUS) %>% 
  
  distinct() %>% 
  
  pull(GENUS)

genus_functional_groups <- genus_all %>% 
  
  inner_join(., classified_genera, by = "GENUS") %>% 
  
  filter(GENUS %in% abund_genera) %>% 
  
  filter(
    
    FUNCTIONAL.GROUP == "SAP.LIG" | 
      
      FUNCTIONAL.GROUP == "SAP.NONLIG" | 
      
      FUNCTIONAL.GROUP == "ECM.PEROX" | 
      
      FUNCTIONAL.GROUP == "ECM.NOPEROX"
    
  ) %>% 
  
  summarise(total_seqs = sum(COUNT)) %>% 
  
  pull(total_seqs)

genus_assigned_pcwde <- genus_all %>% 
  
  filter(GENUS %in% genus_level_normalized_pcwde) %>% 
  
  summarise(total_seqs = sum(COUNT)) %>% 
  
  pull(total_seqs)

asv <- roots_asv %>% 
  
  filter(PLOT %in% unique(genus_all$PLOT)) %>% 
  
  group_by(ASV) %>% 
  
  summarise(total = sum(COUNT)) %>% 
  
  ungroup() %>% 
  
  filter(total > 0)
```

```{r fast_pcwde_summary}
fast_pcwde_summary <- fast_pcwde_input %>% 
  
  filter(z.median > 0) %>% 
  
  mutate(substrate2 = ifelse(substrate == "Hemicellulose" | substrate == "Cellulose and hemicellulose" | substrate == "Cellulose" | substrate == "Hemicellulose and cutin", "Cellulose and/or hemicellulose", as.character(substrate))) %>% 
  
  group_by(substrate2) %>% 
  
  summarise(substrate_perc = 100 * (n() / 23)) %>% 
  
  ungroup() %>% 
  
  arrange(desc(substrate_perc))
```

```{r format_enzyme_data}
#### Normalize ####

## Normalize enzyme activity to ash-free dry mass of roots
enzymes <- read_tsv(file = "data/enzymes/enzyme_data_uncorrected.txt") %>% 
  
  # Format plot column and convert to long format
  mutate(PLOT = paste("Plot", PLOT, sep = "_")) %>% 
  
  pivot_longer(
    
    cols = BGLUC : PEROX, 
    
    names_to = "ENZYME", 
    
    values_to = "ACTIVITY"
    
  ) %>% 
  
  # Add ash free proportion, correct enzyme data, and pivot wider
  inner_join(., decomp_ash, by = "PLOT") %>% 
  
  mutate(ACTIVITY.CORRECTED = ACTIVITY / ASH.FREE.PROP) %>% 
  
  select(PLOT, ENZYME, ACTIVITY.CORRECTED) %>% 
  
  pivot_wider(
    
    id_cols = PLOT, 
    
    names_from = "ENZYME", 
    
    values_from = "ACTIVITY.CORRECTED"
    
  )
```

```{r env_pca}
#### Environment PCA ####

## Get input data for PCA
env_pca_input <- env_data_trimmed %>% 
  
  select(PLOT, N.MIN, SOIL.PH, TEMP.MEAN, VWC.MEAN) %>% 
  
  column_to_rownames(var = "PLOT")

## Run and summarize environmental PCA
env_pca <- rda(env_pca_input, scale = TRUE)

env_pca_summary <- summary(env_pca)

## Format eigenvalues
env_pca_pc1_eig <- round(100 * (as.numeric(env_pca_summary$cont$importance[ ,1][2])), 0)
env_pca_pc2_eig <- round(100 * (as.numeric(env_pca_summary$cont$importance[ ,2][2])), 0)

## Format plot loadings
env_pca_points <- as.data.frame(env_pca_summary$sites) %>% 
  
  rownames_to_column(var = "PLOT") %>% 
  
  as_tibble(.)

## Format variable loadings
env_pca_variables <- as.data.frame(env_pca_summary$species) %>% 
  
  rownames_to_column(var = "ENV.VARIABLE") %>% 
  
  as_tibble(.) %>% 
  
  mutate(
    
    VARIABLE.LABEL = NA, 
    
    VARIABLE.LABEL = ifelse(ENV.VARIABLE == "N.MIN", "Inorg. N", VARIABLE.LABEL), 
    
    VARIABLE.LABEL = ifelse(ENV.VARIABLE == "SOIL.PH", "Soil pH", VARIABLE.LABEL), 
    
    VARIABLE.LABEL = ifelse(ENV.VARIABLE == "TEMP.MEAN", "Soil temp.", VARIABLE.LABEL), 
    
    VARIABLE.LABEL = ifelse(ENV.VARIABLE == "VWC.MEAN", "Soil water", VARIABLE.LABEL)
    
  ) %>% 
  
  mutate(
    
    PC1.NEW = ifelse(PC1 < 0, PC1 - 0.4, PC1 + 0.4), 
    
    PC2.NEW = ifelse(PC2 < 0, PC2 - 0.15, PC2 + 0.15), 
    
    PC2.NEW = ifelse(ENV.VARIABLE == "N.MIN", PC2 - 0.05, PC2.NEW)
    
  )
```

```{r fig_s6_env_pca_fig}
#### Fig. Environment PCA ####
fig_env_pca <- ggplot() + 
  
  ## Horizontal and vertical dashed lines
  geom_hline(
    
    yintercept = 0, 
    
    linetype = "dashed", 
    
    size = 0.5, 
    
    colour = "grey30"
    
  ) + 
  
  geom_vline(
    
    xintercept = 0, 
    
    linetype = "dashed", 
    
    size = 0.5, 
    
    colour = "grey30"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = env_pca_points, 
    
    aes(x = PC1, y = PC2), 
    
    size = 1.5, 
    
    pch = 1
    
  ) + 
  
  ## Add vectors
  geom_segment(
    
    data = env_pca_variables, 
    
    aes(x = 0, y = 0, xend = PC1, yend = PC2), 
    
    colour = "grey30", 
    
    arrow = arrow(angle = 30, length = unit(0.03, "npc"), type = "open")
    
  ) + 
  
  ## Labels
  geom_text(
    
    data = env_pca_variables, 
    
    aes(x = PC1.NEW, y = PC2.NEW, label = VARIABLE.LABEL), 
    
    colour = "black", 
    
    size = 2.5
    
  ) + 
  
  ## Scale limits
  scale_x_continuous(limits = c(-2.5, 2.5)) + 
  
  scale_y_continuous(limits = c(-2.5, 2.5)) + 
  
  ## Add plot and axis titles
  labs(
    
    title = NULL, 
    
    x = bquote('PC1 (' * .(env_pca_pc1_eig) * '%)'),
    
    y = bquote('PC2 (' * .(env_pca_pc2_eig) * '%)')
    
  ) +
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA),
    
    panel.grid.major = element_blank(),
    
    panel.grid.minor = element_blank()
    
  ) +
  
  ## Format text
  theme(
    
    plot.title = element_text(colour = "black", size = 12, hjust = -0.28), 
    
    axis.title.x = element_text(colour = "black", size = 8), 
    
    axis.title.y = element_text(colour = "black", size = 8), 
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black")
    
  ) + 
  
  ## Format legends
  theme( 
    
    legend.key = element_blank(), 
    
    legend.title = element_text(colour = "black", size = 10), 
    
    legend.text = element_text(colour = "black", size = 8)
    
  )

## Save figure
ggsave2(

  "figures/MolEcol_revision/Figure_S6.pdf",

  fig_env_pca,

  device = "pdf",

  width = 6.5,

  height = 6.5,

  units = "in"

)
```

```{r combine_env_and_fungal_data}
## Format figure input
env_fungi_data <- functional_group_relabund %>% 
  
  pivot_wider(id_cols = PLOT, names_from = "FUNCTIONAL.GROUP", values_from = "PROP.ABUND") %>% 
  
  inner_join(., fast_pcwde_cwm, by = "PLOT") %>% 
  
  inner_join(., total_pcwde_cwm, by = "PLOT") %>% 
  
  inner_join(., roots_copy_number, by = "PLOT") %>% 
  
  inner_join(., enzymes, by = "PLOT") %>% 
  
  inner_join(., env_data, by = "PLOT") %>% 
  
  ## Enzymes
  
  mutate(

    CELL.ENZYMES = BGLUC + CELLO,

    LIG.ENZYMES = PHENOX + PEROX

  ) %>%
  
  # PCA loadings
  inner_join(., env_pca_points, by = "PLOT") %>% 
  
  ## Absolute abundances
  mutate(
    
    ECM.PEROX.ABS = ROOTS.COPY.NUMBER * ECM.PEROX, 
    
    ECM.NOPEROX.ABS = ROOTS.COPY.NUMBER * ECM.NOPEROX, 
    
    SAP.LIG.ABS = ROOTS.COPY.NUMBER * SAP.LIG, 
    
    SAP.NONLIG.ABS = ROOTS.COPY.NUMBER * SAP.NONLIG
    
  ) %>% 
  
  mutate(GROUP = "Group") %>% 
  
  filter(PLOT != "Plot_19" & PROP.MASS.LOSS < 0.4)
```

```{r variable_correlations}
# Input data
env_fungi_corr_input <- env_fungi_data %>% 
  
  select(PLOT, N.MIN, VWC.MEAN, TEMP.MEAN, SOIL.PH, PC1, PC2) %>% 
  
  column_to_rownames(var = "PLOT") %>% 
  
  as.data.frame()

# Correlations
env_fungi_corr <- corr.test( env_fungi_corr_input, adjust = "none", ci = FALSE)
```

```{r function_to_get_gamm_pval}
## Write function to get pvalue from gam summary
gamm_pval <- function(x) {
  
  ## Label
  if(x < 0.001) {
    
    tmp1 <- "italic(P) < 0.001"
    
  } else if(x <= 0.05 & x >= 0.001) {
    
    tmp1 <- paste("italic(P) == ", format(round(x, 3), nsmall = 3), sep = "")
    
  } else if(x > 0.05) {
    
    tmp1 <- "italic(n.s.)"
    
  } else {
    
    tmp1 <- NA
    
  }
  
  ## Return output
  return(tmp1)
  
}
```

```{r gamm_for_decay_and_pcwde}
#### Decay ~ all PCWDE GAMM ####
decay_pcwde_gamm <- gamm(
  
  PROP.MASS.LOSS ~ s(TOTAL.CWM, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

decay_pcwde_gamm_summary <- summary(decay_pcwde_gamm$gam)

decay_pcwde_gamm_plot <- plot(decay_pcwde_gamm$gam, residuals = TRUE)

decay_pcwde_fig_data <- decay_pcwde_gamm_plot[[1]]

decay_pcwde_fit <- as_tibble(decay_pcwde_fig_data[c("x", "se", "fit")])

decay_pcwde_input <- as_tibble(decay_pcwde_fig_data[c("raw", "p.resid")])

decay_pcwde_gamm_stats <- tibble(
  
  PVAL = decay_pcwde_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 62.5, 
  
  YPOS = 0.155) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE), 
         
         PVAL.LABEL = paste("atop(R[adj.]^2 == ", 
                            round(decay_pcwde_gamm_summary$r.sq, 3), 
                            ", ", 
                            PVAL.LABEL, ")", 
                            sep = ""))

#### Decay ~ fast PCWDE GAMM ####
decay_fast_pcwde_gamm <- gamm(
  
  PROP.MASS.LOSS ~ s(CWM.FAST.TOTAL, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

decay_fast_pcwde_gamm_summary <- summary(decay_fast_pcwde_gamm$gam)

decay_fast_pcwde_gamm_plot <- plot(decay_fast_pcwde_gamm$gam, residuals = TRUE)

decay_fast_pcwde_fig_data <- decay_fast_pcwde_gamm_plot[[1]]

decay_fast_pcwde_fit <- as_tibble(decay_fast_pcwde_fig_data[c("x", "se", "fit")])

decay_fast_pcwde_input <- as_tibble(decay_fast_pcwde_fig_data[c("raw", "p.resid")])

decay_fast_pcwde_gamm_stats <- tibble(
  
  PVAL = decay_fast_pcwde_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 20, 
  
  YPOS = 0.155) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE), 
         
         PVAL.LABEL = paste("atop(R[adj.]^2 == ", 
                            round(decay_fast_pcwde_gamm_summary$r.sq, 3), 
                            ", ", 
                            PVAL.LABEL, ")", 
                            sep = ""))
```

```{r decay_pcwde_env_gamm}
#### Decay ~ env PCWDE GAMM ####
decay_pcwde_env_gamm <- gamm(
  
  PROP.MASS.LOSS ~ s(TOTAL.CWM, k = -1) + s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

summary(decay_pcwde_env_gamm$gam)

#### Decay ~ fast env PCWDE GAMM ####
decay_fast_pcwde_env_gamm <- gamm(
  
  PROP.MASS.LOSS ~ s(CWM.FAST.TOTAL, k = -1) + s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

summary(decay_fast_pcwde_env_gamm$gam)
```

```{r check_decay_pcwde_gamm}
# All PCWDE
gam.check(decay_pcwde_gamm$gam)

# Rapid PCWDE
gam.check(decay_fast_pcwde_gamm$gam)

# All PCWDE env
gam.check(decay_pcwde_env_gamm$gam)

# Rapid PCWDE env
gam.check(decay_fast_pcwde_env_gamm$gam)
```

## Figure 1b  

```{r fig1b_decay_all_pcwde}
## Fig
decay_all_pcwde_fig1b <- ggplot() + 
  
  ## CI
  geom_ribbon(
    
    data = decay_pcwde_fit, 
    
    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL), 
    
    fill = "grey80"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = decay_pcwde_input, 
    
    aes(x = raw, y = p.resid), 
    
    colour = "#233E6CFF", 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(
    
    data = decay_pcwde_fit, 
    
    aes(x = x, y = fit + se), 
    
    linetype = 2, 
    
    colour = "#233E6CFF"
    
  ) + 
  
  ## Lower CI
  geom_line(
    
    data = decay_pcwde_fit, 
    
    aes(x = x, y = fit - se), 
    
    linetype = 2, 
    
    colour = "#233E6CFF"
    
  ) + 
  
  ## Line
  geom_line(
    
    data = decay_pcwde_fit, 
    
    aes(x = x, y = fit), 
    
    colour = "#233E6CFF"
    
  ) + 
  
  ## P values
  geom_text(
    
    data = decay_pcwde_gamm_stats, 
    
    aes(x = XPOS, y = YPOS, label = PVAL.LABEL), 
    
    parse = TRUE, 
    
    stat = "identity", 
    
    size = 3
    
  ) + 
  
  ## Titles
  labs(
    
    title = "(b)", 
    
    x = "All PCWDE per 10,000 genes", 
    
    y = "Partial fine root decay"
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 10), 
    
    axis.title.y = element_text(colour = "black", size = 10),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Figure 1c  

```{r decay_fast_pcwde_fig1c}
## Fig
decay_fast_pcwde_fig1c <- ggplot() + 
  
  ## CI
  geom_ribbon(
    
    data = decay_fast_pcwde_fit, 
    
    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL), 
    
    fill = "grey80"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = decay_fast_pcwde_input, 
    
    aes(x = raw, y = p.resid), 
    
    colour = "#C8B86AFF", 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(

    data = decay_fast_pcwde_fit,

    aes(x = x, y = fit + se),

    linetype = 2,

    colour = "#C8B86AFF"

  ) +
  
  ## Lower CI
  geom_line(

    data = decay_fast_pcwde_fit,

    aes(x = x, y = fit - se),

    linetype = 2,

    colour = "#C8B86AFF"

  ) +
  
  ## Line
  geom_line(

    data = decay_fast_pcwde_fit,

    aes(x = x, y = fit),

    colour = "#C8B86AFF"
    
  ) +
  
  ## P values
  geom_text(

    data = decay_fast_pcwde_gamm_stats,

    aes(x = XPOS, y = YPOS, label = PVAL.LABEL),

    parse = TRUE,

    stat = "identity",

    size = 3

  ) +
  
  ## Titles
  labs(
    
    title = "(c)", 
    
    x = "Rapid-decay PCWDE per 10,000 genes", 
    
    y = "Partial fine root decay"
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 10), 
    
    axis.title.y = element_text(colour = "black", size = 10),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Assemble Figure 1  

```{r assemble_fig1}
## Assemble figure
Figure_1_aligned <- align_plots(
  
  fig1a_fast_pcwde_z, 
  
  decay_all_pcwde_fig1b, 
  
  align = "v", 
  
  axis = "l"
  
)

Figure_1_bottomrow <- plot_grid(
  
  Figure_1_aligned[[2]], decay_fast_pcwde_fig1c, 
  
  labels = NULL, 
  
  align = "hv"
  
  )

Figure_1 <- plot_grid(
  
  Figure_1_aligned[[1]],  Figure_1_bottomrow, 
  
  ncol = 1, 
  
  rel_heights = c(1.33, 1)
  
  )

## Save figure
ggsave2(

  "figures/MolEcol_revision/Figure_1.pdf",

  Figure_1,

  device = "pdf",

  width = 6.5,

  height = 7.5,

  units = "in"

)
```

## Figure 2a

```{r fig2a_pcwde_by_functional_group}
#### Write a function to get functional group labels ####
functional_group_labels <- function(x) {
  
  if(x == "ECM.PEROX") {
    
    tmp1 <- "ECM\nperox."
    
  } else if(x == "ECM.NOPEROX") {
    
    tmp1 <- "ECM no\nperox."
    
  } else if(x == "SAP.LIG") {
    
    tmp1 <- "Lig.\nsaps."
    
  } else if(x == "SAP.NONLIG") {
    
    tmp1 <- "Non-lig.\nsaps."
    
  } else if(x == "OTHER.MYCORRHIZA") {
    
    tmp1 <- "Other myc."
    
  } else if(x == "OTHER") {
    
    tmp1 <- "Other"
    
  } else {
    
    tmp1 <- NA
    
  }
  
  ## Return result
  return(tmp1)
  
}

#### Get all PCWDE ####
pcwde_functional_groups <- species_level_normalized_pcwde_counts %>% 
  
  group_by(FUNCTIONAL.GROUP, GENUS, mycocosm_species_code) %>% 
  
  summarise(normalized_gene_number = sum(normalized_gene_number)) %>% 
  
  ungroup() %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = sapply(FUNCTIONAL.GROUP, functional_group_labels, simplify = TRUE), 
    
    FUNCTIONAL.GROUP.LABEL = factor(FUNCTIONAL.GROUP.LABEL, levels = c("Lig.\nsaps.", "Non-lig.\nsaps.", "ECM\nperox.", "ECM no\nperox."))
    
  ) %>% 
  
  filter(!is.na(FUNCTIONAL.GROUP.LABEL)) %>% 
  
  mutate(PCWDE = "All PCWDEs")

# ANOVA
pcwde_functional_groups_aov <- aov(normalized_gene_number ~ FUNCTIONAL.GROUP.LABEL, data = pcwde_functional_groups)
TukeyHSD(pcwde_functional_groups_aov)

pcwde_functional_groups_aov_stats <- pcwde_functional_groups %>% 
  
  select(FUNCTIONAL.GROUP.LABEL) %>% 
  
  distinct() %>% 
  
  arrange(FUNCTIONAL.GROUP.LABEL) %>% 
  
  mutate(
    
    SIG.LABEL = c("a", "b", "c", "bc"), 
    
    YPOS = 225
    
  ) %>% 
  
  mutate(PCWDE = "All PCWDEs")

pcwde_mean_functional_groups <- species_level_normalized_pcwde_counts %>% 
  
  group_by(FUNCTIONAL.GROUP, GENUS, mycocosm_species_code) %>% 
  
  summarise(normalized_gene_number = sum(normalized_gene_number)) %>% 
  
  ungroup() %>% 
  
  group_by(FUNCTIONAL.GROUP) %>% 
  
  summarise(
    
    MEAN.NORMALIZED.GENE.NUMBER = mean(normalized_gene_number), 
    
    UCI = mean(normalized_gene_number) + se(normalized_gene_number), 
    
    LCI = mean(normalized_gene_number) - se(normalized_gene_number)
    
  ) %>% 
  
  ungroup() %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = sapply(FUNCTIONAL.GROUP, functional_group_labels, simplify = TRUE), 
    
    FUNCTIONAL.GROUP.LABEL = factor(FUNCTIONAL.GROUP.LABEL, levels = c("Lig.\nsaps.", "Non-lig.\nsaps.", "ECM\nperox.", "ECM no\nperox."))
    
  ) %>% 
  
  filter(!is.na(FUNCTIONAL.GROUP.LABEL)) %>% 
  
  mutate(PCWDE = "All PCWDEs")

#### Get fast PCWDE ####
fast_pcwde_functional_groups <- species_level_normalized_pcwde_counts %>% 
  
  inner_join(fast_pcwde, ., by = "cazyme") %>% 
  
  group_by(FUNCTIONAL.GROUP, GENUS, mycocosm_species_code) %>% 
  
  summarise(normalized_gene_number = sum(normalized_gene_number)) %>% 
  
  ungroup() %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = sapply(FUNCTIONAL.GROUP, functional_group_labels, simplify = TRUE), 
    
    FUNCTIONAL.GROUP.LABEL = factor(FUNCTIONAL.GROUP.LABEL, levels = c("Lig.\nsaps.", "Non-lig.\nsaps.", "ECM\nperox.", "ECM no\nperox."))
    
  ) %>% 
  
  filter(!is.na(FUNCTIONAL.GROUP.LABEL)) %>% 
  
  mutate(PCWDE = "Rapid-decay PCWDEs")

# ANOVA
fast_pcwde_functional_groups_aov <- aov(normalized_gene_number ~ FUNCTIONAL.GROUP.LABEL, data = fast_pcwde_functional_groups)
TukeyHSD(fast_pcwde_functional_groups_aov)

fast_pcwde_functional_groups_aov_stats <- fast_pcwde_functional_groups %>% 
  
  select(FUNCTIONAL.GROUP.LABEL) %>% 
  
  distinct() %>% 
  
  arrange(FUNCTIONAL.GROUP.LABEL) %>% 
  
  mutate(
    
    SIG.LABEL = c("a", "b", "c", "cd"), 
    
    YPOS = 100
    
  ) %>% 
  
  mutate(PCWDE = "Rapid-decay PCWDEs")
  
fast_pcwde_mean_functional_groups <- species_level_normalized_pcwde_counts %>% 
  
  inner_join(fast_pcwde, ., by = "cazyme") %>% 
  
  group_by(FUNCTIONAL.GROUP, GENUS, mycocosm_species_code) %>% 
  
  summarise(normalized_gene_number = sum(normalized_gene_number)) %>% 
  
  ungroup() %>% 
  
  group_by(FUNCTIONAL.GROUP) %>% 
  
  summarise(
    
    MEAN.NORMALIZED.GENE.NUMBER = mean(normalized_gene_number), 
    
    UCI = mean(normalized_gene_number) + se(normalized_gene_number), 
    
    LCI = mean(normalized_gene_number) - se(normalized_gene_number)
    
  ) %>% 
  
  ungroup() %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = sapply(FUNCTIONAL.GROUP, functional_group_labels, simplify = TRUE), 
    
    FUNCTIONAL.GROUP.LABEL = factor(FUNCTIONAL.GROUP.LABEL, levels = c("Lig.\nsaps.", "Non-lig.\nsaps.", "ECM\nperox.", "ECM no\nperox."))
    
  ) %>% 
  
  filter(!is.na(FUNCTIONAL.GROUP.LABEL)) %>% 
  
  mutate(PCWDE = "Rapid-decay PCWDEs")

#### Combine data ####
combined_pcwde_functional_groups <- bind_rows(
  
  pcwde_functional_groups, 
  
  fast_pcwde_functional_groups
  
) %>% 
  
  mutate(PCWDE = factor(PCWDE, levels = c("All PCWDEs", "Rapid-decay PCWDEs")))

combined_pcwde_functional_groups_aov_stats <- bind_rows(
  
  pcwde_functional_groups_aov_stats, 
  
  fast_pcwde_functional_groups_aov_stats
  
) %>% 
  
  mutate(PCWDE = factor(PCWDE, levels = c("All PCWDEs", "Rapid-decay PCWDEs")))

combined_pcwde_mean_functional_groups <- bind_rows(
  
  pcwde_mean_functional_groups, 
  
  fast_pcwde_mean_functional_groups
  
) %>% 
  
  mutate(PCWDE = factor(PCWDE, levels = c("All PCWDEs", "Rapid-decay PCWDEs")))


#### Figure 2a ####
combined_pcwde_functional_group_fig <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = combined_pcwde_functional_groups, 
    
    aes(x = FUNCTIONAL.GROUP.LABEL, y = normalized_gene_number, colour = FUNCTIONAL.GROUP.LABEL), 
    
    position = position_jitter(width = 0.1), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Lines for means
  geom_segment(
    
    data = combined_pcwde_mean_functional_groups, 
    
    aes(
      
      x = as.integer(FUNCTIONAL.GROUP.LABEL) - 0.4, xend = as.integer(FUNCTIONAL.GROUP.LABEL) + 0.4, 
      
      y = MEAN.NORMALIZED.GENE.NUMBER, yend = MEAN.NORMALIZED.GENE.NUMBER, 
      
      colour = FUNCTIONAL.GROUP.LABEL
      
    ), 
    
    size = 0.5
    
  ) + 
  
  ## Confidence intervals
  geom_errorbar(
    
    data = combined_pcwde_mean_functional_groups, 
    
    aes(x = FUNCTIONAL.GROUP.LABEL, ymin = LCI, ymax = UCI, colour = FUNCTIONAL.GROUP.LABEL), 
    
    width = 0.35, 
    
    size = 0.5
    
  ) + 
  
  ## P values
  geom_text(
    
    data = combined_pcwde_functional_groups_aov_stats, 
    
    aes(x = FUNCTIONAL.GROUP.LABEL, y = YPOS, label = SIG.LABEL), 
    
    parse = TRUE, 
    
    stat = "identity", 
    
    size = 3
    
  ) + 
  
  ## Scale colors
  scale_colour_manual(
    
    values = c("#443A83FF", "#287C8EFF", "#31B57BFF", "#B9DE28FF"), 
    
    guide = "none"
    
  ) + 
  
  facet_wrap(~ PCWDE) + 
  
  ## Add plot and axis titles
  labs(
    
    title = "(a)", 
    
    x = NULL, 
    
    y = "Gene copies per genome"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.y = element_text(colour = "black", size = 10), 
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Figure 2b  

```{r fig2b_decay_all_pcwde}
#### GAMM ####
pcwde_fungi_gamm <- gamm(
  
  TOTAL.CWM ~ s(SAP.LIG.ABS, k = 3) + s(SAP.NONLIG.ABS, k = 3) + s(ECM.PEROX.ABS, k = -1) + s(ECM.NOPEROX.ABS, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

pcwde_fungi_gamm_summary <- summary(pcwde_fungi_gamm$gam)

pcwde_fungi_gamm_plot <- plot(pcwde_fungi_gamm$gam, residuals = TRUE)

#### PCWDE ~ lig saps ####
pcwde_saplig_fig_data <- pcwde_fungi_gamm_plot[[1]]

pcwde_saplig_fit <- as_tibble(pcwde_saplig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

pcwde_saplig_input <- as_tibble(pcwde_saplig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

pcwde_saplig_gamm_stats <- tibble(
  
  PVAL = pcwde_fungi_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 2500000000, 
  
  YPOS = -55) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

#### PCWDE ~ non-lig saps ####
pcwde_sapnonlig_fig_data <- pcwde_fungi_gamm_plot[[2]]

pcwde_sapnonlig_fit <- as_tibble(pcwde_sapnonlig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

pcwde_sapnonlig_input <- as_tibble(pcwde_sapnonlig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

pcwde_sapnonlig_gamm_stats <- tibble(
  
  PVAL = pcwde_fungi_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 750000000, 
  
  YPOS = -55) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

#### PCWDE ~ ECM perox ####
pcwde_ecmperox_fig_data <- pcwde_fungi_gamm_plot[[3]]

pcwde_ecmperox_fit <- as_tibble(pcwde_ecmperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

pcwde_ecmperox_input <- as_tibble(pcwde_ecmperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

pcwde_ecmperox_gamm_stats <- tibble(
  
  PVAL = pcwde_fungi_gamm_summary$s.table[3, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1650000000, 
  
  YPOS = -55) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

#### PCWDE ~ ECM no perox ####
pcwde_ecmnoperox_fig_data <- pcwde_fungi_gamm_plot[[4]]

pcwde_ecmnoperox_fit <- as_tibble(pcwde_ecmnoperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")

pcwde_ecmnoperox_input <- as_tibble(pcwde_ecmnoperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")

pcwde_ecmnoperox_gamm_stats <- tibble(
  
  PVAL = pcwde_fungi_gamm_summary$s.table[4, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 900000000, 
  
  YPOS = -55) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")

#### Combine input data ####
pcwde_fungi_fit <- bind_rows(
  
  pcwde_saplig_fit, 
  
  pcwde_sapnonlig_fit, 
  
  pcwde_ecmperox_fit, 
  
  pcwde_ecmnoperox_fit
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  ) %>% 
  
  mutate(
    
    fit = ifelse(FUNCTIONAL.GROUP.LABEL == "ECM without\nperoxidases", NA, fit), 
    
    se = ifelse(FUNCTIONAL.GROUP.LABEL == "ECM without\nperoxidases", NA, se), 
    
    fit = ifelse(FUNCTIONAL.GROUP.LABEL == "Non-ligninolytic\nsaprotrophs", NA, fit), 
    
    se = ifelse(FUNCTIONAL.GROUP.LABEL == "Non-ligninolytic\nsaprotrophs", NA, se)
    
  )

pcwde_fungi_input <- bind_rows(
  
  pcwde_saplig_input, 
  
  pcwde_sapnonlig_input, 
  
  pcwde_ecmperox_input, 
  
  pcwde_ecmnoperox_input
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  )

pcwde_fungi_gamm_stats <- bind_rows(
  
  pcwde_saplig_gamm_stats, 
  
  pcwde_sapnonlig_gamm_stats, 
  
  pcwde_ecmperox_gamm_stats, 
  
  pcwde_ecmnoperox_gamm_stats
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  )

#### Function for scientific notation ####

scientific_10 <- function(x) {
  
  tmp1 <- format(x, scientific = TRUE)
  
  tmp1 <- gsub("0e\\+00", "0", tmp1)
  
  tmp1 <- gsub("^(.*)e", "'\\1'e", tmp1)
  
  tmp1 <- gsub("e", "%*%10^", tmp1)
  
  tmp1 <- gsub("\\+0?", "", tmp1)
  
  parse(text = tmp1)
  
}

#### Figure 2b ####
pcwde_fungi_fig2b <- ggplot() + 
  
  ## CI
  geom_ribbon(
    
    data = pcwde_fungi_fit, 
    
    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL), 
    
    fill = "grey80"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = pcwde_fungi_input, 
    
    aes(x = raw, y = p.resid, colour = FUNCTIONAL.GROUP.LABEL), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(

    data = pcwde_fungi_fit,

    aes(x = x, y = fit + se, colour = FUNCTIONAL.GROUP.LABEL),

    linetype = 2

  ) +
  
  ## Lower CI
  geom_line(

    data = pcwde_fungi_fit,

    aes(x = x, y = fit - se, colour = FUNCTIONAL.GROUP.LABEL),

    linetype = 2

  ) + 
  
  ## Line
  geom_line(
    
    data = pcwde_fungi_fit, 
    
    aes(x = x, y = fit, colour = FUNCTIONAL.GROUP.LABEL)
    
  ) + 
  
  ## Color
  scale_colour_manual(
    
    values = c("#443A83FF", "#287C8EFF", "#31B57BFF", "#B9DE28FF"), 
    
    guide = "none"
    
  ) + 
  
  ## Fill
  scale_fill_manual(
    
    values = c("#443A83FF", "#287C8EFF", "#31B57BFF", "#B9DE28FF"), 
    
    guide = "none"
    
  ) + 
  
  ## P values
  geom_text(

    data = pcwde_fungi_gamm_stats,

    aes(x = XPOS, y = YPOS, label = PVAL.LABEL),

    parse = TRUE,

    stat = "identity",

    size = 3

  ) +
  
  ## Titles
  labs(
    
    title = "(b)", 
    
    x = bquote('Fungal functional group abundance (ITS copies'%.%g^-1~'fine root litter)'), 
    
    y = "Partial PCWDEs (all)"
    
  ) + 
  
  scale_x_continuous(
    
    breaks = breaks_extended(n = 3), 
    
    limits = c(0, NA), 
    
    expand = c(0.1, 0.1), 
    
    labels = scientific_10
    
  ) +  
  
  ## Wrap
  facet_wrap(
    
    ~ FUNCTIONAL.GROUP.LABEL, 
    
    nrow = 1, 
    
    ncol = 4, 
    
    scales = "free_x"
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 10), 
    
    axis.title.y = element_text(colour = "black", size = 10),  
    
    axis.text.x = element_text(colour = "black", size = 7.5), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Figure 2c  

```{r fig2c_decay_fast_pcwde}
#### GAMM ####
fast_pcwde_fungi_gamm <- gamm(
  
  CWM.FAST.TOTAL ~ s(SAP.LIG.ABS, k = 3) + s(SAP.NONLIG.ABS, k = 3) + s(ECM.PEROX.ABS, k = -1) + s(ECM.NOPEROX.ABS, k = -1),
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

fast_pcwde_fungi_gamm_summary <- summary(fast_pcwde_fungi_gamm$gam)

fast_pcwde_fungi_gamm_plot <- plot(fast_pcwde_fungi_gamm$gam, residuals = TRUE)

#### PCWDE ~ lig saps ####
fast_pcwde_saplig_fig_data <- fast_pcwde_fungi_gamm_plot[[1]]

fast_pcwde_saplig_fit <- as_tibble(fast_pcwde_saplig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

fast_pcwde_saplig_input <- as_tibble(fast_pcwde_saplig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

fast_pcwde_saplig_gamm_stats <- tibble(
  
  PVAL = fast_pcwde_fungi_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 2500000000, 
  
  YPOS = -21) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

#### PCWDE ~ non-lig saps ####
fast_pcwde_sapnonlig_fig_data <- fast_pcwde_fungi_gamm_plot[[2]]

fast_pcwde_sapnonlig_fit <- as_tibble(fast_pcwde_sapnonlig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

fast_pcwde_sapnonlig_input <- as_tibble(fast_pcwde_sapnonlig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

fast_pcwde_sapnonlig_gamm_stats <- tibble(
  
  PVAL = fast_pcwde_fungi_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 750000000, 
  
  YPOS = -21) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

#### PCWDE ~ ECM perox ####
fast_pcwde_ecmperox_fig_data <- fast_pcwde_fungi_gamm_plot[[3]]

fast_pcwde_ecmperox_fit <- as_tibble(fast_pcwde_ecmperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

fast_pcwde_ecmperox_input <- as_tibble(fast_pcwde_ecmperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

fast_pcwde_ecmperox_gamm_stats <- tibble(
  
  PVAL = fast_pcwde_fungi_gamm_summary$s.table[3, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1650000000, 
  
  YPOS = -21) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

#### PCWDE ~ ECM no perox ####
fast_pcwde_ecmnoperox_fig_data <- fast_pcwde_fungi_gamm_plot[[4]]

fast_pcwde_ecmnoperox_fit <- as_tibble(fast_pcwde_ecmnoperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")

fast_pcwde_ecmnoperox_input <- as_tibble(fast_pcwde_ecmnoperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")

fast_pcwde_ecmnoperox_gamm_stats <- tibble(
  
  PVAL = fast_pcwde_fungi_gamm_summary$s.table[4, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 900000000, 
  
  YPOS = -21) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")

#### Combine input data ####
fast_pcwde_fungi_fit <- bind_rows(
  
  fast_pcwde_saplig_fit, 
  
  fast_pcwde_sapnonlig_fit, 
  
  fast_pcwde_ecmperox_fit, 
  
  fast_pcwde_ecmnoperox_fit
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  ) %>% 
  
  mutate(
    
    fit = ifelse(FUNCTIONAL.GROUP.LABEL == "ECM without\nperoxidases", NA, fit), 
    
    se = ifelse(FUNCTIONAL.GROUP.LABEL == "ECM without\nperoxidases", NA, se), 
    
    fit = ifelse(FUNCTIONAL.GROUP.LABEL == "Non-ligninolytic\nsaprotrophs", NA, fit), 
    
    se = ifelse(FUNCTIONAL.GROUP.LABEL == "Non-ligninolytic\nsaprotrophs", NA, se)
    
  )

fast_pcwde_fungi_input <- bind_rows(
  
  fast_pcwde_saplig_input, 
  
  fast_pcwde_sapnonlig_input, 
  
  fast_pcwde_ecmperox_input, 
  
  fast_pcwde_ecmnoperox_input
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  )

fast_pcwde_fungi_gamm_stats <- bind_rows(
  
  fast_pcwde_saplig_gamm_stats, 
  
  fast_pcwde_sapnonlig_gamm_stats, 
  
  fast_pcwde_ecmperox_gamm_stats, 
  
  fast_pcwde_ecmnoperox_gamm_stats
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  )

#### Figure 2c ####
fast_pcwde_fungi_fig2c <- ggplot() + 
  
  ## CI
  geom_ribbon(
    
    data = fast_pcwde_fungi_fit, 
    
    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL), 
    
    fill = "grey80"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = fast_pcwde_fungi_input, 
    
    aes(x = raw, y = p.resid, colour = FUNCTIONAL.GROUP.LABEL), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(

    data = fast_pcwde_fungi_fit,

    aes(x = x, y = fit + se, colour = FUNCTIONAL.GROUP.LABEL),

    linetype = 2#, 
    
    #colour = "grey60"

  ) +
  
  ## Lower CI
  geom_line(

    data = fast_pcwde_fungi_fit,

    aes(x = x, y = fit - se, colour = FUNCTIONAL.GROUP.LABEL), 

    linetype = 2

  ) + 
  
  ## Line
  geom_line(
    
    data = fast_pcwde_fungi_fit, 
    
    aes(x = x, y = fit, colour = FUNCTIONAL.GROUP.LABEL)
    
  ) + 
  
  ## Color
  scale_colour_manual(
    
    values = c("#443A83FF", "#287C8EFF", "#31B57BFF", "#B9DE28FF"), 
    
    guide = "none"
    
  ) + 
  
  ## P values
  geom_text(
    
    data = fast_pcwde_fungi_gamm_stats, 
    
    aes(x = XPOS, y = YPOS, label = PVAL.LABEL), 
    
    parse = TRUE, 
    
    stat = "identity", 
    
    size = 3
    
  ) + 
  
  ## Titles
  labs(
    
    title = "(c)", 
    
    x = bquote('Fungal functional group abundance (ITS copies'%.%g^-1~'fine root litter)'), 
    
    y = "Partial PCWDEs (rapid decay)"
    
  ) + 
  
  scale_x_continuous(
    
    breaks = breaks_extended(n = 3), 
    
    limits = c(0, NA), 
    
    expand = c(0.1, 0.1), 
    
    labels = scientific_10
    
  ) +  
  
  ## Wrap
  facet_wrap(
    
    ~ FUNCTIONAL.GROUP.LABEL, 
    
    nrow = 1, 
    
    ncol = 4, 
    
    scales = "free_x"
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 10), 
    
    axis.title.y = element_text(colour = "black", size = 10),  
    
    axis.text.x = element_text(colour = "black", size = 7.5), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Assemble Figure 2  

```{r assemble_fig2}
## Assemble figure
Figure_2 <- plot_grid(
  
  combined_pcwde_functional_group_fig, 
  
  pcwde_fungi_fig2b, 
  
  fast_pcwde_fungi_fig2c, 
  
  ncol = 1, 
  
  rel_heights = c(1, 1, 1), 
  
  align = "v", 
  
  axis = "lr"
  
  )

## Save figure
ggsave2(

  "figures/MolEcol_revision/Figure_2.pdf",

  Figure_2,

  device = "pdf",

  width = 6.5,

  height = 7.5,

  units = "in"

)
```

```{r check_pcwde_fungi_gamm}
# All PCWDE
gam.check(pcwde_fungi_gamm$gam)

# Rapid PCWDE
gam.check(fast_pcwde_fungi_gamm$gam)
```

## GAMM for Figure 3a-d  

```{r gamm_decay_fungi}
#### Decay GAMM ####
decay_gamm <- gamm(
  
  PROP.MASS.LOSS ~ s(SAP.LIG.ABS, k = 3) + s(SAP.NONLIG.ABS, k = 3) + s(ECM.PEROX.ABS, k = -1) + s(ECM.NOPEROX.ABS, k = -1),
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

decay_gamm_summary <- summary(decay_gamm$gam)

decay_gamm_plot <- plot(decay_gamm$gam, residuals = TRUE)

#### Decay ~ lig saps ####
decay_saplig_fig_data <- decay_gamm_plot[[1]]

decay_saplig_fit <- as_tibble(decay_saplig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

decay_saplig_input <- as_tibble(decay_saplig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

decay_saplig_gamm_stats <- tibble(
  
  PVAL = decay_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 500000000, 
  
  YPOS = 0.1625) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs")

#### Decay ~ non-lig saps ####
decay_sapnonlig_fig_data <- decay_gamm_plot[[2]]

decay_sapnonlig_fit <- as_tibble(decay_sapnonlig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

decay_sapnonlig_input <- as_tibble(decay_sapnonlig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

decay_sapnonlig_gamm_stats <- tibble(
  
  PVAL = decay_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 850000000, 
  
  YPOS = 0.17) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs")

#### Decay ~ ECM perox ####
decay_ecmperox_fig_data <- decay_gamm_plot[[3]]

decay_ecmperox_fit <- as_tibble(decay_ecmperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

decay_ecmperox_input <- as_tibble(decay_ecmperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

decay_ecmperox_gamm_stats <- tibble(
  
  PVAL = decay_gamm_summary$s.table[3, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1750000000, 
  
  YPOS = 0.18) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases")

#### Decay ~ ECM no perox ####
decay_ecmnoperox_fig_data <- decay_gamm_plot[[4]]

decay_ecmnoperox_fit <- as_tibble(decay_ecmnoperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")

decay_ecmnoperox_input <- as_tibble(decay_ecmnoperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")

decay_ecmnoperox_gamm_stats <- tibble(
  
  PVAL = decay_gamm_summary$s.table[4, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1000000000, 
  
  YPOS = 0.165) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases")
```

## Figure 3a  

```{r fig_3a_decay_saplig}
#### Figure 3a ####
decay_saplig_fig3a <- ggplot() + 
  
  ## CI
  geom_ribbon(
    
    data = decay_saplig_fit, 
    
    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL), 
    
    fill = "grey80"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = decay_saplig_input, 
    
    aes(x = raw, y = p.resid), 
    
    colour = "#443A83FF", 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(

    data = decay_saplig_fit,

    aes(x = x, y = fit + se),

    linetype = 2, 
    
    colour = "#443A83FF"

  ) +
  
  ## Lower CI
  geom_line(

    data = decay_saplig_fit,

    aes(x = x, y = fit - se), 
    
    colour = "#443A83FF", 

    linetype = 2

  ) + 
  
  ## Line
  geom_line(
    
    data = decay_saplig_fit, 
    
    aes(x = x, y = fit), 
    
    colour = "#443A83FF"
    
  ) + 
  
  ## P values
  geom_text(
    
    data = decay_saplig_gamm_stats, 
    
    aes(x = XPOS, y = YPOS, label = PVAL.LABEL), 
    
    parse = TRUE, 
    
    stat = "identity", 
    
    size = 3
    
  ) + 
  
  ## Titles
  labs(
    
    title = "(a)", 
    
    x = bquote(atop('Ligninolytic saprotrophs', '(ITS copies'%.%g^-1~'fine root litter)')), 
    
    y = "Partial fine root decay"
    
  ) + 
  
  scale_x_continuous(
    
    breaks = breaks_extended(n = 3), 
    
    limits = c(0, NA), 
    
    expand = c(0.1, 0.1), 
    
    labels = scientific_10
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 9), 
    
    axis.title.y = element_text(colour = "black", size = 9),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Figure 3b  

```{r fig_3b_decay_sapnonlig}
#### Figure 3b ####
decay_sapnonlig_fig3b <- ggplot() + 
  
  ## CI
  geom_ribbon(
    
    data = decay_sapnonlig_fit, 
    
    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL), 
    
    fill = "grey80"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = decay_sapnonlig_input, 
    
    aes(x = raw, y = p.resid), 
    
    colour = "#287C8EFF", 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(

    data = decay_sapnonlig_fit,

    aes(x = x, y = fit + se),

    linetype = 2, 
    
    colour = "#287C8EFF"

  ) +
  
  ## Lower CI
  geom_line(

    data = decay_sapnonlig_fit,

    aes(x = x, y = fit - se), 
    
    colour = "#287C8EFF", 

    linetype = 2

  ) + 
  
  ## Line
  geom_line(
    
    data = decay_sapnonlig_fit, 
    
    aes(x = x, y = fit), 
    
    colour = "#287C8EFF"
    
  ) + 
  
  ## P values
  geom_text(

    data = decay_sapnonlig_gamm_stats,

    aes(x = XPOS, y = YPOS, label = PVAL.LABEL),

    parse = TRUE,

    stat = "identity",

    size = 3

  ) +
  
  ## Titles
  labs(
    
    title = "(b)", 
    
    x = bquote(atop('Non-ligninolytic saprotrophs', '(ITS copies'%.%g^-1~'fine root litter)')), 
    
    y = "Partial fine root decay"
    
  ) + 
  scale_x_continuous(
    
    breaks = breaks_extended(n = 3), 
    
    limits = c(0, NA), 
    
    expand = c(0.1, 0.1), 
    
    labels = scientific_10
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 9), 
    
    axis.title.y = element_text(colour = "black", size = 9),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Figure 3c  

```{r fig_3c_decay_ecmperox}
#### Figure 3c ####
decay_ecmperox_fig3c <- ggplot() + 
  
  ## CI
  geom_ribbon(
    
    data = decay_ecmperox_fit, 
    
    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL), 
    
    fill = "grey80"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = decay_ecmperox_input, 
    
    aes(x = raw, y = p.resid), 
    
    colour = "#31B57BFF", 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(

    data = decay_ecmperox_fit,

    aes(x = x, y = fit + se),

    linetype = 2, 
    
    colour = "#31B57BFF"

  ) +
  
  ## Lower CI
  geom_line(

    data = decay_ecmperox_fit,

    aes(x = x, y = fit - se), 
    
    colour = "#31B57BFF", 

    linetype = 2

  ) + 
  
  ## Line
  geom_line(
    
    data = decay_ecmperox_fit, 
    
    aes(x = x, y = fit), 
    
    colour = "#31B57BFF"
    
  ) + 
  
  ## P values
  geom_text(

    data = decay_ecmperox_gamm_stats,

    aes(x = XPOS, y = YPOS, label = PVAL.LABEL),

    parse = TRUE,

    stat = "identity",

    size = 3

  ) +
  
  ## Titles
  labs(
    
    title = "(c)", 
    
    x = bquote(atop('ECM with peroxidases', '(ITS copies'%.%g^-1~'fine root litter)')), 
    
    y = "Partial fine root decay"
    
  ) + 
  
  scale_x_continuous(
    
    breaks = breaks_extended(n = 3), 
    
    limits = c(0, NA), 
    
    expand = c(0.1, 0.1), 
    
    labels = scientific_10
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 9), 
    
    axis.title.y = element_text(colour = "black", size = 9),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Figure 3d  

```{r fig_3d_decay_ecmnoperox}
#### Figure 3d ####
decay_ecmnoperox_fig3d <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = decay_ecmnoperox_input, 
    
    aes(x = raw, y = p.resid), 
    
    colour = "#31B57BFF", 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## P values
  geom_text(

    data = decay_ecmnoperox_gamm_stats,

    aes(x = XPOS, y = YPOS, label = PVAL.LABEL),

    parse = TRUE,

    stat = "identity",

    size = 3

  ) + 
  
  ## Titles
  labs(
    
    title = "(d)", 
    
    x = bquote(atop('ECM without peroxidases', '(ITS copies'%.%g^-1~'fine root litter)')), 
    
    y = "Partial fine root decay"
    
  ) + 
  
  scale_x_continuous(
    
    breaks = breaks_extended(n = 3), 
    
    limits = c(0, NA), 
    
    expand = c(0.1, 0.1), 
    
    labels = scientific_10
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 9), 
    
    axis.title.y = element_text(colour = "black", size = 9),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Decay - environment GAMM  

```{r decay_env_gamm}
#### Decay env GAMM ####
decay_env_gamm <- gamm(
  
  PROP.MASS.LOSS ~ s(PC1, k = -1) + s(PC2, k = -1),
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

decay_env_gamm_summary <- summary(decay_env_gamm$gam)

decay_env_gamm_plot <- plot(decay_env_gamm$gam, residuals = TRUE)

#### Decay ~ PC1 ####
decay_pc1_fig_data <- decay_env_gamm_plot[[1]]

decay_pc1_fit <- as_tibble(decay_pc1_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(ENV.LABEL = "PC1")

decay_pc1_input <- as_tibble(decay_pc1_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(ENV.LABEL = "PC1")

decay_pc1_gamm_stats <- tibble(
  
  PVAL = decay_env_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = -0.5, 
  
  YPOS = 0.15) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(ENV.LABEL = "PC1")

#### Decay ~ PC2 ####
decay_pc2_fig_data <- decay_env_gamm_plot[[2]]

decay_pc2_fit <- as_tibble(decay_pc2_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(ENV.LABEL = "PC2")

decay_pc2_input <- as_tibble(decay_pc2_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(ENV.LABEL = "PC2")

decay_pc2_gamm_stats <- tibble(
  
  PVAL = decay_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.9, 
  
  YPOS = 0.17) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(ENV.LABEL = "PC2")
```

## Figure 3e  

```{r fig3e_decay_pc1}
#### Figure 3e ####
decay_pc1_fig3e <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = decay_pc1_input, 
    
    aes(x = raw, y = p.resid), 
    
    colour = "#501D4CFF", 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## P values
  geom_text(
    
    data = decay_pc1_gamm_stats, 
    
    aes(x = XPOS, y = YPOS, label = PVAL.LABEL), 
    
    parse = TRUE, 
    
    stat = "identity", 
    
    size = 3
    
  ) + 
  
  ## Titles
  labs(
    
    title = "(e)", 
    
    x = bquote(atop("Environmental gradient 1", "(PC1)")), 
    
    y = "Partial fine root decay"
    
  ) + 
  
  scale_y_continuous(
    
    breaks = c(-0.1, 0, 0.1)
    
  ) +
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 9), 
    
    axis.title.y = element_text(colour = "black", size = 9),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Figure 3f  

```{r fig3f_decay_pc2}
#### Figure 3f ####
decay_pc2_fig3f <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = decay_pc2_input, 
    
    aes(x = raw, y = p.resid), 
    
    colour = "#EC4B3EFF", 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## P values
  geom_text(
    
    data = decay_pc2_gamm_stats, 
    
    aes(x = XPOS, y = YPOS, label = PVAL.LABEL), 
    
    parse = TRUE, 
    
    stat = "identity", 
    
    size = 3
    
  ) + 
  
  ## Titles
  labs(
    
    title = "(f)", 
    
    x = bquote(atop("Environmental gradient 2", "(PC2)")), 
    
    y = "Partial fine root decay"
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 9), 
    
    axis.title.y = element_text(colour = "black", size = 9),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )
```

## Assemble Figure 3  

```{r assemble_fig3}
## Assemble figure
Figure_3 <- plot_grid(
  
  decay_saplig_fig3a, 
  
  decay_sapnonlig_fig3b, 
  
  decay_ecmperox_fig3c, 
  
  decay_ecmnoperox_fig3d, 
  
  decay_pc1_fig3e, 
  
  decay_pc2_fig3f, 
  
  ncol = 3, 
  
  nrow = 2, 
  
  rel_heights = c(1, 1), 
  
  axis = "lr", 
  
  align = "hv"
  
  )

## Save figure
ggsave2(

  "figures/MolEcol_revision/Figure_3.pdf",

  Figure_3,

  device = "pdf",

  width = 6.5,

  height = 5,

  units = "in"

)
```

```{r decay_fungi_env_gamm}
decay_fungi_env_gamm <- gamm(
  
  PROP.MASS.LOSS ~ s(SAP.LIG.ABS, k = 3) + s(SAP.NONLIG.ABS, k = 3) + s(ECM.PEROX.ABS, k = -1) + s(ECM.NOPEROX.ABS, k = -1) + s(PC1, k = -1) + s(PC2, k = -1),
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

decay_fungi_env_gamm_summary <- summary(decay_fungi_env_gamm$gam)
```

```{r check_decay_gamm}
gam.check(decay_gamm$gam)

gam.check(decay_env_gamm$gam)

gam.check(decay_fungi_env_gamm$gam)
```

## Figure 4  

```{r fig4_fungi_env_gamm}
#### Lig saps ~ env GAMM ####
saplig_env_gamm <- gamm(
  
  SAP.LIG.ABS ~ s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

saplig_env_gamm_summary <- summary(saplig_env_gamm$gam)

saplig_env_gamm_plot <- plot(saplig_env_gamm$gam, residuals = TRUE)

#### Non-lig saps ~ env GAMM ####
sapnonlig_env_gamm <- gamm(
  
  SAP.NONLIG.ABS ~ s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

sapnonlig_env_gamm_summary <- summary(sapnonlig_env_gamm$gam)

sapnonlig_env_gamm_plot <- plot(sapnonlig_env_gamm$gam, residuals = TRUE)

#### ECM perox ~ env GAMM ####
ecmperox_env_gamm <- gamm(
  
  ECM.PEROX.ABS ~ s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

ecmperox_env_gamm_summary <- summary(ecmperox_env_gamm$gam)

ecmperox_env_gamm_plot <- plot(ecmperox_env_gamm$gam, residuals = TRUE)

#### ECM no perox ~ env GAMM ####
ecmnoperox_env_gamm <- gamm(
  
  ECM.NOPEROX.ABS ~ s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

ecmnoperox_env_gamm_summary <- summary(ecmnoperox_env_gamm$gam)

ecmnoperox_env_gamm_plot <- plot(ecmnoperox_env_gamm$gam, residuals = TRUE)
```

```{r fig4}
#### Lig saps ~ PC1 input data ####
saplig_pc1_fig_data <- saplig_env_gamm_plot[[1]]

saplig_pc1_fit <- as_tibble(saplig_pc1_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

saplig_pc1_input <- as_tibble(saplig_pc1_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

saplig_pc1_gamm_stats <- tibble(
  
  PVAL = saplig_env_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.85, 
  
  YPOS = 1900000000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

#### Lig saps ~ PC1 input data ####
saplig_pc2_fig_data <- saplig_env_gamm_plot[[2]]

saplig_pc2_fit <- as_tibble(saplig_pc2_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

saplig_pc2_input <- as_tibble(saplig_pc2_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

saplig_pc2_gamm_stats <- tibble(
  
  PVAL = saplig_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.85, 
  
  YPOS = 1600000000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

#### Non-lig saps ~ PC1 input data ####
sapnonlig_pc1_fig_data <- sapnonlig_env_gamm_plot[[1]]

sapnonlig_pc1_fit <- as_tibble(sapnonlig_pc1_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

sapnonlig_pc1_input <- as_tibble(sapnonlig_pc1_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

sapnonlig_pc1_gamm_stats <- tibble(
  
  PVAL = sapnonlig_env_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.85, 
  
  YPOS = 1900000000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

#### Non-lig saps ~ PC2 input data ####
sapnonlig_pc2_fig_data <- sapnonlig_env_gamm_plot[[2]]

sapnonlig_pc2_fit <- as_tibble(sapnonlig_pc2_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

sapnonlig_pc2_input <- as_tibble(sapnonlig_pc2_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

sapnonlig_pc2_gamm_stats <- tibble(
  
  PVAL = sapnonlig_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.85, 
  
  YPOS = 1600000000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic saps.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

#### ECM perox ~ PC1 input data ####
ecmperox_pc1_fig_data <- ecmperox_env_gamm_plot[[1]]

ecmperox_pc1_fit <- as_tibble(ecmperox_pc1_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM with perox.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

ecmperox_pc1_input <- as_tibble(ecmperox_pc1_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM with perox.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

ecmperox_pc1_gamm_stats <- tibble(
  
  PVAL = ecmperox_env_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.85, 
  
  YPOS = 1900000000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM with perox.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

#### ECM perox ~ PC2 input data ####
ecmperox_pc2_fig_data <- ecmperox_env_gamm_plot[[2]]

ecmperox_pc2_fit <- as_tibble(ecmperox_pc2_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM with perox.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

ecmperox_pc2_input <- as_tibble(ecmperox_pc2_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM with perox.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

ecmperox_pc2_gamm_stats <- tibble(
  
  PVAL = ecmperox_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.85, 
  
  YPOS = 1600000000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM with perox.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

#### ECM no perox ~ PC1 input data ####
ecmnoperox_pc1_fig_data <- ecmnoperox_env_gamm_plot[[1]]

ecmnoperox_pc1_fit <- as_tibble(ecmnoperox_pc1_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM without perox.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

ecmnoperox_pc1_input <- as_tibble(ecmnoperox_pc1_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM without perox.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

ecmnoperox_pc1_gamm_stats <- tibble(
  
  PVAL = ecmnoperox_env_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.85, 
  
  YPOS = 1900000000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM without perox.", 
    
    ENV.LABEL = "Environ. PC1"
    
  )

#### ECM no perox ~ PC2 input data ####
ecmnoperox_pc2_fig_data <- ecmnoperox_env_gamm_plot[[2]]

ecmnoperox_pc2_fit <- as_tibble(ecmnoperox_pc2_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM without perox.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

ecmnoperox_pc2_input <- as_tibble(ecmnoperox_pc2_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM without perox.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

ecmnoperox_pc2_gamm_stats <- tibble(
  
  PVAL = ecmnoperox_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.85, 
  
  YPOS = 1600000000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = "ECM without perox.", 
    
    ENV.LABEL = "Environ. PC2"
    
  )

#### Combine input data ####
fungi_env_fit <- bind_rows(
  
  saplig_pc1_fit, 
  
  sapnonlig_pc1_fit, 
  
  ecmperox_pc1_fit, 
  
  ecmnoperox_pc1_fit, 
  
  saplig_pc2_fit, 
  
  sapnonlig_pc2_fit, 
  
  ecmperox_pc2_fit, 
  
  ecmnoperox_pc2_fit
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic saps.", 
        
        "Non-ligninolytic saps.", 
        
        "ECM with perox.", 
        
        "ECM without perox."
        
      )
      
    ), 
    
    ENV.LABEL = factor(ENV.LABEL, levels = c("Environ. PC1", "Environ. PC2"))
    
  ) %>% 
  
  mutate(
    
    fit = ifelse(FUNCTIONAL.GROUP.LABEL == "Non-ligninolytic saps." & ENV.LABEL == "Environ. PC2", NA, fit), 
    
    se = ifelse(FUNCTIONAL.GROUP.LABEL == "Non-ligninolytic saps." & ENV.LABEL == "Environ. PC2", NA, se), 
    
    fit = ifelse(FUNCTIONAL.GROUP.LABEL == "ECM without perox." & ENV.LABEL == "Environ. PC2", NA, fit), 
    
    se = ifelse(FUNCTIONAL.GROUP.LABEL == "ECM without perox." & ENV.LABEL == "Environ. PC2", NA, se)
    
  )

fungi_env_input <- bind_rows(
  
  saplig_pc1_input, 
  
  sapnonlig_pc1_input, 
  
  ecmperox_pc1_input, 
  
  ecmnoperox_pc1_input, 
  
  saplig_pc2_input, 
  
  sapnonlig_pc2_input, 
  
  ecmperox_pc2_input, 
  
  ecmnoperox_pc2_input
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic saps.", 
        
        "Non-ligninolytic saps.", 
        
        "ECM with perox.", 
        
        "ECM without perox."
        
      )
      
    ), 
    
    ENV.LABEL = factor(ENV.LABEL, levels = c("Environ. PC1", "Environ. PC2"))
    
  )

fungi_env_gamm_stats <- bind_rows(
  
  saplig_pc1_gamm_stats, 
  
  sapnonlig_pc1_gamm_stats, 
  
  ecmperox_pc1_gamm_stats, 
  
  ecmnoperox_pc1_gamm_stats, 
  
  saplig_pc2_gamm_stats, 
  
  sapnonlig_pc2_gamm_stats, 
  
  ecmperox_pc2_gamm_stats, 
  
  ecmnoperox_pc2_gamm_stats
  
) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic saps.", 
        
        "Non-ligninolytic saps.", 
        
        "ECM with perox.", 
        
        "ECM without perox."
        
      )
      
    ), 
    
    ENV.LABEL = factor(ENV.LABEL, levels = c("Environ. PC1", "Environ. PC2"))
    
  )

#### Assemble figure ####
fungi_env_fig4 <- ggplot() + 
  
  ## CI
  geom_ribbon(

    data = fungi_env_fit,

    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL),

    fill = "grey80"

  ) +
  
  ## Points
  geom_point(
    
    data = fungi_env_input, 
    
    aes(x = raw, y = p.resid, colour = FUNCTIONAL.GROUP.LABEL), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(

    data = fungi_env_fit,

    aes(x = x, y = fit + se, colour = FUNCTIONAL.GROUP.LABEL),

    linetype = 2

  ) +
  
  ## Lower CI
  geom_line(

    data = fungi_env_fit,

    aes(x = x, y = fit - se, colour = FUNCTIONAL.GROUP.LABEL), 

    linetype = 2

  ) + 
  
  ## Line
  geom_line(
    
    data = fungi_env_fit, 
    
    aes(x = x, y = fit, colour = FUNCTIONAL.GROUP.LABEL)
    
  ) + 
  
  ## Color
  scale_colour_manual(
    
    values = c("#443A83FF", "#287C8EFF", "#31B57BFF", "#B9DE28FF"), 
    
    guide = "none"
    
  ) + 
  
  ## P values
  geom_text(

    data = fungi_env_gamm_stats,

    aes(x = XPOS, y = YPOS, label = PVAL.LABEL),

    parse = TRUE,

    stat = "identity",

    size = 3

  ) +
  
  ## Titles
  labs(
    
    title = NULL, 
    
    x = "Environmental gradient (PCA axis)", 
    
    y = "Partial fungal abundance"
    
  ) + 
  
  scale_y_continuous(
    
    labels = scientific_10
    
  ) + 
  
  ## Wrap
  facet_grid(
    
    cols = vars(FUNCTIONAL.GROUP.LABEL), 
    
    rows = vars(ENV.LABEL), 
    
    scales = "free"
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 10), 
    
    axis.title.y = element_text(colour = "black", size = 10),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 9)
    
  )

## Save figure
ggsave2(

  "figures/MolEcol_revision/Figure_4.pdf",

  fungi_env_fig4,

  device = "pdf",

  width = 6.5,

  height = 3.25,

  units = "in"

)
```

```{r check_fungi_env_gamm}
gam.check(saplig_env_gamm$gam)

gam.check(sapnonlig_env_gamm$gam)

gam.check(ecmperox_env_gamm$gam)

gam.check(ecmnoperox_env_gamm$gam)
```

```{r figs1_map}
library(usmap)
library(rgdal)
library(ggrepel)

#### 4. Map of study sites ####

stand.locations.df <- read_tsv("data/site/plot.locations.updated.2019.10.19.and.20.V2.txt") %>%
  dplyr::select(Site, Plot, `Bag angle`, `Bag distance (m)`, `Site latitude degrees (N)`, `Site latitude minutes (N)`, `Site longitude degrees (W)`, `Site longitude minutes (W)`) %>%
  dplyr::rename(STAND = "Site",
                PLOT = "Plot",
                PLOT.BEARING = "Bag angle",
                PLOT.DIST = "Bag distance (m)",
                STAND.LAT.DEG = "Site latitude degrees (N)",
                STAND.LAT.MIN = "Site latitude minutes (N)",
                STAND.LON.DEG = "Site longitude degrees (W)",
                STAND.LON.MIN = "Site longitude minutes (W)") %>%
  dplyr::filter(PLOT <= 72) %>%
  mutate(STAND = paste("Stand", STAND, sep = "_")) %>%
  # Decimal degrees
  mutate(STAND.LAT = STAND.LAT.DEG + (STAND.LAT.MIN / 60),
         STAND.LON = -1 * (STAND.LON.DEG + (STAND.LON.MIN / 60))) %>%
  distinct(STAND, .keep_all = TRUE) %>%
  select(STAND, STAND.LON, STAND.LAT) %>%
  as.data.frame(.)

row.names(stand.locations.df) <- stand.locations.df$STAND
stand.locations.df <- stand.locations.df[2 : ncol(stand.locations.df)]

stand.locations.data <- usmap_transform(stand.locations.df)
stand.locations.data$STAND <- row.names(stand.locations.df)

mi.map.plot <- plot_usmap(include = "MI") +
  geom_point(data = stand.locations.data,
             aes(x = STAND.LON.1, y = STAND.LAT.1),
             size = 1) +

  geom_text_repel(data = stand.locations.data,
            aes(x = STAND.LON.1, y = STAND.LAT.1, label = STAND),
            size = 2,
            colour = "blue",
            max.overlaps = 100)

# ggsave("figures/MolEcol_revision/Figure_S1.pdf", mi.map.plot, device = "pdf", width = 5, height = 5, units = "in")
```

```{r figs2_env}
#### Table for figure ####
figs2_env_input <- env_fungi_data %>% 
  
  group_by(SITE) %>% 
  
  mutate(MEAN.NMIN = mean(N.MIN)) %>% 
  
  ungroup() %>% 
  
  arrange(N.MIN) %>% 
  
  mutate(
    
    SITE.LABEL = gsub("\\_", " ", SITE), 
    
    SITE.LABEL = factor(SITE.LABEL, levels = unique(SITE.LABEL))
    
  )

#### Figure s2a ####
env_figs2a <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = figs2_env_input, 
    
    aes(x = SITE.LABEL, y = N.MIN, colour = SITE.LABEL), 
    
    position = position_jitter(width = 0.1), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Boxplots
  geom_boxplot(
    
    data = figs2_env_input, 
    
    aes(x = SITE.LABEL, y = N.MIN, colour = SITE.LABEL), 
    
    fill = NA, 
    
    outlier.shape = NA
    
  ) + 
  
  ## Scale error bar colors
  scale_colour_viridis(
    
    discrete = TRUE, 
    
    option = "plasma", 
    
    guide = "none"
    
  ) + 
  
  ## Add plot and axis titles
  labs(
    
    title = "(a)", 
    
    x = NULL, 
    
    y = bquote('Inorganic N availability ('*mu*g~N%.%g^-1%.%d^-1*')')
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(size = 12, hjust = 0), 
    
    axis.title.x = element_text(size = 8), 
    
    axis.title.y = element_text(size = 8), 
    
    axis.text.x = element_text(size = 8, angle = 45, hjust = 1), 
    
    axis.text.y = element_text(size = 8)
    
  )

#### Figure s2b ####
env_figs2b <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = figs2_env_input, 
    
    aes(x = SITE.LABEL, y = VWC.MEAN, colour = SITE.LABEL), 
    
    position = position_jitter(width = 0.1), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Boxplots
  geom_boxplot(
    
    data = figs2_env_input, 
    
    aes(x = SITE.LABEL, y = VWC.MEAN, colour = SITE.LABEL), 
    
    fill = NA, 
    
    outlier.shape = NA
    
  ) + 
  
  ## Scale error bar colors
  scale_colour_viridis(
    
    discrete = TRUE, 
    
    option = "plasma", 
    
    guide = "none"
    
  ) + 
  
  ## Add plot and axis titles
  labs(
    
    title = "(b)", 
    
    x = NULL, 
    
    y = bquote(theta~'('*m^-3%.%m^-3*')')
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(size = 12, hjust = 0), 
    
    axis.title.x = element_text(size = 8), 
    
    axis.title.y = element_text(size = 8), 
    
    axis.text.x = element_text(size = 8, angle = 45, hjust = 1), 
    
    axis.text.y = element_text(size = 8)
    
  )

#### Figure s2c ####
env_figs2c <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = figs2_env_input, 
    
    aes(x = SITE.LABEL, y = TEMP.MEAN, colour = SITE.LABEL), 
    
    position = position_jitter(width = 0.1), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Boxplots
  geom_boxplot(
    
    data = figs2_env_input, 
    
    aes(x = SITE.LABEL, y = TEMP.MEAN, colour = SITE.LABEL), 
    
    fill = NA, 
    
    outlier.shape = NA
    
  ) + 
  
  ## Scale error bar colors
  scale_colour_viridis(
    
    discrete = TRUE, 
    
    option = "plasma", 
    
    guide = "none"
    
  ) + 
  
  ## Add plot and axis titles
  labs(
    
    title = "(c)", 
    
    x = NULL, 
    
    y = bquote('Soil temperature ('*degree*C*')')
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(size = 12, hjust = 0), 
    
    axis.title.x = element_text(size = 8), 
    
    axis.title.y = element_text(size = 8), 
    
    axis.text.x = element_text(size = 8, angle = 45, hjust = 1), 
    
    axis.text.y = element_text(size = 8)
    
  )

#### Figure s2d ####
env_figs2d <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = figs2_env_input, 
    
    aes(x = SITE.LABEL, y = SOIL.PH, colour = SITE.LABEL), 
    
    position = position_jitter(width = 0.1), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Boxplots
  geom_boxplot(
    
    data = figs2_env_input, 
    
    aes(x = SITE.LABEL, y = SOIL.PH, colour = SITE.LABEL), 
    
    fill = NA, 
    
    outlier.shape = NA
    
  ) + 
  
  ## Scale error bar colors
  scale_colour_viridis(
    
    discrete = TRUE, 
    
    option = "plasma", 
    
    guide = "none"
    
  ) + 
  
  ## Add plot and axis titles
  labs(
    
    title = "(d)", 
    
    x = NULL, 
    
    y = "Soil pH"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(size = 12, hjust = 0), 
    
    axis.title.x = element_text(size = 8), 
    
    axis.title.y = element_text(size = 8), 
    
    axis.text.x = element_text(size = 8, angle = 45, hjust = 1), 
    
    axis.text.y = element_text(size = 8)
    
  )

#### Assemble figure ####
Figure_S2 <- plot_grid(
  
  env_figs2a, 
  
  env_figs2b, 
  
  env_figs2c, 
  
  env_figs2d, 
  
  ncol = 2, 
  
  nrow = 2, 
  
  rel_heights = c(1, 1, 1, 1), 
  
  rel_widths = c(1, 1, 1, 1), 
  
  align = "hv"
  
  )

## Save figure
ggsave2(

  "figures/MolEcol_revision/Figure_S2.pdf",

  Figure_S2,

  device = "pdf",

  width = 6.5,

  height = 6.5,

  units = "in"

)
```

```{r figs4_n_covariation}
# Read in
Nmin.aug.2018.mean.tbl <- read_csv("data/root_and_soil_chem/complete.min.august.manistee.csv") %>% 
  rename(TREE = "Tree ID", 
         INCUBATION = "Time", 
         REP = "Rep", 
         NITRATE = "Nitrate", 
         AMMONIUM = "Ammonia") %>% 
  select(INCUBATION, TREE, REP, NITRATE, AMMONIUM) %>% 
  mutate(NITRATE = ifelse(NITRATE < 0, 0, NITRATE), 
         AMMONIUM = ifelse(AMMONIUM < 0, 0, AMMONIUM)) %>% 
  group_by(INCUBATION, TREE) %>% 
  summarise(NITRATE = sum(NITRATE), 
            AMMONIUM = sum(AMMONIUM)) %>% 
  ungroup() %>% 
  mutate(TOTAL = NITRATE + AMMONIUM) %>% 
  select(INCUBATION, TREE, TOTAL) %>% 
  pivot_wider(id_cols = TREE, 
              names_from = "INCUBATION", 
              values_from = "TOTAL") %>% 
  mutate(N.MIN = (Post - Pre) / 14) %>% 
  select(TREE, N.MIN) %>% 
  separate(TREE, into = c("STAND", "TREE"), sep = "_") %>% 
  mutate(STAND = paste("Stand", STAND, sep = "_")) %>% 
  filter(!is.na(N.MIN)) %>% 
  group_by(STAND) %>% 
  summarise(NMIN.MEAN = mean(N.MIN), 
            NMIN.SE = se(N.MIN)) %>% 
  ungroup()

# Root CN
root.CN.tbl <- read_csv(file = "data/root_and_soil_chem/Zak_WA_pre_roots_3-26-19_1.csv") %>% 
  filter(!is.na(Sample) & Sample != "Blank" & Sample != "Standard") %>% 
  separate(Sample, into = c("STAND", "REP1", "REP2"), sep = "_") %>% 
  mutate(STAND = paste("Stand", STAND, sep = "_")) %>% 
  group_by(STAND, REP1) %>% 
  summarise(NITROGEN = mean(Nitrogen), 
            CARBON = mean(Carbon)) %>% 
  ungroup() %>% 
  mutate(ROOT.CN = CARBON / NITROGEN) %>% 
  group_by(STAND) %>% 
  summarise(ROOTCN.MEAN = mean(ROOT.CN), 
            ROOTCN.SE = se(ROOT.CN)) %>% 
  ungroup() %>% 
  inner_join(Nmin.aug.2018.mean.tbl, ., by = "STAND")

# LM
root.CN.lm <- lm(ROOTCN.MEAN ~ NMIN.MEAN, data = root.CN.tbl)
root.CN.lm.summary <- summary(root.CN.lm)
root.CN.lm.Pval = format(round(root.CN.lm.summary$coefficients[2, 4], 3), scientific = FALSE)
root.CN.lm.P <- tibble(PVAL.LABEL = paste("atop(R[adj.]^2 == ", 
                                          round(root.CN.lm.summary$adj.r.squared, 3), 
                                          ", ", 
                                          "italic(P) == ", 
                                          root.CN.lm.Pval, 
                                          ")", 
                                          sep = ""), 
                    XPOS = 0.65, 
                    YPOS = 55)

# Figure
root.CN.plot <- ggplot() + 
  
  # CI
  geom_smooth(data = root.CN.tbl, 
              aes(x = NMIN.MEAN, y = ROOTCN.MEAN), 
              formula = y ~ x,  
              method = "lm", 
              colour = "grey80", 
              fill = "grey80") + 
  
  # Vertical error bars
  geom_errorbar(data = root.CN.tbl, 
                aes(x = NMIN.MEAN, y = ROOTCN.MEAN, ymin = ROOTCN.MEAN - ROOTCN.SE, ymax = ROOTCN.MEAN + ROOTCN.SE), 
                colour = "#7B02A8FF") + 
  
  # Horizontal error bars
  geom_errorbar(data = root.CN.tbl, 
                aes(x = NMIN.MEAN, y = ROOTCN.MEAN, xmin = NMIN.MEAN - NMIN.SE, xmax = NMIN.MEAN + NMIN.SE), 
                colour = "#7B02A8FF") + 
  
  # Points
  geom_point(data = root.CN.tbl, 
             aes(x = NMIN.MEAN, y = ROOTCN.MEAN), 
             colour = "#7B02A8FF", 
             size = 1.5) + 
  
  # Pvalues
  geom_text(data = root.CN.lm.P, 
            aes(x = XPOS, y = YPOS, label = PVAL.LABEL), 
            parse = TRUE, 
            stat = "identity", 
            size = 3) + 
  
  # Line
  geom_smooth(data = root.CN.tbl, 
              aes(x = NMIN.MEAN, y = ROOTCN.MEAN), 
              formula = y ~ x,  
              method = "lm", 
              colour = "#7B02A8FF", 
              se = FALSE) + 
  
  # Titles
  labs(title = "(a)", 
       x = bquote('Inorganic N ('*mu*g~N%.%g^-1%.%d^-1*')'), 
       y = "Fine root C/N") + 
  
  # Format
  theme(panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
        panel.background = element_rect(fill = NA), 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        plot.title = element_text(size = 12, hjust = 0), 
        axis.title.x = element_text(colour = "black", size = 7), 
        axis.title.y = element_text(colour = "black", size = 7), 
        axis.ticks = element_line(colour = "black", size = 0.25), 
        # x axis text
        axis.text.x = element_text(colour = "black", size = 6), 
        axis.text.y = element_text(colour = "black", size = 6))

# GAM
soil.CN.gam <- gam(SOIL.CN ~ s(N.MIN), data = env_fungi_data)
soil.CN.gam.summary <- summary(soil.CN.gam)
soil.CN.gam.P <- tibble(PVAL.LABEL = paste("atop(R[adj.]^2 == ", 
                                          round(soil.CN.gam.summary$r.sq, 3), 
                                          ", ", 
                                          "italic(P) < 0.001", 
                                          ")", 
                                          sep = ""), 
                    XPOS = 0.85, 
                    YPOS = 28)

# Figure
soil.CN.plot <- ggplot() + 
  
   # Line
  geom_smooth(data = env_fungi_data, 
              aes(x = N.MIN, y = SOIL.CN), 
              formula = y ~ s(x),  
              method = "gam", 
              colour = "grey80", 
              fill = "grey80") + 
  
  # Points
  geom_point(data = env_fungi_data, 
             aes(x = N.MIN, y = SOIL.CN), 
             colour = "#FEBE2AFF", 
             size = 1.5) + 
  
  # Pvalues
  geom_text(data = soil.CN.gam.P, 
            aes(x = XPOS, y = YPOS, label = PVAL.LABEL), 
            parse = TRUE, 
            stat = "identity", 
            size = 3) + 
  
  # Line
  geom_smooth(data = env_fungi_data, 
              aes(x = N.MIN, y = SOIL.CN), 
              formula = y ~ s(x),  
              method = "gam", 
              colour = "#FEBE2AFF", 
              fill = NA) + 
  
  # Titles
  labs(title = "(b)", 
       x = bquote('Inorganic N ('*mu*g~N%.%g^-1%.%d^-1*')'), 
       y = "Soil C/N") + 
  
  # Format
  theme(panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
        panel.background = element_rect(fill = NA), 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        plot.title = element_text(size = 12, hjust = 0), 
        axis.title.x = element_text(colour = "black", size = 7), 
        axis.title.y = element_text(colour = "black", size = 7),  
        axis.ticks = element_line(colour = "black", size = 0.25), 
        # x axis text
        axis.text.x = element_text(colour = "black", size = 6), 
        axis.text.y = element_text(colour = "black", size = 6))

#### Assemble figure ####
N_fig <- ggarrange(
  
  root.CN.plot, 
  
  soil.CN.plot, 
  
  nrow = 1, 
  
  ncol = 2, 
  
  align = "hv"
  
)

#### Save figure ####
ggsave2(

  "figures/MolEcol_revision/Figure_S4.pdf",

  N_fig,

  device = "pdf",

  width = 6.5,

  height = 3.25,

  units = "in"

)
```

```{r figs5_root_decay}
## Table for figure
root_decomp_fig_input <- env_fungi_data %>% 
  
  group_by(SITE) %>% 
  
  mutate(MEAN.DECOMP = mean(PROP.MASS.LOSS)) %>% 
  
  ungroup() %>% 
  
  arrange(MEAN.DECOMP) %>% 
  
  mutate(
    
    SITE.LABEL = gsub("\\_", " ", SITE), 
    
    SITE.LABEL = factor(SITE.LABEL, levels = c("Site 20", "Site 50", "Site 3", "Site 31", "Site 58", "Site 9", "Site 6", "Site 7", "Site 41", "Site 24", "Site 100", "Site 22")))

## Figure
root_decomp_fig <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = root_decomp_fig_input, 
    
    aes(x = SITE.LABEL, y = PROP.MASS.LOSS, colour = SITE.LABEL), 
    
    position = position_jitter(width = 0.1), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  geom_boxplot(
    
    data = root_decomp_fig_input, 
    
    aes(x = SITE.LABEL, y = PROP.MASS.LOSS, colour = SITE.LABEL), 
    
    fill = NA, 
    
    outlier.shape = NA
    
  ) + 
  
  ## Scale error bar colors
  scale_colour_viridis(
    
    discrete = TRUE, 
    
    option = "plasma", 
    
    guide = "none"
    
  ) + 
  
  ## Scale y axis
  scale_y_continuous(limits = c(0, NA)) + 
  
  ## Add plot and axis titles
  labs(
    
    title = NULL, 
    
    x = NULL, 
    
    y = "Proportional mass loss"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank()
    
  ) + 
  
  ## Format text
  theme(
    
    axis.title.x = element_text(size = 8), 
    
    axis.title.y = element_text(size = 8), 
    
    axis.text.x = element_text(size = 8), 
    
    axis.text.y = element_text(size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  )

#### Save figure ####
ggsave(

  "figures/MolEcol_revision/Figure_S5.pdf",

  root_decomp_fig,

  device = "pdf",

  width = 6.5,

  height = 4.5,

  units = "in"

)
```

```{r figs7_functional_group_summary}
#### Format data ####

## Get total seqs
total_seqs <- sum(genus_all$COUNT)

## Data for figure
functional_group_summary_input <- genus_all %>% 
  
  ## Add functional groups
  inner_join(., classified_genera, by = "GENUS") %>% 
  
  group_by(FUNCTIONAL.GROUP) %>% 
  
  summarise(TOTAL.COUNT = sum(COUNT)) %>% 
  
  ungroup() %>% 
  
  mutate(
    
    TOTAL.PERC = TOTAL.COUNT / total_seqs, 
    
    PERC.OF.ASSIGNED = TOTAL.COUNT / sum(TOTAL.COUNT)
    
  ) %>% 
  
  arrange(desc(TOTAL.PERC)) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = sapply(FUNCTIONAL.GROUP, functional_group_labels, simplify = TRUE), 
    
    FUNCTIONAL.GROUP.LABEL = factor(FUNCTIONAL.GROUP.LABEL, levels = FUNCTIONAL.GROUP.LABEL), 
    
    PERC.LABEL = format(round(100 * TOTAL.PERC, 1), nsmall = 1)
    
  )

perc_of_total_seqs <- 100 * (sum(functional_group_summary_input$TOTAL.COUNT) / total_seqs)

perc_of_assigned_seqs <- 100 * sum(functional_group_summary_input$PERC.OF.ASSIGNED[1 : 4])

#### Fig. S7, functional group summary ####
figs7_functional_group_summary <- ggplot() + 
  
  ## Bars
  geom_bar(
    
    data = functional_group_summary_input, 
    
    aes(
      
      x = FUNCTIONAL.GROUP.LABEL, 
      
      y = 100 * TOTAL.PERC, 
      
      fill = FUNCTIONAL.GROUP.LABEL
      
    ), 
    
    stat = "identity", 
    
    colour = "black"
    
  ) + 
  
  ## Scale fill
  scale_fill_viridis(
    
    discrete = TRUE, 
    
    option = "plasma", 
    
    guide = "none"
    
  ) + 
  
  ## Bar labels
  geom_text(
    
    data = functional_group_summary_input, 
    
    aes(x = FUNCTIONAL.GROUP.LABEL, y = (100 * TOTAL.PERC) + 3, label = PERC.LABEL)
    
  ) + 
  
  ## Axes
  scale_y_continuous(
    
    limits = c(0, 36), 
    
    expand = expansion(c(0, 0.05))
    
  ) + 
  
  ## Labels
  labs(
    
    title = "Fungal functional groups", 
    
    x = NULL, 
    
    y = "% of sequences"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank()
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0.5), 
    
    axis.title.y = element_text(colour = "black", size = 8), 
    
    axis.title.x = element_text(colour = "black", size = 8), 
    
    axis.text.x = element_text(colour = "black", size = 8, angle = 45, hjust = 1), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  )

#### Save figure ####
ggsave(

  "figures/MolEcol_revision/Figure_S7.pdf",

  figs7_functional_group_summary,

  device = "pdf",

  width = 6.5,

  height = 4.5,

  units = "in"

)
```

```{r figs8_functional_group_genus_summary}
##### Get figure data ####

## Overall
functional_group_genus_summary <- genus_relabund %>% 
  
  select(GENUS) %>% 
  
  inner_join(., classified_genera, by = "GENUS") %>% 
  
  inner_join(genus_all, ., by = "GENUS") %>%  
  
  filter(FUNCTIONAL.GROUP != "OTHER" & FUNCTIONAL.GROUP != "OTHER.MYCORRHIZA") %>% 
  
  group_by(FUNCTIONAL.GROUP, GENUS) %>% 
  
  summarise(GENUS.TOTAL = sum(COUNT)) %>% 
  
  ungroup() %>% 
  
  group_by(FUNCTIONAL.GROUP) %>% 
  
  mutate(GENUS.PERC = 100 * (GENUS.TOTAL / sum(GENUS.TOTAL))) %>% 
  
  ungroup() %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = sapply(FUNCTIONAL.GROUP, functional_group_labels, simplify = TRUE), 
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c("Lig.\nsaps.", "Non-lig.\nsaps.", "ECM\nperox.", "ECM no\nperox.")
      
    )
    
  ) %>% 
  
  arrange(FUNCTIONAL.GROUP.LABEL, desc(GENUS.PERC)) %>% 
  
  mutate(
    
    GENUS.LABEL = GENUS, 
    
    GENUS.LABEL = factor(GENUS.LABEL, levels = GENUS.LABEL)
    
  )

## Ligninolytic saprotrophs
saplig_genus_summary <- functional_group_genus_summary %>% 
  
  filter(FUNCTIONAL.GROUP == "SAP.LIG")

## Non-ligninolytic saprotrophs
nonsaplig_genus_summary <- functional_group_genus_summary %>% 
  
  filter(FUNCTIONAL.GROUP == "SAP.NONLIG")

## ECM perox
ecmperox_genus_summary <- functional_group_genus_summary %>% 
  
  filter(FUNCTIONAL.GROUP == "ECM.PEROX")

## ECM without perox
ecmnoperox_genus_summary <- functional_group_genus_summary %>% 
  
  filter(FUNCTIONAL.GROUP == "ECM.NOPEROX")

#### Fig. S8A, sap lig ####
figs8a_saplig_summary <- ggplot() + 
  
  geom_bar(
    
    data = saplig_genus_summary, 
    
    aes(x = FUNCTIONAL.GROUP.LABEL, y = GENUS.PERC, fill = GENUS.LABEL), 
    
    colour = "black", 
    
    stat = "identity"
    
  ) + 
  
  ## Labels
  labs(
    
    title = "Ligninolytic saprotrophs", 
    
    x = NULL, 
    
    y = "% of sequences in functional group"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank()
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0.5), 
    
    axis.title.y = element_text(colour = "black", size = 8), 
    
    axis.title.x = element_text(colour = "black", size = 8), 
    
    axis.text.x = element_blank(), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks.y = element_line(colour = "black", size = 0.25), 
    
    axis.ticks.x = element_blank()
    
  ) + 

    guides(fill = guide_legend(title = "Genus"))

#### Fig. S8B, sap lig ####
figs8b_sapnonlig_summary <- ggplot() + 
  
  geom_bar(
    
    data = nonsaplig_genus_summary, 
    
    aes(x = FUNCTIONAL.GROUP.LABEL, y = GENUS.PERC, fill = GENUS.LABEL), 
    
    colour = "black", 
    
    stat = "identity"
    
  ) + 
  
  ## Labels
  labs(
    
    title = "Non-ligninolytic saprotrophs", 
    
    x = NULL, 
    
    y = "% of sequences in functional group"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    axis.line = element_line(colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank()
    
  ) + 
  
  ## Format text
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    axis.title.y = element_text(colour = "black", size = 8), 
    
    axis.title.x = element_text(colour = "black", size = 8), 
    
    axis.text.x = element_blank(), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks.y = element_line(colour = "black", size = 0.25), 
    
    axis.ticks.x = element_blank()
    
  ) + 

    guides(fill = guide_legend(title = "Genus"))

#### Fig. S8C, ECM perox ####
figs8c_ecmperox_summary <- ggplot() + 
  
  geom_bar(
    
    data = ecmperox_genus_summary, 
    
    aes(x = FUNCTIONAL.GROUP.LABEL, y = GENUS.PERC, fill = GENUS.LABEL), 
    
    colour = "black", 
    
    stat = "identity"
    
  ) + 
  
  ## Labels
  labs(
    
    title = "ECM with peroxidases", 
    
    x = NULL, 
    
    y = "% of sequences in functional group"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank()
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0.5), 
    
    axis.title.y = element_text(colour = "black", size = 8), 
    
    axis.title.x = element_text(colour = "black", size = 8), 
    
    axis.text.x = element_blank(), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks.y = element_line(colour = "black", size = 0.25), 
    
    axis.ticks.x = element_blank()
    
  ) + 

    guides(fill = guide_legend(title = "Genus"))

#### Fig. S8D, ECM no perox ####
figs8d_ecmnoperox_summary <- ggplot() + 
  
  geom_bar(
    
    data = ecmnoperox_genus_summary, 
    
    aes(x = FUNCTIONAL.GROUP.LABEL, y = GENUS.PERC, fill = GENUS.LABEL), 
    
    colour = "black", 
    
    stat = "identity"
    
  ) + 
  
  ## Labels
  labs(
    
    title = "ECM without peroxidases", 
    
    x = NULL, 
    
    y = "% of sequences in functional group"
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank()
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0.5), 
    
    axis.title.y = element_text(colour = "black", size = 8), 
    
    axis.title.x = element_text(colour = "black", size = 8), 
    
    axis.text.x = element_blank(), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks.y = element_line(colour = "black", size = 0.25), 
    
    axis.ticks.x = element_blank()
    
  ) + 

    guides(fill = guide_legend(title = "Genus"))

#### Assemble Fig S8 ####
figs8_functional_group_genus_summary <- ggarrange(
  
  figs8a_saplig_summary, 

  figs8b_sapnonlig_summary,

  figs8c_ecmperox_summary, 
  
  figs8d_ecmnoperox_summary, 
  
  nrow = 2,

  ncol = 2,

  align = "hv",

  common.legend = FALSE

)

#### Save figure ####
ggsave(

  "figures/MolEcol_revision/Figure_S8.pdf",

  figs8_functional_group_genus_summary,

  device = "pdf",

  width = 6.5,

  height = 10,

  units = "in"

)
```

```{r enzyme_gamms}
#### Cell fungi GAMM ####
cell_fungi_gamm <- gamm(
  
  CELL.ENZYMES ~ s(SAP.LIG.ABS, k = -1) + s(SAP.NONLIG.ABS, k = -1) + s(ECM.PEROX.ABS, k = -1) + s(ECM.NOPEROX.ABS, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

cell_fungi_gamm_summary <- summary(cell_fungi_gamm$gam)

cell_fungi_gamm_plot <- plot(cell_fungi_gamm$gam, residuals = TRUE)

#### Cell env GAMM ####
cell_env_gamm <- gamm(
  
  CELL.ENZYMES ~ s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

cell_env_gamm_summary <- summary(cell_env_gamm$gam)

cell_env_gamm_plot <- plot(cell_env_gamm$gam, residuals = TRUE)

#### Lig fungi GAMM ####
lig_fungi_gamm <- gamm(
  
  LIG.ENZYMES ~ s(SAP.LIG.ABS, k = -1) + s(SAP.NONLIG.ABS, k = -1) + s(ECM.PEROX.ABS, k = -1) + s(ECM.NOPEROX.ABS, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

lig_fungi_gamm_summary <- summary(lig_fungi_gamm$gam)

lig_fungi_gamm_plot <- plot(lig_fungi_gamm$gam, residuals = TRUE)

#### Lig env GAMM ####
lig_env_gamm <- gamm(
  
  LIG.ENZYMES ~ s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

lig_env_gamm_summary <- summary(lig_env_gamm$gam)

lig_env_gamm_plot <- plot(lig_env_gamm$gam, residuals = TRUE)

#### NAG fungi GAMM ####
nag_fungi_gamm <- gamm(
  
  NAG ~ s(SAP.LIG.ABS, k = -1) + s(SAP.NONLIG.ABS, k = -1) + s(ECM.PEROX.ABS, k = -1) + s(ECM.NOPEROX.ABS, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

nag_fungi_gamm_summary <- summary(nag_fungi_gamm$gam)

nag_fungi_gamm_plot <- plot(nag_fungi_gamm$gam, residuals = TRUE)

#### NAG env GAMM ####
nag_env_gamm <- gamm(
  
  NAG ~ s(PC1, k = -1) + s(PC2, k = -1), 
  
  correlation = corSpatial(form = ~LON + LAT, type = "gaussian"), 
  
  data = env_fungi_data
  
)

nag_env_gamm_summary <- summary(nag_env_gamm$gam)

nag_env_gamm_plot <- plot(nag_env_gamm$gam, residuals = TRUE)
```

```{r figs9_enzymes_fungi}
#### Cell ~ lig saps ####
cell_saplig_fig_data <- cell_fungi_gamm_plot[[1]]

cell_saplig_fit <- as_tibble(cell_saplig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_saplig_input <- as_tibble(cell_saplig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_saplig_gamm_stats <- tibble(
  
  PVAL = cell_fungi_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 2750000000, 
  
  YPOS = 1800) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

#### Cell ~ non-lig saps ####
cell_sapnonlig_fig_data <- cell_fungi_gamm_plot[[2]]

cell_sapnonlig_fit <- as_tibble(cell_sapnonlig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_sapnonlig_input <- as_tibble(cell_sapnonlig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_sapnonlig_gamm_stats <- tibble(
  
  PVAL = cell_fungi_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 750000000, 
  
  YPOS = 1800) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

#### Cell ~ ECM perox ####
cell_ecmperox_fig_data <- cell_fungi_gamm_plot[[3]]

cell_ecmperox_fit <- as_tibble(cell_ecmperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_ecmperox_input <- as_tibble(cell_ecmperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_ecmperox_gamm_stats <- tibble(
  
  PVAL = cell_fungi_gamm_summary$s.table[3, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1750000000, 
  
  YPOS = 1800) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

#### Cell ~ ECM no perox ####
cell_ecmnoperox_fig_data <- cell_fungi_gamm_plot[[4]]

cell_ecmnoperox_fit <- as_tibble(cell_ecmnoperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_ecmnoperox_input <- as_tibble(cell_ecmnoperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_ecmnoperox_gamm_stats <- tibble(
  
  PVAL = cell_fungi_gamm_summary$s.table[4, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1100000000, 
  
  YPOS = 1800) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

#### Lig ~ lig saps ####
lig_saplig_fig_data <- lig_fungi_gamm_plot[[1]]

lig_saplig_fit <- as_tibble(lig_saplig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_saplig_input <- as_tibble(lig_saplig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_saplig_gamm_stats <- tibble(
  
  PVAL = lig_fungi_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 2750000000, 
  
  YPOS = 11) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

#### Lig ~ non-lig saps ####
lig_sapnonlig_fig_data <- lig_fungi_gamm_plot[[2]]

lig_sapnonlig_fit <- as_tibble(lig_sapnonlig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_sapnonlig_input <- as_tibble(lig_sapnonlig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_sapnonlig_gamm_stats <- tibble(
  
  PVAL = lig_fungi_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 750000000, 
  
  YPOS = 11) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

#### Lig ~ ECM perox ####
lig_ecmperox_fig_data <- lig_fungi_gamm_plot[[3]]

lig_ecmperox_fit <- as_tibble(lig_ecmperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_ecmperox_input <- as_tibble(lig_ecmperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_ecmperox_gamm_stats <- tibble(
  
  PVAL = lig_fungi_gamm_summary$s.table[3, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1750000000, 
  
  YPOS = 11) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

#### Lig ~ ECM no perox ####
lig_ecmnoperox_fig_data <- lig_fungi_gamm_plot[[4]]

lig_ecmnoperox_fit <- as_tibble(lig_ecmnoperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_ecmnoperox_input <- as_tibble(lig_ecmnoperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_ecmnoperox_gamm_stats <- tibble(
  
  PVAL = lig_fungi_gamm_summary$s.table[4, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1100000000, 
  
  YPOS = 11) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

#### NAG ~ lig saps ####
nag_saplig_fig_data <- nag_fungi_gamm_plot[[1]]

nag_saplig_fit <- as_tibble(nag_saplig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_saplig_input <- as_tibble(nag_saplig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_saplig_gamm_stats <- tibble(
  
  PVAL = nag_fungi_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 750000000, 
  
  YPOS = 11000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

#### NAG ~ non-lig saps ####
nag_sapnonlig_fig_data <- nag_fungi_gamm_plot[[2]]

nag_sapnonlig_fit <- as_tibble(nag_sapnonlig_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_sapnonlig_input <- as_tibble(nag_sapnonlig_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_sapnonlig_gamm_stats <- tibble(
  
  PVAL = nag_fungi_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 750000000, 
  
  YPOS = 11000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "Non-ligninolytic\nsaprotrophs") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

#### NAG ~ ECM perox ####
nag_ecmperox_fig_data <- nag_fungi_gamm_plot[[3]]

nag_ecmperox_fit <- as_tibble(nag_ecmperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_ecmperox_input <- as_tibble(nag_ecmperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_ecmperox_gamm_stats <- tibble(
  
  PVAL = nag_fungi_gamm_summary$s.table[3, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1750000000, 
  
  YPOS = 11000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM with\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

#### NAG ~ ECM no perox ####
nag_ecmnoperox_fig_data <- nag_fungi_gamm_plot[[4]]

nag_ecmnoperox_fit <- as_tibble(nag_ecmnoperox_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_ecmnoperox_input <- as_tibble(nag_ecmnoperox_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_ecmnoperox_gamm_stats <- tibble(
  
  PVAL = nag_fungi_gamm_summary$s.table[4, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 900000000, 
  
  YPOS = 11000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(FUNCTIONAL.GROUP.LABEL = "ECM without\nperoxidases") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

#### Combine input data ####
enzyme_fungi_fit <- bind_rows(
  
  cell_saplig_fit, 
  
  cell_sapnonlig_fit, 
  
  cell_ecmperox_fit, 
  
  cell_ecmnoperox_fit, 
  
  lig_saplig_fit, 
  
  lig_sapnonlig_fit, 
  
  lig_ecmperox_fit, 
  
  lig_ecmnoperox_fit, 
  
  nag_saplig_fit, 
  
  nag_sapnonlig_fit, 
  
  nag_ecmperox_fit, 
  
  nag_ecmnoperox_fit
  
) %>% 
  
  mutate(
    
    ENZYME.LABEL = factor(
      
      ENZYME.LABEL, 
      
      levels = c(
        
        "Cellulolytic\nenzymes", 
        
        "Ligninolytic\nenzymes", 
        
        "N-acetyl-\nglucosaminidase"
        
      )
      
    )
    
  ) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  ) %>% 
  
  mutate(
    
    fit = ifelse(ENZYME.LABEL == "Cellulolytic\nenzymes", NA, fit), 
    
    se = ifelse(ENZYME.LABEL == "Cellulolytic\nenzymes", NA, se), 
    
    fit = ifelse(ENZYME.LABEL == "Ligninolytic\nenzymes", NA, fit), 
    
    se = ifelse(ENZYME.LABEL == "Ligninolytic\nenzymes", NA, se), 
    
    fit = ifelse(FUNCTIONAL.GROUP.LABEL == "ECM with\nperoxidases", NA, fit), 
    
    se = ifelse(FUNCTIONAL.GROUP.LABEL == "ECM with\nperoxidases", NA, se)
    
  )

enzyme_fungi_input <- bind_rows(
  
  cell_saplig_input, 
  
  cell_sapnonlig_input, 
  
  cell_ecmperox_input, 
  
  cell_ecmnoperox_input, 
  
  lig_saplig_input, 
  
  lig_sapnonlig_input, 
  
  lig_ecmperox_input, 
  
  lig_ecmnoperox_input, 
  
  nag_saplig_input, 
  
  nag_sapnonlig_input, 
  
  nag_ecmperox_input, 
  
  nag_ecmnoperox_input
  
) %>% 
  
  mutate(
    
    ENZYME.LABEL = factor(
      
      ENZYME.LABEL, 
      
      levels = c(
        
        "Cellulolytic\nenzymes", 
        
        "Ligninolytic\nenzymes", 
        
        "N-acetyl-\nglucosaminidase"
        
      )
      
    )
    
  ) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  )

enzyme_fungi_gamm_stats <- bind_rows(
  
  cell_saplig_gamm_stats, 
  
  cell_sapnonlig_gamm_stats, 
  
  cell_ecmperox_gamm_stats, 
  
  cell_ecmnoperox_gamm_stats, 
  
  lig_saplig_gamm_stats, 
  
  lig_sapnonlig_gamm_stats, 
  
  lig_ecmperox_gamm_stats, 
  
  lig_ecmnoperox_gamm_stats, 
  
  nag_saplig_gamm_stats, 
  
  nag_sapnonlig_gamm_stats, 
  
  nag_ecmperox_gamm_stats, 
  
  nag_ecmnoperox_gamm_stats
  
) %>% 
  
  mutate(
    
    ENZYME.LABEL = factor(
      
      ENZYME.LABEL, 
      
      levels = c(
        
        "Cellulolytic\nenzymes", 
        
        "Ligninolytic\nenzymes", 
        
        "N-acetyl-\nglucosaminidase"
        
      )
      
    )
    
  ) %>% 
  
  mutate(
    
    FUNCTIONAL.GROUP.LABEL = factor(
      
      FUNCTIONAL.GROUP.LABEL, 
      
      levels = c(
        
        "Ligninolytic\nsaprotrophs", 
        
        "Non-ligninolytic\nsaprotrophs", 
        
        "ECM with\nperoxidases", 
        
        "ECM without\nperoxidases"
        
      )
      
    )
    
  )

#### Figure S9 ####
enzyme_fungi_figs9 <- ggplot() + 
  
  ## CI
  geom_ribbon(
    
    data = enzyme_fungi_fit, 
    
    aes(x = x, ymin = fit - se, ymax = fit + se, y = NULL), 
    
    fill = "grey80"
    
  ) + 
  
  ## Points
  geom_point(
    
    data = enzyme_fungi_input, 
    
    aes(x = raw, y = p.resid, colour = FUNCTIONAL.GROUP.LABEL), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Upper CI
  geom_line(

    data = enzyme_fungi_fit,

    aes(x = x, y = fit + se, colour = FUNCTIONAL.GROUP.LABEL),

    linetype = 2

  ) +
  
  ## Lower CI
  geom_line(

    data = enzyme_fungi_fit,

    aes(x = x, y = fit - se, colour = FUNCTIONAL.GROUP.LABEL), 

    linetype = 2

  ) + 
  
  ## Line
  geom_line(
    
    data = enzyme_fungi_fit, 
    
    aes(x = x, y = fit, colour = FUNCTIONAL.GROUP.LABEL)
    
  ) + 
  
  ## Color
  scale_colour_manual(
    
    values = c("#443A83FF", "#287C8EFF", "#31B57BFF", "#B9DE28FF"), 
    
    guide = "none"
    
  ) + 
  
  ## P values
  geom_text(
    
    data = enzyme_fungi_gamm_stats, 
    
    aes(x = XPOS, y = YPOS, label = PVAL.LABEL), 
    
    parse = TRUE, 
    
    stat = "identity", 
    
    size = 3
    
  ) + 
  
  ## Titles
  labs(
    
    x = bquote('Fungal functional group abundance (ITS copies'%.%g^-1~'fine root litter)'), 
    
    y = "Partial enzyme activity"
    
  ) + 
  
  scale_x_continuous(
    
    breaks = breaks_extended(n = 3), 
    
    limits = c(0, NA), 
    
    expand = c(0.1, 0.1), 
    
    labels = scientific_10
    
  ) + 
  
  ## Wrap
  facet_grid(
    
    cols = vars(FUNCTIONAL.GROUP.LABEL), 
    
    rows = vars(ENZYME.LABEL), 
    
    scales = "free"
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 10), 
    
    axis.title.y = element_text(colour = "black", size = 10),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )

## Save figure
ggsave2(

  "figures/MolEcol_revision/Figure_S9.pdf",

  enzyme_fungi_figs9,

  device = "pdf",

  width = 6.5,

  height = 5,

  units = "in"

)
```

```{r figs10_enzymes_env}
#### Cell ~ PC1 ####
cell_pc1_fig_data <- cell_env_gamm_plot[[1]]

cell_pc1_fit <- as_tibble(cell_pc1_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_pc1_input <- as_tibble(cell_pc1_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_pc1_gamm_stats <- tibble(
  
  PVAL = cell_env_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1, 
  
  YPOS = 1500) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

#### Cell ~ PC2 ####
cell_pc2_fig_data <- cell_env_gamm_plot[[2]]

cell_pc2_fit <- as_tibble(cell_pc2_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_pc2_input <- as_tibble(cell_pc2_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

cell_pc2_gamm_stats <- tibble(
  
  PVAL = cell_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.75, 
  
  YPOS = 1500) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "Cellulolytic\nenzymes")

#### Lig ~ PC1 ####
lig_pc1_fig_data <- lig_env_gamm_plot[[1]]

lig_pc1_fit <- as_tibble(lig_pc1_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_pc1_input <- as_tibble(lig_pc1_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_pc1_gamm_stats <- tibble(
  
  PVAL = lig_env_gamm_summary$s.table[1, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1, 
  
  YPOS = 11) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

#### Lig ~ PC2 ####
lig_pc2_fig_data <- lig_env_gamm_plot[[2]]

lig_pc2_fit <- as_tibble(lig_pc2_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_pc2_input <- as_tibble(lig_pc2_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

lig_pc2_gamm_stats <- tibble(
  
  PVAL = lig_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.75, 
  
  YPOS = 11) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "Ligninolytic\nenzymes")

#### NAG ~ PC1 ####
nag_pc1_fig_data <- nag_env_gamm_plot[[1]]

nag_pc1_fit <- as_tibble(nag_pc1_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_pc1_input <- as_tibble(nag_pc1_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_pc1_gamm_stats <- tibble(
  
  PVAL = nag_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 1, 
  
  YPOS = 11000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(ENV.LABEL = "Environ. PC1") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

#### NAG ~ PC2 ####
nag_pc2_fig_data <- nag_env_gamm_plot[[2]]

nag_pc2_fit <- as_tibble(nag_pc2_fig_data[c("x", "se", "fit")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_pc2_input <- as_tibble(nag_pc2_fig_data[c("raw", "p.resid")]) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

nag_pc2_gamm_stats <- tibble(
  
  PVAL = nag_env_gamm_summary$s.table[2, 4], 
  
  PVAL.LABEL = NA, 
  
  XPOS = 0.75, 
  
  YPOS = 11000) %>% 
  
  mutate(PVAL.LABEL = sapply(PVAL, gamm_pval, simplify = TRUE)) %>% 
  
  mutate(ENV.LABEL = "Environ. PC2") %>% 
  
  mutate(ENZYME.LABEL = "N-acetyl-\nglucosaminidase")

enzyme_env_input <- bind_rows(
  
  cell_pc1_input, 
  
  cell_pc2_input, 
  
  lig_pc1_input, 
  
  lig_pc2_input, 
  
  nag_pc1_input, 
  
  nag_pc2_input
  
) %>% 
  
  mutate(
    
    ENZYME.LABEL = factor(
      
      ENZYME.LABEL, 
      
      levels = c(
        
        "Cellulolytic\nenzymes", 
        
        "Ligninolytic\nenzymes", 
        
        "N-acetyl-\nglucosaminidase"
        
      )
      
    )
    
  ) %>% 
  
  mutate(
    
    ENV.LABEL = factor(
      
      ENV.LABEL, 
      
      levels = c(
        
        "Environ. PC1", 
        
        "Environ. PC2"
        
      )
      
    )
    
  )

enzyme_env_gamm_stats <- bind_rows(
  
  cell_pc1_gamm_stats, 
  
  cell_pc2_gamm_stats,  
  
  lig_pc1_gamm_stats, 
  
  lig_pc2_gamm_stats, 
  
  nag_pc1_gamm_stats, 
  
  nag_pc2_gamm_stats
  
) %>% 
  
  mutate(
    
    ENZYME.LABEL = factor(
      
      ENZYME.LABEL, 
      
      levels = c(
        
        "Cellulolytic\nenzymes", 
        
        "Ligninolytic\nenzymes", 
        
        "N-acetyl-\nglucosaminidase"
        
      )
      
    )
    
  ) %>% 
  
  mutate(
    
    ENV.LABEL = factor(
      
      ENV.LABEL, 
      
      levels = c(
        
        "Environ. PC1", 
        
        "Environ. PC2"
        
      )
      
    )
    
  )

#### Figure S10 ####
enzyme_fungi_figs10 <- ggplot() + 
  
  ## Points
  geom_point(
    
    data = enzyme_env_input, 
    
    aes(x = raw, y = p.resid, colour = ENV.LABEL), 
    
    pch = 1, 
    
    size = 1
    
  ) + 
  
  ## Color
  scale_colour_manual(
    
    values = c("#501D4CFF", "#EC4B3EFF"), 
    
    guide = "none"
    
  ) + 
  
  ## P values
  geom_text(

    data = enzyme_env_gamm_stats,

    aes(x = XPOS, y = YPOS, label = PVAL.LABEL),

    parse = TRUE,

    stat = "identity",

    size = 3

  ) +
  
  ## Titles
  labs(
    
    x = "Environmental gradient", 
    
    y = "Partial enzyme activity"
    
  ) + 
  
  ## Wrap
  facet_grid(
    
    cols = vars(ENV.LABEL), 
    
    rows = vars(ENZYME.LABEL), 
    
    scales = "free"
    
  ) + 
  
  # Format
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank(), 
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0), 
    
    axis.title.x = element_text(colour = "black", size = 10), 
    
    axis.title.y = element_text(colour = "black", size = 10),  
    
    axis.text.x = element_text(colour = "black", size = 8), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25), 
    
    strip.background = element_blank(), 
    
    strip.text = element_text(colour = "black", size = 10)
    
  )

## Save figure
ggsave2(

  "figures/MolEcol_revision/Figure_S10.pdf",

  enzyme_fungi_figs10,

  device = "pdf",

  width = 3.75,

  height = 4.5,

  units = "in"

)
```

```{r enzyme_gamm_checks}
gam.check(cell_fungi_gamm$gam)

gam.check(lig_fungi_gamm$gam)

gam.check(nag_fungi_gamm$gam)

gam.check(cell_env_gamm$gam)

gam.check(lig_env_gamm$gam)

gam.check(nag_env_gamm$gam)
```

```{r figs11_average_root_biochemistry}
#### Format individual points data ####
pre_root_pygcms_source_input <- pre_root_pygcms_source %>% 
  
  pivot_longer(
    
    cols = AROMATIC : UNKNOWN.ORIGIN, 
    
    names_to = "COMPOUND.SOURCE", 
    
    values_to = "PROP") %>% 
  
  group_by(COMPOUND.SOURCE) %>% 
  
  mutate(PROP.MEAN = mean(PROP)) %>% 
  
  ungroup() %>%
  
  arrange(desc(PROP.MEAN)) %>% 
  
  mutate(
    
    SOURCE.LABEL = NA, 
    
    SOURCE.LABEL = ifelse(COMPOUND.SOURCE == "AROMATIC", "Aromatic", SOURCE.LABEL), 
    
    SOURCE.LABEL = ifelse(COMPOUND.SOURCE == "LIGNIN", "Lignin", SOURCE.LABEL), 
    
    SOURCE.LABEL = ifelse(COMPOUND.SOURCE == "LIPID", "Lipids", SOURCE.LABEL), 
    
    SOURCE.LABEL = ifelse(COMPOUND.SOURCE == "N.BEARING", "N-bearing", SOURCE.LABEL), 
    
    SOURCE.LABEL = ifelse(COMPOUND.SOURCE == "PHENOL", "Phenol", SOURCE.LABEL), 
    
    SOURCE.LABEL = ifelse(COMPOUND.SOURCE == "POLYSACCHARIDE", "Polysaccharides", SOURCE.LABEL), 
    
    SOURCE.LABEL = ifelse(COMPOUND.SOURCE == "PROTEIN", "Proteins", SOURCE.LABEL), 
    
    SOURCE.LABEL = ifelse(COMPOUND.SOURCE == "UNKNOWN.ORIGIN", "Unknown origin", SOURCE.LABEL), 
    
    SOURCE.LABEL = factor(SOURCE.LABEL, levels = unique(SOURCE.LABEL))
    
  )

#### Fig. S11 root biochemistry ####
figs11_root_biochem <- ggplot() +
  
  ## Points
  geom_point(
    
    data = pre_root_pygcms_source_input,
    
    aes(x = SOURCE.LABEL, y = PROP, colour = SOURCE.LABEL), 
    
    pch = 21, 
    
    position = position_jitter(width = 0.1),
    
    size = 1
    
  ) +
  
  geom_boxplot(
    
    data = pre_root_pygcms_source_input,
    
    aes(x = SOURCE.LABEL, y = PROP, colour = SOURCE.LABEL), 
    
    fill = NA, 
    
    outlier.shape = NA
    
  ) + 
  
  ## Scale error bar colors
  scale_colour_viridis(
    
    option = "plasma", 
    
    discrete = TRUE, 
    
    guide = "none"
    
  ) + 
  
  
  ## Add plot and axis titles
  labs(
    
    title = "Root biochemistry",
    
    x = "Compound class",
    
    y = "Relative abundance"
    
  ) + 
  
  scale_y_continuous(
    
    limits = c(0, 0.6)
    
  ) + 
  
  ## Format background and axis lines
  theme(
    
    panel.border = element_rect(fill = NA, colour = "black", size = 0.5), 
    
    panel.background = element_rect(fill = NA), 
    
    panel.grid.major = element_blank(), 
    
    panel.grid.minor = element_blank()
    
  ) + 
  
  ## Format text
  theme(
    
    plot.title = element_text(colour = "black", size = 12, hjust = 0.5), 
    
    axis.title.y = element_text(colour = "black", size = 8), 
    
    axis.title.x = element_text(colour = "black", size = 8), 
    
    axis.text.x = element_text(colour = "black", size = 8, angle = 45, hjust = 1), 
    
    axis.text.y = element_text(colour = "black", size = 8), 
    
    axis.ticks = element_line(colour = "black", size = 0.25)
    
  ) + 
  
  ## Format legends
  theme( 
    
    legend.key = element_blank(), 
    
    legend.title = element_text(size = 10), 
    
    legend.text = element_text(size = 8)
    
  )

#### Save figure ####
ggsave(

  "figures/MolEcol_revision/Figure_S11.pdf",

  figs11_root_biochem,

  device = "pdf",

  width = 6.5,

  height = 4.5,

  units = "in"

)
```

```{r summed_root_biochem}
root_biochem_totals <- env_fungi_data %>% 
  
  select(PLOT, PRE.ROOT.MEAN.AROMATIC : PRE.ROOT.MEAN.UNKNOWN.ORIGIN) %>% 
  
  pivot_longer(
    
    cols = PRE.ROOT.MEAN.AROMATIC : PRE.ROOT.MEAN.UNKNOWN.ORIGIN, 
    
    names_to = "CLASS", 
    
    values_to = "PROP"
    
  ) %>% 
  
  group_by(PLOT) %>% 
  
  summarise(TOTAL = sum(PROP))
```

